<html>
<head>
    <title>KOC vs KOSALT</title>
    <meta charset="UTF-8">
    <style>
    body {display:flex; height:100%; width: 100%; margin:0; background-color:white; font-family:"Segoe UI",Arial,sans-serif;}
    #wrapper{display:flex; position: absolute; width: 100%; height: 100%; margin:0; min-height:0;}
    #title_bar_container{position:fixed;height:15%;width:100%;background-color:black;color:white;opacity:0.3}
    #side_bar_container{}
        #side_bar_container > ul{list-style-type:none;padding:0; margin:0; border-right: solid 2px black}
            ul > li:first-child {padding: 0 0 0 0; margin-bottom: -4px;}
            ul > li:first-child:hover {filter: opacity(50%)}
            ul > li.side_bar_header {background-color: #616161!important; color:white}
            ul > li{background-color:  #cc0000; padding: 8px; 16px;}
            ul > li:hover{background-color: #777777;}
                li > a{color:white; text-decoration:none; display:block;}

     h1 {margin: 0; font-weight:400;}
     h3 {margin: 0; font-weight:400;}

    .report_header{color:white; margin: 0 -1.5em; padding: 1em; background-color:#616161!important;}
    #report_body_container{position: relative;}
        .plot_sections{background-color:white; position: relative; margin-bottom:3em;}
        #report_body_container > div:first-child {background-color:white; position: relative; margin-bottom: 0;}

            .plots{background-color: white; margin-bottom:1.5em;}
            .plot_legend{display:none; padding: 15px; color:black; border: solid black; border-width: 0 1px 1px 1px;}
            .plot_description{display:none; padding: 15px; border: solid black; border-width: 0 1px 1px 1px;}
            .r_code{display:none; position:relative; background-color:white;font-family: "Lucida Console", Monaco, monospace; font-size: 12px; max-height: 90%; overflow-y: scroll; overflow-x: scroll; white-space:nowrap; padding:15px; border: solid black; border-width: 0 1px 1px 1px;}
                .r_code > span.r_comment {color: blue; display: inline;}
            img {display: inline; width: 40%;}
                p.plot_title {display: inline; position: absolute; padding: 4%; font-weight: bold;}
             h3.sub_title {display: block}
            .text:last-child{padding: 0 0 50px 0;}

    .collapsible {color: white; background-color: #999999; cursor: pointer; padding: 5px; width: 100%; text-align: center; font-size: 15px; border: 1px solid black; margin-top: 5px;}

    .collapsible:hover {background-color: #cccccc; color: black;}

    .active.collapsible {background-color: #cccccc; color:black;}
    .active.collapsible:hover {background-color: #cccccc; color:black;}

    .plot_title {cursor: pointer;}
    .plot_tite:hover {color: #999999}
    .plot_title.active {color: #999999;}
    </style>
    <!-- move css to html head -->
</head>
<body>
    <!-- start of report-->
    <div id="wrapper">
        <div id="side_bar_container" style="width:20%;overflow-y:auto;"><ul><li><a href=#tp><img src=plots/logo.png style=width:100%;></a></li><li class="side_bar_header">Expression</li><li><a href="#Distribution_of_Expression_Values">Distribution of Expression Values
</a></li>
<li><a href="#PCA_Contribution_of_Components">PCA Contribution of Components
</a></li>
<li><a href="#PCA_Scatterplots">PCA Scatterplots
</a></li>
<li><a href="#Correlation_Analysis_Heatmap">Correlation Analysis Heatmap
</a></li>
<li class="side_bar_header">Differential Expression</li><li><a href="#Volcano_Plot">Volcano Plot
</a></li>
<li><a href="#MA_Plot">MA Plot
</a></li>
<li><a href="#Number_of_Significant_Genes">Number of Significant Genes
</a></li>
<li><a href="#Significant_Genes_Heatmap">Significant Genes Heatmap
</a></li>
<li><a href="#Most_Differential_Genes_Tables">Most Differential Genes Tables
</a></li>
<li><a href="#Most_Differential_Genes_Boxplots">Most Differential Genes Boxplots
</a></li>
<li class="side_bar_header">Over Representation Analysis</li><li><a href="#Enrichment_Summary">Enrichment Summary
</a></li>
<li><a href="#Enrichment_Boxplots">Enrichment Boxplots
</a></li>
<li><a href="#Enrichment_Network_Plots">Enrichment Network Plots
</a></li>
<li><a href="#Underenrichment_Barcharts">Underenrichment Barcharts
</a></li>
<li class="side_bar_header">Spatial Analysis</li><li><a href="#Spatial_Enrichment_Table">Spatial Enrichment Table
</a></li>
<li><a href="#Spatial_Enrichment_Distribution_Plots">Spatial Enrichment Distribution Plots
</a></li>
<li class="side_bar_header">Methodology</li><li><a href=#Core Workflow>Core Workflow</a></li><li><a href=#Spatial Analysis>Spatial Analysis</a></li><li><a href=#Over Representation Analysis>Over Representation Analysis</a></li><li class="side_bar_header">Extras</li><li><a href=#Citation>Citation</a></li></ul></div>
        <div id="report_body_container" style="width:80%;padding:0 1.5em;overflow-y:auto;"><div id="tp"></div>
<div class="report_header">
    <h1>Rapid analysis and visualisation for bulk RNA-seq, psuedo-bulk RNA-seq, GeoMx and Proteomic datasets</h1><br>
    <h3>Glasgow Bioinformatic Core (2023)</h3><br>
</div>
<br><div id="Distribution_of_Expression_Values" class="plot_sections">
    <h2 class="plot_title">Distribution of Expression Values
</h2>
    <div class="plot_wrapper" style="display: block;">
        <div class="plots"><img style="width:16.666666666666664%;" src="plots/distribution_of_expression_values/KOSALT12_S4_001_distribution_of_expression_values.png
"/><img style="width:16.666666666666664%;" src="plots/distribution_of_expression_values/KOSALT15_S10_001_distribution_of_expression_values.png
"/><img style="width:16.666666666666664%;" src="plots/distribution_of_expression_values/KOSALT19_S8_001_distribution_of_expression_values.png
"/><img style="width:16.666666666666664%;" src="plots/distribution_of_expression_values/KOC12_S3_001_distribution_of_expression_values.png
"/><img style="width:16.666666666666664%;" src="plots/distribution_of_expression_values/KOC15_S9_001_distribution_of_expression_values.png
"/><img style="width:16.666666666666664%;" src="plots/distribution_of_expression_values/KOC19_S6_001_distribution_of_expression_values.png
"/></div>

        <!-- plot_description is replaced by bin/plot_description/... -->
        <button class="collapsible">plot description</button>
        <div class="plot_description">Here we show for each sample the distribution of expression values. Generally these are expected to look highly similar across all samples regardless of sample group, and so are useful for assessing the quality of the experiment technically. Outlier samples with atypical distributions of expression values can be indicative of technical issues with that sample âˆ’ such as insufficient quantities of RNA during library preparation.
</div>

        <!-- plot_legend is replaced by bin/figure_legends/... -->
        <button class="collapsible">figure legend</button>
        <div class="plot_legend"><p>Density plots showing the overall distribution of expression values for each sample. Expression values are given on x-axis and are on a log 10 scale. The gene density is given on the y-axis.
</p></div>

        <!-- r_code is replaced by -->
        <button class="collapsible" id="r_code">r code</button>
        <div class="r_code"><span class="r_comment">##---- general libraries ----####</span><br><br>library(ggplot2)<br>library(ggrepel)<br>library(reshape)<br>library(ggridges)<br><br><span class="r_comment">##---- DE sample and group names ----####</span><br><br>samples = c("KOSALT12_S4_001","KOSALT15_S10_001","KOSALT19_S8_001","KOC12_S3_001","KOC15_S9_001","KOC19_S6_001")<br>sample_groups = c("KOSALT","KOC")<br>sample_groups = factor(sample_groups, levels = sample_groups)<br>sample_groupings = c("KOSALT","KOSALT","KOSALT","KOC","KOC","KOC")<br>sample_groupings = factor(sample_groupings, levels = sample_groups)<br>samples_by_sample_group = list(c("KOSALT12_S4_001","KOSALT15_S10_001","KOSALT19_S8_001"),c("KOC12_S3_001","KOC15_S9_001","KOC19_S6_001"))<br>comparisons = c("KOC vs KOSALT")<br><br><span class="r_comment">##---- sets the working directory. If you move the SL2 folder please update this path ----####</span><br><br>setwd("/home/john/Downloads/mahesh_run2/2x75/SL2/all_genes/de_workflows/KOC_vs_KOSALT")<br><br><span class="r_comment">##---- de input files ----####</span><br><br>de_annotated = read.table(file="data/de_annotated.csv", header=TRUE,row.names = 2, sep='\t', quote='',check.names = TRUE)<br>de_annotated_sig = read.table(file="data/de_annotated_significant.csv", header=TRUE,row.names = 2, sep='\t', quote='',check.names = TRUE)<br><br><span class="r_comment">##---- de parse ne matrix ----####</span><br><br>ne_matrix = de_annotated[,samples]<br>ne_matrix_sig = de_annotated_sig[,samples]<br><br><span class="r_comment">##---- SL2 Theme ----####</span><br><br>theme_SL2 <- function () { <br> &emsp; &emsp;theme_bw() %+replace% <br> &emsp; &emsp; &emsp; &emsp;theme(<br>	 &emsp; &emsp;panel.grid = element_blank(),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;panel.background = element_blank(),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;panel.border = element_rect(colour = "black", fill=NA, linewidth=1),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.background = element_blank(), <br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.background = element_rect(fill="transparent", colour=NA),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.key = element_rect(fill="transparent", colour=NA),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.title = element_text(size=12, margin = margin(b = 5),hjust=0,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;title = element_text(size = 12, margin = margin(b = 5),hjust=0,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.text.y = element_text(size = 10, margin = margin(r = 5),hjust=1,vjust=0.5, face="bold",colour="black"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.text.x = element_text(size = 10, margin = margin(t = 5),hjust=0.5,vjust=1, face="bold",colour="black"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.title.y = element_text(size = 11, margin = margin(r = 10),angle = 90,hjust=0.5,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.title.x = element_text(size = 11, margin = margin(t = 10),hjust=0.5,vjust=1, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.text=element_text(size=11, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.title=element_blank(), <br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.key.size=unit(2.5,"line"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.margin=unit(c(0.4,0.4,0.4,0.4), "cm")<br> &emsp; &emsp; &emsp; &emsp;)<br>}<br><br><span class="r_comment">##----- Default GGplot Colours Function -----####</span><br><br>gg_color_hue <- function(n)<br>{<br> &emsp;# default colour blind palettes<br> &emsp;cblind_palette = c("#88CCEE", "#CC6677", "#DDCC77", "#117733", "#332288", "#AA4499", "#44AA99", "#999933", "#882255", "#661100", "#6699CC", "#888888")<br><br> &emsp;# append ggplot colours if groups goes above colour blind palette length<br> &emsp;if (n <= length(cblind_palette))<br> &emsp;{<br> &emsp; &emsp;return(cblind_palette[1:n])<br> &emsp;}<br> &emsp;else<br> &emsp;{<br> &emsp; &emsp;nn = n - length(cblind_palette)<br> &emsp; &emsp;hues = seq(15, 375, length = nn + 1)<br> &emsp; &emsp;return(append(cblind_palette,hcl(h = hues, l = 65, c = 100)[1:nn]))<br> &emsp;}<br>}<br><br><span class="r_comment">##---- Default Sample and Sample Group Colours &emsp;----####</span><br><br>number_of_sample_groups = length(sample_groups)<br>default_sample_group_colours = gg_color_hue(number_of_sample_groups)<br>default_samples_colours = c(default_sample_group_colours[1],default_sample_group_colours[1],default_sample_group_colours[1],default_sample_group_colours[2],default_sample_group_colours[2],default_sample_group_colours[2])<br><br><span class="r_comment">##---- Default Sample Labels &emsp;----####</span><br><br>default_sample_labels = gsub("_"," ",c("KOSALT12_S4_001","KOSALT15_S10_001","KOSALT19_S8_001","KOC12_S3_001","KOC15_S9_001","KOC19_S6_001")) # note: changing this won't change the order the samples appear on the plots. Merely what they are labelled as.<br><br><span class="r_comment">##---- Save Plot Function -----####</span><br><br>save_plot <- function(ggp,plot_height,plot_width,plot_path)<br>{<br> &emsp;png(plot_path, height=plot_height, width=plot_width, pointsize=5)<br> &emsp;print(ggp)<br> &emsp;dev.off()<br><br> &emsp;svg(gsub(".png", ".svg", plot_path), height=plot_height/90, width=plot_width/90)<br> &emsp;print(ggp)<br> &emsp;dev.off()<br><br> &emsp;# clears all devices - as a safety measure<br> &emsp;while (dev.cur()>1) dev.off()<br>}<br><br><span class="r_comment">##---- Distribution of Expression Values Function ----####</span><br><br>make_distribution_of_expression_values_plot <- function(plot_sample, plot_sample_label, plot_colour,transparency, line_thickness, x_axis_label, y_axis_label, legend_position)<br>{<br> &emsp;ggp = ggplot(ne_matrix, aes(log10(x=ne_matrix[[plot_sample]]+0.001))) +<br> &emsp; &emsp;geom_density(colour=plot_colour,fill=plot_colour, alpha=transparency,size=line_thickness) +<br> &emsp; &emsp;labs(x=x_axis_label, y=y_axis_label, title=plot_sample_label) +<br> &emsp; &emsp;scale_x_continuous(expand=c(0,0)) +<br> &emsp; &emsp;scale_y_continuous(expand=c(0,0)) +<br> &emsp; &emsp;theme_SL2() + <br> &emsp; &emsp;theme(legend.position=legend_position)<br><br> &emsp;return(ggp)<br>}<br><br><span class="r_comment">##---- Distribution of Expression Values ----####</span><br><br>plot_height = 250<br>plot_width = 250<br>transparency = 0.5<br>line_thickness = 1<br>x_axis_label = expression("expression (log"[10]* ")")<br>y_axis_label = "density"<br>legend_position = "none"<br>plot_labels = default_sample_labels<br><br>for(plot_sample_index in 1:length(samples))<br>{<br> &emsp;plot_sample = samples[plot_sample_index]<br> &emsp;plot_sample_label = plot_labels[plot_sample_index]<br> &emsp;plot_colour = default_samples_colours[plot_sample_index]<br> &emsp;ggp = make_distribution_of_expression_values_plot(plot_sample, plot_sample_label, plot_colour, transparency, line_thickness, x_axis_label, y_axis_label, legend_position)<br> &emsp;save_plot(ggp,plot_height,plot_width,paste("plots/distribution_of_expression_values/",plot_sample,"_distribution_of_expression_values.png",sep=""))<br>}<br><br></div>
    </div>
</div>

<div id="PCA_Contribution_of_Components" class="plot_sections">
    <h2 class="plot_title">PCA Contribution of Components
</h2>
    <div class="plot_wrapper" style="display: block;">
        <div class="plots"><img style="width:25.0%;" src="plots/PCA/PCA_contribution_of_components.png
"/></div>

        <!-- plot_description is replaced by bin/plot_description/... -->
        <button class="collapsible">plot description</button>
        <div class="plot_description">In simple terms, Principal Components Analysis (PCA) is a technique that uses all expression values to find underlying variables (known as principal components) that best differentiate your samples. Principal components are dimensions along which your data points are most spread out. Each component will describe a proportion of the total underlying variation between genes and samples (i.e. the global variation in the dataset). Components are ordered by the % of the underlying variation that they describe, with PC1 explaining the most variation.
<br />
<br />
A PCA contribution of components plot shows what % of the global variation in the dataset is explained by each component. This is useful for determining how many biological groups there are in the data and how big an impact on the data each biological group has. For example if there were two biological groups with very strong differences, and no other biological groups we might expect a large proportion of variance in PC1 and a low proportion in all other components.
</div>

        <!-- plot_legend is replaced by bin/figure_legends/... -->
        <button class="collapsible">figure legend</button>
        <div class="plot_legend"><p>Gene expression data Principal Component Analysis (PCA) contribution of components waterfall plot. Showing all components in ascending order on the x-axis and the contribution of each component on the y-axis. To control for over representation of very highly expressed genes, all gene expression values were scaled on a gene by gene basis using the z-score transformation, prior to the PCA. 
</p></div>

        <!-- r_code is replaced by -->
        <button class="collapsible" id="r_code">r code</button>
        <div class="r_code"><span class="r_comment">##---- general libraries ----####</span><br><br>library(ggplot2)<br>library(ggrepel)<br>library(reshape)<br>library(ggridges)<br><br><span class="r_comment">##---- DE sample and group names ----####</span><br><br>samples = c("KOSALT12_S4_001","KOSALT15_S10_001","KOSALT19_S8_001","KOC12_S3_001","KOC15_S9_001","KOC19_S6_001")<br>sample_groups = c("KOSALT","KOC")<br>sample_groups = factor(sample_groups, levels = sample_groups)<br>sample_groupings = c("KOSALT","KOSALT","KOSALT","KOC","KOC","KOC")<br>sample_groupings = factor(sample_groupings, levels = sample_groups)<br>samples_by_sample_group = list(c("KOSALT12_S4_001","KOSALT15_S10_001","KOSALT19_S8_001"),c("KOC12_S3_001","KOC15_S9_001","KOC19_S6_001"))<br>comparisons = c("KOC vs KOSALT")<br><br><span class="r_comment">##---- sets the working directory. If you move the SL2 folder please update this path ----####</span><br><br>setwd("/home/john/Downloads/mahesh_run2/2x75/SL2/all_genes/de_workflows/KOC_vs_KOSALT")<br><br><span class="r_comment">##---- de input files ----####</span><br><br>de_annotated = read.table(file="data/de_annotated.csv", header=TRUE,row.names = 2, sep='\t', quote='',check.names = TRUE)<br>de_annotated_sig = read.table(file="data/de_annotated_significant.csv", header=TRUE,row.names = 2, sep='\t', quote='',check.names = TRUE)<br><br><span class="r_comment">##---- de parse ne matrix ----####</span><br><br>ne_matrix = de_annotated[,samples]<br>ne_matrix_sig = de_annotated_sig[,samples]<br><br><span class="r_comment">##---- transpose and scale ne matrix ----####</span><br><br>ne_matrix_transposed = data.frame(t(ne_matrix))<br>colnames(ne_matrix_transposed) = rownames(ne_matrix)<br><br>ne_matrix_scaled = data.frame(t(scale(t(ne_matrix))))<br>rownames(ne_matrix_scaled) = rownames(ne_matrix)<br>ne_matrix_scaled[do.call(cbind, lapply(ne_matrix_scaled, is.nan))] <- 0<br>ne_matrix_scaled = ne_matrix_scaled[is.finite(rowSums(ne_matrix_scaled)), ]<br><br>ne_matrix_scaled_transposed = data.frame(t(ne_matrix_scaled))<br>colnames(ne_matrix_scaled_transposed) = rownames(ne_matrix_scaled)<br><br><span class="r_comment">##---- SL2 Theme ----####</span><br><br>theme_SL2 <- function () { <br> &emsp; &emsp;theme_bw() %+replace% <br> &emsp; &emsp; &emsp; &emsp;theme(<br>	 &emsp; &emsp;panel.grid = element_blank(),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;panel.background = element_blank(),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;panel.border = element_rect(colour = "black", fill=NA, linewidth=1),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.background = element_blank(), <br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.background = element_rect(fill="transparent", colour=NA),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.key = element_rect(fill="transparent", colour=NA),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.title = element_text(size=12, margin = margin(b = 5),hjust=0,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;title = element_text(size = 12, margin = margin(b = 5),hjust=0,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.text.y = element_text(size = 10, margin = margin(r = 5),hjust=1,vjust=0.5, face="bold",colour="black"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.text.x = element_text(size = 10, margin = margin(t = 5),hjust=0.5,vjust=1, face="bold",colour="black"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.title.y = element_text(size = 11, margin = margin(r = 10),angle = 90,hjust=0.5,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.title.x = element_text(size = 11, margin = margin(t = 10),hjust=0.5,vjust=1, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.text=element_text(size=11, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.title=element_blank(), <br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.key.size=unit(2.5,"line"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.margin=unit(c(0.4,0.4,0.4,0.4), "cm")<br> &emsp; &emsp; &emsp; &emsp;)<br>}<br><br><span class="r_comment">##----- Default GGplot Colours Function -----####</span><br><br>gg_color_hue <- function(n)<br>{<br> &emsp;# default colour blind palettes<br> &emsp;cblind_palette = c("#88CCEE", "#CC6677", "#DDCC77", "#117733", "#332288", "#AA4499", "#44AA99", "#999933", "#882255", "#661100", "#6699CC", "#888888")<br><br> &emsp;# append ggplot colours if groups goes above colour blind palette length<br> &emsp;if (n <= length(cblind_palette))<br> &emsp;{<br> &emsp; &emsp;return(cblind_palette[1:n])<br> &emsp;}<br> &emsp;else<br> &emsp;{<br> &emsp; &emsp;nn = n - length(cblind_palette)<br> &emsp; &emsp;hues = seq(15, 375, length = nn + 1)<br> &emsp; &emsp;return(append(cblind_palette,hcl(h = hues, l = 65, c = 100)[1:nn]))<br> &emsp;}<br>}<br><br><span class="r_comment">##---- Default Sample and Sample Group Colours &emsp;----####</span><br><br>number_of_sample_groups = length(sample_groups)<br>default_sample_group_colours = gg_color_hue(number_of_sample_groups)<br>default_samples_colours = c(default_sample_group_colours[1],default_sample_group_colours[1],default_sample_group_colours[1],default_sample_group_colours[2],default_sample_group_colours[2],default_sample_group_colours[2])<br><br><span class="r_comment">##---- Default Sample Labels &emsp;----####</span><br><br>default_sample_labels = gsub("_"," ",c("KOSALT12_S4_001","KOSALT15_S10_001","KOSALT19_S8_001","KOC12_S3_001","KOC15_S9_001","KOC19_S6_001")) # note: changing this won't change the order the samples appear on the plots. Merely what they are labelled as.<br><br><span class="r_comment">##---- Default Sample Group Labels &emsp;----####</span><br><br>default_sample_group_labels = gsub("_"," ",c("KOSALT","KOC")) # note: changing this won't change the order the groups appear on the plots. Merely what they are labelled as.<br><br><span class="r_comment">##---- Save Plot Function -----####</span><br><br>save_plot <- function(ggp,plot_height,plot_width,plot_path)<br>{<br> &emsp;png(plot_path, height=plot_height, width=plot_width, pointsize=5)<br> &emsp;print(ggp)<br> &emsp;dev.off()<br><br> &emsp;svg(gsub(".png", ".svg", plot_path), height=plot_height/90, width=plot_width/90)<br> &emsp;print(ggp)<br> &emsp;dev.off()<br><br> &emsp;# clears all devices - as a safety measure<br> &emsp;while (dev.cur()>1) dev.off()<br>}<br><br><span class="r_comment">##---- PCA Contribution of Components Function ----####</span><br><br>make_PCA_contribution_of_components_plot <- function(x_axis_label,y_axis_label,dot_size,dot_transparency,dot_colour,line_type,line_colour,line_size)<br>{<br><br> &emsp;prcomp_data = prcomp(as.matrix(sapply(ne_matrix_scaled_transposed, as.numeric)))<br> &emsp;vars = apply(prcomp_data$x, 2, var)<br> &emsp;prcomp_props = data.frame(round(vars / sum(vars),4) * 100)<br> &emsp;prcomp_props$PC = row.names(prcomp_props)<br> &emsp;colnames(prcomp_props) = c("prop","PC")<br> &emsp;prcomp_props$PC = factor(prcomp_props$PC, levels = row.names(prcomp_props))<br><br> &emsp;ggp = ggplot(data=prcomp_props,aes(x=PC,y=prop,group=1)) +<br> &emsp; &emsp;geom_line(linetype=line_type,colour=line_colour,size=line_size) +<br> &emsp; &emsp;geom_point(size=dot_size,alpha=dot_transparency,colour=dot_colour) +<br> &emsp; &emsp;xlab(x_axis_label) +<br> &emsp; &emsp;ylab(y_axis_label) +<br> &emsp; &emsp;theme_SL2() +<br> &emsp; &emsp;theme(legend.position="None", legend.title = element_blank())<br><br> &emsp;return(ggp)<br>}<br><br><span class="r_comment">##---- PCA (Contribution of Components) ----####</span><br><br>plot_height = 300<br>plot_width = 400<br>x_axis_label = "component"<br>y_axis_label = "proportion of variance (%)"<br>dot_size = 4<br>dot_transparency = 1<br>dot_colour = "red"<br>line_type = "solid"<br>line_colour = "black"<br>line_size = 1<br><br>ggp = make_PCA_contribution_of_components_plot(x_axis_label,y_axis_label,dot_size,dot_transparency,dot_colour,line_type,line_colour,line_size)<br>save_plot(ggp,plot_height,plot_width,"plots/PCA/PCA_contribution_of_components.png")<br><br></div>
    </div>
</div>

<div id="PCA_Scatterplots" class="plot_sections">
    <h2 class="plot_title">PCA Scatterplots
</h2>
    <div class="plot_wrapper" style="display: block;">
        <div class="plots"><img style="width:25.0%;" src="plots/PCA/PCA_1_vs_PCA_2_scatter_plot_labelled.png
"/><img style="width:25.0%;" src="plots/PCA/PCA_1_vs_PCA_2_scatter_plot_unlabelled.png
"/><img style="width:25.0%;" src="plots/PCA/PCA_3_vs_PCA_4_scatter_plot_labelled.png
"/><img style="width:25.0%;" src="plots/PCA/PCA_3_vs_PCA_4_scatter_plot_unlabelled.png
"/></div>

        <!-- plot_description is replaced by bin/plot_description/... -->
        <button class="collapsible">plot description</button>
        <div class="plot_description">In simple terms, Principal Components Analysis (PCA) is a technique that uses all expression values to find underlying variables (known as principal components) that best differentiate your samples. Principal components are dimensions along which your data points are most spread out. Each component will describe a proportion of the total underlying variation between genes and samples (i.e. the global variation in the dataset). Components are ordered by the % of the underlying variation that they describe, with PC1 explaining the most variation.
<br />
<br />
Here we plot each sample based on its PC1 vs PC2 values (left and middle left) and its PC3 vs PC4 values (middle right and right). Generally we expect sample groups to separate and replicates to cluster together. Groups not separating suggests little difference between groups, and replicates not clustering suggests sample heterogeneity. 
</div>

        <!-- plot_legend is replaced by bin/figure_legends/... -->
        <button class="collapsible">figure legend</button>
        <div class="plot_legend"><p>Gene expression data Principal Component Analysis (PCA) scatterplots. showing PC1 vs PC2 (left) and PC3 vs PC4 (right). Individual samples are represented by dots. The percentage of total variation explained by each component is given in the x and y-axis as appropriate. To control over representation of very highly expressed genes, all gene expression values were scaled on a gene by gene basis using the z-score transformation, prior to the PCA.  
</p></div>

        <!-- r_code is replaced by -->
        <button class="collapsible" id="r_code">r code</button>
        <div class="r_code"><span class="r_comment">##---- general libraries ----####</span><br><br>library(ggplot2)<br>library(ggrepel)<br>library(reshape)<br>library(ggridges)<br><br><span class="r_comment">##---- DE sample and group names ----####</span><br><br>samples = c("KOSALT12_S4_001","KOSALT15_S10_001","KOSALT19_S8_001","KOC12_S3_001","KOC15_S9_001","KOC19_S6_001")<br>sample_groups = c("KOSALT","KOC")<br>sample_groups = factor(sample_groups, levels = sample_groups)<br>sample_groupings = c("KOSALT","KOSALT","KOSALT","KOC","KOC","KOC")<br>sample_groupings = factor(sample_groupings, levels = sample_groups)<br>samples_by_sample_group = list(c("KOSALT12_S4_001","KOSALT15_S10_001","KOSALT19_S8_001"),c("KOC12_S3_001","KOC15_S9_001","KOC19_S6_001"))<br>comparisons = c("KOC vs KOSALT")<br><br><span class="r_comment">##---- sets the working directory. If you move the SL2 folder please update this path ----####</span><br><br>setwd("/home/john/Downloads/mahesh_run2/2x75/SL2/all_genes/de_workflows/KOC_vs_KOSALT")<br><br><span class="r_comment">##---- de input files ----####</span><br><br>de_annotated = read.table(file="data/de_annotated.csv", header=TRUE,row.names = 2, sep='\t', quote='',check.names = TRUE)<br>de_annotated_sig = read.table(file="data/de_annotated_significant.csv", header=TRUE,row.names = 2, sep='\t', quote='',check.names = TRUE)<br><br><span class="r_comment">##---- de parse ne matrix ----####</span><br><br>ne_matrix = de_annotated[,samples]<br>ne_matrix_sig = de_annotated_sig[,samples]<br><br><span class="r_comment">##---- transpose and scale ne matrix ----####</span><br><br>ne_matrix_transposed = data.frame(t(ne_matrix))<br>colnames(ne_matrix_transposed) = rownames(ne_matrix)<br><br>ne_matrix_scaled = data.frame(t(scale(t(ne_matrix))))<br>rownames(ne_matrix_scaled) = rownames(ne_matrix)<br>ne_matrix_scaled[do.call(cbind, lapply(ne_matrix_scaled, is.nan))] <- 0<br>ne_matrix_scaled = ne_matrix_scaled[is.finite(rowSums(ne_matrix_scaled)), ]<br><br>ne_matrix_scaled_transposed = data.frame(t(ne_matrix_scaled))<br>colnames(ne_matrix_scaled_transposed) = rownames(ne_matrix_scaled)<br><br><span class="r_comment">##---- SL2 Theme ----####</span><br><br>theme_SL2 <- function () { <br> &emsp; &emsp;theme_bw() %+replace% <br> &emsp; &emsp; &emsp; &emsp;theme(<br>	 &emsp; &emsp;panel.grid = element_blank(),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;panel.background = element_blank(),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;panel.border = element_rect(colour = "black", fill=NA, linewidth=1),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.background = element_blank(), <br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.background = element_rect(fill="transparent", colour=NA),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.key = element_rect(fill="transparent", colour=NA),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.title = element_text(size=12, margin = margin(b = 5),hjust=0,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;title = element_text(size = 12, margin = margin(b = 5),hjust=0,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.text.y = element_text(size = 10, margin = margin(r = 5),hjust=1,vjust=0.5, face="bold",colour="black"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.text.x = element_text(size = 10, margin = margin(t = 5),hjust=0.5,vjust=1, face="bold",colour="black"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.title.y = element_text(size = 11, margin = margin(r = 10),angle = 90,hjust=0.5,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.title.x = element_text(size = 11, margin = margin(t = 10),hjust=0.5,vjust=1, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.text=element_text(size=11, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.title=element_blank(), <br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.key.size=unit(2.5,"line"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.margin=unit(c(0.4,0.4,0.4,0.4), "cm")<br> &emsp; &emsp; &emsp; &emsp;)<br>}<br><br><span class="r_comment">##----- Default GGplot Colours Function -----####</span><br><br>gg_color_hue <- function(n)<br>{<br> &emsp;# default colour blind palettes<br> &emsp;cblind_palette = c("#88CCEE", "#CC6677", "#DDCC77", "#117733", "#332288", "#AA4499", "#44AA99", "#999933", "#882255", "#661100", "#6699CC", "#888888")<br><br> &emsp;# append ggplot colours if groups goes above colour blind palette length<br> &emsp;if (n <= length(cblind_palette))<br> &emsp;{<br> &emsp; &emsp;return(cblind_palette[1:n])<br> &emsp;}<br> &emsp;else<br> &emsp;{<br> &emsp; &emsp;nn = n - length(cblind_palette)<br> &emsp; &emsp;hues = seq(15, 375, length = nn + 1)<br> &emsp; &emsp;return(append(cblind_palette,hcl(h = hues, l = 65, c = 100)[1:nn]))<br> &emsp;}<br>}<br><br><span class="r_comment">##---- Default Sample and Sample Group Colours &emsp;----####</span><br><br>number_of_sample_groups = length(sample_groups)<br>default_sample_group_colours = gg_color_hue(number_of_sample_groups)<br>default_samples_colours = c(default_sample_group_colours[1],default_sample_group_colours[1],default_sample_group_colours[1],default_sample_group_colours[2],default_sample_group_colours[2],default_sample_group_colours[2])<br><br><span class="r_comment">##---- Default Sample Labels &emsp;----####</span><br><br>default_sample_labels = gsub("_"," ",c("KOSALT12_S4_001","KOSALT15_S10_001","KOSALT19_S8_001","KOC12_S3_001","KOC15_S9_001","KOC19_S6_001")) # note: changing this won't change the order the samples appear on the plots. Merely what they are labelled as.<br><br><span class="r_comment">##---- Default Sample Group Labels &emsp;----####</span><br><br>default_sample_group_labels = gsub("_"," ",c("KOSALT","KOC")) # note: changing this won't change the order the groups appear on the plots. Merely what they are labelled as.<br><br><span class="r_comment">##---- Save Plot Function -----####</span><br><br>save_plot <- function(ggp,plot_height,plot_width,plot_path)<br>{<br> &emsp;png(plot_path, height=plot_height, width=plot_width, pointsize=5)<br> &emsp;print(ggp)<br> &emsp;dev.off()<br><br> &emsp;svg(gsub(".png", ".svg", plot_path), height=plot_height/90, width=plot_width/90)<br> &emsp;print(ggp)<br> &emsp;dev.off()<br><br> &emsp;# clears all devices - as a safety measure<br> &emsp;while (dev.cur()>1) dev.off()<br>}<br><br><span class="r_comment">##---- PCA Scatter Plot Function ----####</span><br><br>make_PCA_scatterplot <- function(plot_sample_groupings,comp_x,comp_y,x_axis_label,y_axis_label,plot_sample_group_colours,plot_sample_group_labels,plot_sample_labels,dot_size,dot_transparency,legend_position,sample_label_size,show_proportion_of_variance, show_sample_labels)<br>{<br><br> &emsp;# Do PCA<br> &emsp;prcomp_data = prcomp(as.matrix(sapply(ne_matrix_scaled_transposed, as.numeric)))<br> &emsp;prcomp_coordinates = data.frame(prcomp_data$x)<br><br> &emsp;# Get variance<br> &emsp;if (show_proportion_of_variance == TRUE)<br> &emsp;{<br> &emsp; &emsp;vars = apply(prcomp_data$x, 2, var)<br> &emsp; &emsp;prop_x = round(vars[comp_x] / sum(vars),4) * 100<br> &emsp; &emsp;prop_y = round(vars[comp_y] / sum(vars),4) * 100<br><br> &emsp; &emsp;x_axis_label = paste(x_axis_label, " (",prop_x,"%)",sep="")<br> &emsp; &emsp;y_axis_label = paste(y_axis_label, " (",prop_y,"%)",sep="")<br> &emsp;}<br><br> &emsp;# Prepare Table<br> &emsp;prcomp_coordinates = prcomp_coordinates[,c(component_x, component_y)]<br> &emsp;names(prcomp_coordinates) = c("comp_x","comp_y")<br><br> &emsp;# labels<br> &emsp;if (show_sample_labels == TRUE)<br> &emsp;{<br> &emsp; &emsp;plot_sample_labels = plot_sample_labels<br> &emsp;}<br> &emsp;else<br> &emsp;{<br> &emsp; &emsp;plot_sample_labels = NA<br> &emsp;}<br><br> &emsp;#make plot<br> &emsp;ggp = ggplot(data=prcomp_coordinates, aes(x=comp_x, y=comp_y, colour=plot_sample_groupings)) +<br> &emsp; &emsp;geom_point(size=dot_size,alpha=dot_transparency) +<br> &emsp; &emsp;geom_label_repel(aes(label=plot_sample_labels), size=sample_label_size, force = 1,box.padding = 1.5, label.padding = unit(0.5, "lines"), show.legend = FALSE, colour = "black") +<br> &emsp; &emsp;scale_color_manual(values=plot_sample_group_colours,labels=plot_sample_group_labels, name="") +<br> &emsp; &emsp;xlim(c(min(prcomp_coordinates$comp_x)*1.25,max(prcomp_coordinates$comp_x)*1.25)) +<br> &emsp; &emsp;ylim(c(min(prcomp_coordinates$comp_y)*1.25,max(prcomp_coordinates$comp_y)*1.25)) +<br> &emsp; &emsp;labs(x=x_axis_label, y=y_axis_label) +<br> &emsp; &emsp;theme_SL2() +<br> &emsp; &emsp;theme(legend.position=legend_position, legend.spacing.x = unit(0.25, 'cm'))<br><br> &emsp;return(ggp) <br>}<br><br><span class="r_comment">##---- PCA (Scatter Plots) ----####</span><br><br>plot_height = 300<br>plot_width = 300<br>plot_sample_group_colours = default_sample_group_colours<br>plot_sample_group_labels = default_sample_group_labels<br>plot_sample_labels = default_sample_labels<br>plot_sample_groupings = sample_groupings<br>dot_size = 4<br>dot_transparency = 1<br>legend_position = "bottom"<br>sample_label_size = 3.5<br>show_proportion_of_variance = TRUE<br>show_sample_labels = TRUE<br><br>component_x = 1<br>component_y = 2<br>x_axis_label = "PC1"<br>y_axis_label = "PC2"<br>ggp = make_PCA_scatterplot(plot_sample_groupings,component_x,component_y,x_axis_label,y_axis_label,plot_sample_group_colours,plot_sample_group_labels,plot_sample_labels,dot_size,dot_transparency,legend_position,sample_label_size,show_proportion_of_variance, show_sample_labels)<br>save_plot(ggp,plot_height,plot_width,"plots/PCA/PCA_1_vs_PCA_2_scatter_plot_labelled.png")<br><br>show_sample_labels = FALSE<br>ggp = make_PCA_scatterplot(plot_sample_groupings,component_x,component_y,x_axis_label,y_axis_label,plot_sample_group_colours,plot_sample_group_labels,plot_sample_labels,dot_size,dot_transparency,legend_position,sample_label_size,show_proportion_of_variance, show_sample_labels)<br>save_plot(ggp,plot_height,plot_width,"plots/PCA/PCA_1_vs_PCA_2_scatter_plot_unlabelled.png")<br><br>component_x = 3<br>component_y = 4<br>x_axis_label = "PC3"<br>y_axis_label = "PC4"<br>show_sample_labels = TRUE<br>ggp = make_PCA_scatterplot(plot_sample_groupings,component_x,component_y,x_axis_label,y_axis_label,plot_sample_group_colours,plot_sample_group_labels,plot_sample_labels,dot_size,dot_transparency,legend_position,sample_label_size,show_proportion_of_variance, show_sample_labels)<br>save_plot(ggp,plot_height,plot_width,"plots/PCA/PCA_3_vs_PCA_4_scatter_plot_labelled.png")<br><br>show_sample_labels = FALSE<br>ggp = make_PCA_scatterplot(plot_sample_groupings,component_x,component_y,x_axis_label,y_axis_label,plot_sample_group_colours,plot_sample_group_labels,plot_sample_labels,dot_size,dot_transparency,legend_position,sample_label_size,show_proportion_of_variance, show_sample_labels)<br>save_plot(ggp,plot_height,plot_width,"plots/PCA/PCA_3_vs_PCA_4_scatter_plot_unlabelled.png")<br><br></div>
    </div>
</div>

<div id="Correlation_Analysis_Heatmap" class="plot_sections">
    <h2 class="plot_title">Correlation Analysis Heatmap
</h2>
    <div class="plot_wrapper" style="display: block;">
        <div class="plots"><img style="width:50.0%;" src="plots/correlation_analysis/correlation_analysis_heatmap.png
"/><img style="width:50.0%;" src="plots/correlation_analysis/correlation_analysis_heatmap_clustered.png
"/></div>

        <!-- plot_description is replaced by bin/plot_description/... -->
        <button class="collapsible">plot description</button>
        <div class="plot_description">Here we provide a sample by sample correlation analysis heatmap. The correlation analysis heatmap highlights the differences and similarity between samples based on how strongly or weakly they correlate with each other across all gene expression values. Generally we would expect replicates to have similar expression patterns across all genes, and so we would expect replicates to have high correlation values. Similarly we might expect very different sample groups to have very different expression patterns and so to have low correlation values. This is reflected in the heatmap, which shows all possible combinations of correlations (Spearman Correlation Coefficient) between two samples. The diagonal line indicates that when comparing a sample to itself we should always get a correlation of 1. 
<br />
<br />
The left and right plots are identical, however in the right hand plot the samples have been hierarchically clustered, so that more similar samples are grouped together on the x and y axis. We would expect replicate samples to group after clustering and cohorts to separate. Hierarchical clustering can therefore provide an extra level of evidence for sample similarity / dissimilarity and can help to clarify results visually.

</div>

        <!-- plot_legend is replaced by bin/figure_legends/... -->
        <button class="collapsible">figure legend</button>
        <div class="plot_legend"><p>Heatmap of sample correlation. Correlations between each pairwise combination of samples is shown. The correlations were calculated using Spearman Correlation based on all gene expression values. The level of correlation (Spearman Correlation Coefficient) is represented by colour intensity, with strong positive correlation in red, no correlation in light pink and strong anti-correlation in blue. The right had plot has been hierarchically clustered on both axis using, Spearman distances, with UPMGA agglomeration and mean reordering.
</p></div>

        <!-- r_code is replaced by -->
        <button class="collapsible" id="r_code">r code</button>
        <div class="r_code"><span class="r_comment">##---- general libraries ----####</span><br><br>library(ggplot2)<br>library(ggrepel)<br>library(reshape)<br>library(ggridges)<br><br><span class="r_comment">##---- heatmap libraries ----####</span><br><br>library(amap)<br><br><span class="r_comment">##---- DE sample and group names ----####</span><br><br>samples = c("KOSALT12_S4_001","KOSALT15_S10_001","KOSALT19_S8_001","KOC12_S3_001","KOC15_S9_001","KOC19_S6_001")<br>sample_groups = c("KOSALT","KOC")<br>sample_groups = factor(sample_groups, levels = sample_groups)<br>sample_groupings = c("KOSALT","KOSALT","KOSALT","KOC","KOC","KOC")<br>sample_groupings = factor(sample_groupings, levels = sample_groups)<br>samples_by_sample_group = list(c("KOSALT12_S4_001","KOSALT15_S10_001","KOSALT19_S8_001"),c("KOC12_S3_001","KOC15_S9_001","KOC19_S6_001"))<br>comparisons = c("KOC vs KOSALT")<br><br><span class="r_comment">##---- sets the working directory. If you move the SL2 folder please update this path ----####</span><br><br>setwd("/home/john/Downloads/mahesh_run2/2x75/SL2/all_genes/de_workflows/KOC_vs_KOSALT")<br><br><span class="r_comment">##---- de input files ----####</span><br><br>de_annotated = read.table(file="data/de_annotated.csv", header=TRUE,row.names = 2, sep='\t', quote='',check.names = TRUE)<br>de_annotated_sig = read.table(file="data/de_annotated_significant.csv", header=TRUE,row.names = 2, sep='\t', quote='',check.names = TRUE)<br><br><span class="r_comment">##---- de parse ne matrix ----####</span><br><br>ne_matrix = de_annotated[,samples]<br>ne_matrix_sig = de_annotated_sig[,samples]<br><br><span class="r_comment">##---- transpose and scale ne matrix ----####</span><br><br>ne_matrix_transposed = data.frame(t(ne_matrix))<br>colnames(ne_matrix_transposed) = rownames(ne_matrix)<br><br>ne_matrix_scaled = data.frame(t(scale(t(ne_matrix))))<br>rownames(ne_matrix_scaled) = rownames(ne_matrix)<br>ne_matrix_scaled[do.call(cbind, lapply(ne_matrix_scaled, is.nan))] <- 0<br>ne_matrix_scaled = ne_matrix_scaled[is.finite(rowSums(ne_matrix_scaled)), ]<br><br>ne_matrix_scaled_transposed = data.frame(t(ne_matrix_scaled))<br>colnames(ne_matrix_scaled_transposed) = rownames(ne_matrix_scaled)<br><br><span class="r_comment">##---- SL2 Theme ----####</span><br><br>theme_SL2 <- function () { <br> &emsp; &emsp;theme_bw() %+replace% <br> &emsp; &emsp; &emsp; &emsp;theme(<br>	 &emsp; &emsp;panel.grid = element_blank(),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;panel.background = element_blank(),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;panel.border = element_rect(colour = "black", fill=NA, linewidth=1),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.background = element_blank(), <br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.background = element_rect(fill="transparent", colour=NA),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.key = element_rect(fill="transparent", colour=NA),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.title = element_text(size=12, margin = margin(b = 5),hjust=0,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;title = element_text(size = 12, margin = margin(b = 5),hjust=0,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.text.y = element_text(size = 10, margin = margin(r = 5),hjust=1,vjust=0.5, face="bold",colour="black"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.text.x = element_text(size = 10, margin = margin(t = 5),hjust=0.5,vjust=1, face="bold",colour="black"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.title.y = element_text(size = 11, margin = margin(r = 10),angle = 90,hjust=0.5,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.title.x = element_text(size = 11, margin = margin(t = 10),hjust=0.5,vjust=1, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.text=element_text(size=11, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.title=element_blank(), <br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.key.size=unit(2.5,"line"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.margin=unit(c(0.4,0.4,0.4,0.4), "cm")<br> &emsp; &emsp; &emsp; &emsp;)<br>}<br><br><span class="r_comment">##---- Default Three Tone Heatmap Colours &emsp;----####</span><br><br>default_three_tone_heatmap_colours = c("purple","black","yellow")<br><br><span class="r_comment">##---- Default Sample Labels &emsp;----####</span><br><br>default_sample_labels = gsub("_"," ",c("KOSALT12_S4_001","KOSALT15_S10_001","KOSALT19_S8_001","KOC12_S3_001","KOC15_S9_001","KOC19_S6_001")) # note: changing this won't change the order the samples appear on the plots. Merely what they are labelled as.<br><br><span class="r_comment">##---- Save Plot Function -----####</span><br><br>save_plot <- function(ggp,plot_height,plot_width,plot_path)<br>{<br> &emsp;png(plot_path, height=plot_height, width=plot_width, pointsize=5)<br> &emsp;print(ggp)<br> &emsp;dev.off()<br><br> &emsp;svg(gsub(".png", ".svg", plot_path), height=plot_height/90, width=plot_width/90)<br> &emsp;print(ggp)<br> &emsp;dev.off()<br><br> &emsp;# clears all devices - as a safety measure<br> &emsp;while (dev.cur()>1) dev.off()<br>}<br><br><span class="r_comment">##----- Clustering Method -----####</span><br><br>cluster_matrix <- function(mx,cluster_x,cluster_y,distance_method,clustering_method,reorder_function)<br>{<br> &emsp;x.cluster = hclust(Dist(t(mx), method=distance_method), method=clustering_method)<br> &emsp;y.cluster = hclust(Dist(mx, method=distance_method), method=clustering_method)<br> &emsp;<br> &emsp;x.dd = as.dendrogram(x.cluster)<br> &emsp;y.dd = as.dendrogram(y.cluster)<br> &emsp;x.dd.reorder = reorder(x.dd,0,FUN=reorder_function)<br> &emsp;y.dd.reorder = reorder(y.dd,0,FUN=reorder_function)<br> &emsp;x.order = order.dendrogram(x.dd.reorder)<br> &emsp;y.order = order.dendrogram(y.dd.reorder)<br><br> &emsp;if(cluster_x == TRUE && cluster_y == TRUE) <br> &emsp;{<br> &emsp; &emsp; &emsp;mx_clustered = mx[y.order,x.order]<br> &emsp;}<br> &emsp;<br> &emsp;if(cluster_x == TRUE && cluster_y == FALSE)<br> &emsp;{<br> &emsp; &emsp; &emsp;mx_clustered = mx[,x.order]<br> &emsp;}<br> &emsp;<br> &emsp;if(cluster_x == FALSE && cluster_y == TRUE)<br> &emsp;{<br> &emsp; &emsp; &emsp;mx_clustered = mx[y.order,]<br> &emsp;}<br> &emsp;if(cluster_x == FALSE && cluster_y == FALSE) <br> &emsp;{<br> &emsp; &emsp; &emsp;mx_clustered = mx<br> &emsp;}<br> &emsp;return(mx_clustered)<br>}<br><br><span class="r_comment">##---- Correlation Analysis Heatmap Function &emsp;----####</span><br><br>make_correlation_analysis_heatmap <- function(colours,sample_names,cluster_x,cluster_y,distance_method,clustering_method,reorder_function)<br>{<br> &emsp;ne_matrix_scaled_correlated = cor(ne_matrix_scaled,method="spearman")<br> &emsp;row.names(ne_matrix_scaled_correlated) = sample_names<br> &emsp;colnames(ne_matrix_scaled_correlated) = sample_names<br> &emsp;<br> &emsp;ne_matrix_scaled_correlated_clustered = cluster_matrix(ne_matrix_scaled_correlated,cluster_x,cluster_y,distance_method,clustering_method,reorder_function)<br> &emsp;ne_matrix_scaled_correlated_clustered_melted <- melt(ne_matrix_scaled_correlated_clustered)<br> &emsp;ne_matrix_scaled_correlated_clustered_melted$X1 = factor(ne_matrix_scaled_correlated_clustered_melted$X1, levels=colnames(ne_matrix_scaled_correlated_clustered))<br> &emsp;ne_matrix_scaled_correlated_clustered_melted$X2 = factor(ne_matrix_scaled_correlated_clustered_melted$X2, levels=row.names(ne_matrix_scaled_correlated_clustered))<br><br> &emsp;ggp = ggplot(ne_matrix_scaled_correlated_clustered_melted, aes(x = X1, y = X2, fill = value)) + <br> &emsp; &emsp;geom_tile() + <br> &emsp; &emsp;scale_fill_gradientn(limits=c(-1, 1),colours = colorRampPalette(colours)(100)) + <br> &emsp; &emsp;ylab('') + <br> &emsp; &emsp;xlab('') + <br> &emsp; &emsp;theme_SL2() + <br> &emsp; &emsp;theme(legend.position="right", legend.title = element_blank(), legend.spacing.x = unit(0.25, 'cm'),axis.text.x = element_text(angle = 90, hjust = 1), axis.ticks=element_blank()) + <br> &emsp; &emsp;scale_x_discrete(expand=c(0,0)) + <br> &emsp; &emsp;scale_y_discrete(expand=c(0,0))<br> &emsp;<br> &emsp;return(ggp)<br>}<br><br><span class="r_comment">##---- Correlation Analysis Heatmap ----####</span><br><br>plot_height = 600<br>plot_width = 750<br>colours = default_three_tone_heatmap_colours<br>sample_names = default_sample_labels<br>distance_method = "spearman"<br>clustering_method = "average"<br>reorder_function = "average"<br>cluster_x = FALSE<br>cluster_y = FALSE<br><br>ggp = make_correlation_analysis_heatmap(colours,sample_names,cluster_x,cluster_y,distance_method,clustering_method,reorder_function)<br>save_plot(ggp,plot_height,plot_width,"plots/correlation_analysis/correlation_analysis_heatmap.png")<br><br>cluster_x = TRUE<br>cluster_y = TRUE<br>ggp = make_correlation_analysis_heatmap(colours,sample_names,cluster_x,cluster_y,distance_method,clustering_method,reorder_function)<br>save_plot(ggp,plot_height,plot_width,"plots/correlation_analysis/correlation_analysis_heatmap_clustered.png")<br><br></div>
    </div>
</div>

<div id="Volcano_Plot" class="plot_sections">
    <h2 class="plot_title">Volcano Plot
</h2>
    <div class="plot_wrapper" style="display: block;">
        <div class="plots"><img style="width:50.0%;" src="plots/volcano_plot/volcano_plot_labelled.png
"/><img style="width:50.0%;" src="plots/volcano_plot/volcano_plot_unlabelled.png
"/></div>

        <!-- plot_description is replaced by bin/plot_description/... -->
        <button class="collapsible">plot description</button>
        <div class="plot_description">Here we provide a volcano plot. The volcano plot shows the relationship between the fold change and the associated pâˆ’value for each gene. Typically it is expected to see a consistent relationship between fold changes and pâˆ’values across all genes. In a volcano plot, the absence of a consistent relationship between fold changes and pâˆ’values may suggest problems with the methods used to call differential expression. Volcano plots also provide an idea of the scale, and balance between direction of the changes between two conditions. The left and right plots are identical, except the gene names have been removed from the right plot.
</div>

        <!-- plot_legend is replaced by bin/figure_legends/... -->
        <button class="collapsible">figure legend</button>
        <div class="plot_legend"><p>Volcano plot for the comparison between koc and kosalt. Significantly differential  genes (p.adj < 0.05, absolute log2 fold > 1.0) are shown in red and nonâˆ’significant genes in black. A positive fold change indicates higher expression in koc than in kosalt.
</p></div>

        <!-- r_code is replaced by -->
        <button class="collapsible" id="r_code">r code</button>
        <div class="r_code"><span class="r_comment">##---- general libraries ----####</span><br><br>library(ggplot2)<br>library(ggrepel)<br>library(reshape)<br>library(ggridges)<br><br><span class="r_comment">##---- DE sample and group names ----####</span><br><br>samples = c("KOSALT12_S4_001","KOSALT15_S10_001","KOSALT19_S8_001","KOC12_S3_001","KOC15_S9_001","KOC19_S6_001")<br>sample_groups = c("KOSALT","KOC")<br>sample_groups = factor(sample_groups, levels = sample_groups)<br>sample_groupings = c("KOSALT","KOSALT","KOSALT","KOC","KOC","KOC")<br>sample_groupings = factor(sample_groupings, levels = sample_groups)<br>samples_by_sample_group = list(c("KOSALT12_S4_001","KOSALT15_S10_001","KOSALT19_S8_001"),c("KOC12_S3_001","KOC15_S9_001","KOC19_S6_001"))<br>comparisons = c("KOC vs KOSALT")<br><br><span class="r_comment">##---- sets the working directory. If you move the SL2 folder please update this path ----####</span><br><br>setwd("/home/john/Downloads/mahesh_run2/2x75/SL2/all_genes/de_workflows/KOC_vs_KOSALT")<br><br><span class="r_comment">##---- de input files ----####</span><br><br>de_annotated = read.table(file="data/de_annotated.csv", header=TRUE,row.names = 2, sep='\t', quote='',check.names = TRUE)<br>de_annotated_sig = read.table(file="data/de_annotated_significant.csv", header=TRUE,row.names = 2, sep='\t', quote='',check.names = TRUE)<br><br><span class="r_comment">##---- SL2 Theme ----####</span><br><br>theme_SL2 <- function () { <br> &emsp; &emsp;theme_bw() %+replace% <br> &emsp; &emsp; &emsp; &emsp;theme(<br>	 &emsp; &emsp;panel.grid = element_blank(),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;panel.background = element_blank(),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;panel.border = element_rect(colour = "black", fill=NA, linewidth=1),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.background = element_blank(), <br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.background = element_rect(fill="transparent", colour=NA),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.key = element_rect(fill="transparent", colour=NA),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.title = element_text(size=12, margin = margin(b = 5),hjust=0,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;title = element_text(size = 12, margin = margin(b = 5),hjust=0,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.text.y = element_text(size = 10, margin = margin(r = 5),hjust=1,vjust=0.5, face="bold",colour="black"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.text.x = element_text(size = 10, margin = margin(t = 5),hjust=0.5,vjust=1, face="bold",colour="black"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.title.y = element_text(size = 11, margin = margin(r = 10),angle = 90,hjust=0.5,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.title.x = element_text(size = 11, margin = margin(t = 10),hjust=0.5,vjust=1, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.text=element_text(size=11, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.title=element_blank(), <br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.key.size=unit(2.5,"line"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.margin=unit(c(0.4,0.4,0.4,0.4), "cm")<br> &emsp; &emsp; &emsp; &emsp;)<br>}<br><br><span class="r_comment">##---- Default Three Way Significance Colours &emsp;----####</span><br><br>default_3way_sig_one_colour = "coral3"<br>default_3way_sig_two_colour = "cornflowerblue"<br>default_3way_non_sig_colour = "grey75"<br><br><span class="r_comment">##---- Save Plot Function -----####</span><br><br>save_plot <- function(ggp,plot_height,plot_width,plot_path)<br>{<br> &emsp;png(plot_path, height=plot_height, width=plot_width, pointsize=5)<br> &emsp;print(ggp)<br> &emsp;dev.off()<br><br> &emsp;svg(gsub(".png", ".svg", plot_path), height=plot_height/90, width=plot_width/90)<br> &emsp;print(ggp)<br> &emsp;dev.off()<br><br> &emsp;# clears all devices - as a safety measure<br> &emsp;while (dev.cur()>1) dev.off()<br>}<br><br><span class="r_comment">##---- Volcano Plot Function ----####</span><br><br>make_volcano_plot <- function(de_annotated, sig_up_colour,sig_down_colour, non_sig_colour, dot_size, dot_transparency, sig_up_name, sig_down_name, non_sig_name, x_axis_label, y_axis_label,legend_position, topn, label_size) <br>{<br><br> &emsp;# create a direction column<br> &emsp;de_volcano = de_annotated<br> &emsp;de_volcano$direction = "c"<br><br> &emsp;de_sig_up = subset(de_volcano, log2fold > 0 & sig == "True")<br> &emsp;de_sig_down = subset(de_volcano, log2fold < 0 & sig == "True")<br> &emsp;de_non_sig = subset(de_volcano, sig == "False")<br><br> &emsp;if (nrow(de_sig_up) > 0)<br> &emsp;{<br> &emsp; &emsp;de_sig_up$direction = "a"<br> &emsp;}<br> &emsp;if (nrow(de_sig_down) > 0)<br> &emsp;{<br> &emsp; &emsp;de_sig_down$direction = "b"<br> &emsp;}<br><br> &emsp;de_volcano = rbind(de_non_sig, de_sig_down, de_sig_up)<br><br> &emsp;<br> &emsp;# get genes to label<br> &emsp;if (topn == "ALL_SIG")<br> &emsp;{<br> &emsp; &emsp;de_topn = subset(de_volcano, sig = "True")<br><br> &emsp;}<br> &emsp;else if (topn == 0)<br> &emsp;{<br> &emsp; &emsp;de_topn = de_volcano[0,]<br> &emsp;}<br> &emsp;else<br> &emsp;{<br> &emsp; &emsp;de_topn = de_volcano[order(de_volcano$p),]<br> &emsp; &emsp;de_topn = de_topn[1:topn,]<br> &emsp;}<br> &emsp;<br> &emsp;# get the limits<br> &emsp;limits = c(-max(abs(de_volcano$log2fold)), max(abs(de_volcano$log2fold)))<br><br> &emsp;# make the plot<br> &emsp;ggp = ggplot(data=de_volcano, aes(x=log2fold, y=-log10(p), colour=direction)) +<br> &emsp; &emsp;geom_point(size=dot_size,alpha=dot_transparency) +<br> &emsp; &emsp;geom_label_repel(data=de_topn, aes(label=rownames(de_topn)), size=label_size, force = 1,box.padding = 1, show.legend = FALSE, colour = "black") +<br> &emsp; &emsp;scale_color_manual(breaks=c("a","b","c"), values=c(sig_up_colour ,sig_down_colour, non_sig_colour), labels=c(sig_up_name, sig_down_name, non_sig_name)) +<br> &emsp; &emsp;labs(x=x_axis_label, y=y_axis_label) +<br> &emsp; &emsp;xlim(limits) +<br> &emsp; &emsp;theme_SL2() + <br> &emsp; &emsp;theme(legend.position=legend_position, legend.title = element_blank())<br> &emsp;<br> &emsp;return(ggp)<br>}<br><br><span class="r_comment">##---- Volcano Plot &emsp;----####</span><br><br>plot_height = 500<br>plot_width = 600<br>sig_up_colour = default_3way_sig_one_colour<br>sig_down_colour = default_3way_sig_two_colour<br>non_sig_colour = default_3way_non_sig_colour<br>dot_size = 1.5<br>dot_transparency = 1<br>sig_up_name = "upregulated"<br>sig_down_name = "downregulated"<br>non_sig_name = "non-significant"<br>topn = 10 # option is an integer or "ALL_SIG"<br>label_size = 3.5<br>x_axis_label = expression("log"[2]* " fold change")<br>y_axis_label = expression("-log"[10]* " p-value")<br>legend_position = "bottom"<br><br>ggp = make_volcano_plot(de_annotated, sig_up_colour,sig_down_colour, non_sig_colour, dot_size, dot_transparency, sig_up_name, sig_down_name, non_sig_name, x_axis_label, y_axis_label,legend_position, topn, label_size)<br>save_plot(ggp,plot_height,plot_width,"plots/volcano_plot/volcano_plot_labelled.png")<br><br>topn = 0<br>ggp = make_volcano_plot(de_annotated, sig_up_colour,sig_down_colour, non_sig_colour, dot_size, dot_transparency, sig_up_name, sig_down_name, non_sig_name, x_axis_label, y_axis_label,legend_position, topn, label_size)<br>save_plot(ggp,plot_height,plot_width,"plots/volcano_plot/volcano_plot_unlabelled.png")<br><br></div>
    </div>
</div>

<div id="MA_Plot" class="plot_sections">
    <h2 class="plot_title">MA Plot
</h2>
    <div class="plot_wrapper" style="display: block;">
        <div class="plots"><img style="width:50.0%;" src="plots/MA_plot/MA_plot_labelled.png
"/><img style="width:50.0%;" src="plots/MA_plot/MA_plot_unlabelled.png
"/></div>

        <!-- plot_description is replaced by bin/plot_description/... -->
        <button class="collapsible">plot description</button>
        <div class="plot_description">Here we provide a MA plot. The MA plot shows the relationship between the expression of each gene and its fold change. As lowly expressed genes tend to have fewer reads (observations) there is often typically statistical power to call significance at these genes compared to highly expressed genes. This results in a bias. I.e. we less likely to be able to call a gene significantly differential if it is lowly expressed than highly expressed. The MA plot gives us an indication of this effect. An obvious point on the MA plot where low expression correlates with fewer significant genes, suggests an expression cut-off can be used. The left and right plots are identical, except for clarity the gene names have been removed from the right plot.
</div>

        <!-- plot_legend is replaced by bin/figure_legends/... -->
        <button class="collapsible">figure legend</button>
        <div class="plot_legend"><p>MA plot for the comparison between koc and kosalt. Significantly differential genes (p.adj < 0.05, absolute log2 fold > 1.0) are shown in red and nonâˆ’significant genes in black. A positive fold change indicates higher expression in koc than in kosalt.
</p></div>

        <!-- r_code is replaced by -->
        <button class="collapsible" id="r_code">r code</button>
        <div class="r_code"><span class="r_comment">##---- general libraries ----####</span><br><br>library(ggplot2)<br>library(ggrepel)<br>library(reshape)<br>library(ggridges)<br><br><span class="r_comment">##---- DE sample and group names ----####</span><br><br>samples = c("KOSALT12_S4_001","KOSALT15_S10_001","KOSALT19_S8_001","KOC12_S3_001","KOC15_S9_001","KOC19_S6_001")<br>sample_groups = c("KOSALT","KOC")<br>sample_groups = factor(sample_groups, levels = sample_groups)<br>sample_groupings = c("KOSALT","KOSALT","KOSALT","KOC","KOC","KOC")<br>sample_groupings = factor(sample_groupings, levels = sample_groups)<br>samples_by_sample_group = list(c("KOSALT12_S4_001","KOSALT15_S10_001","KOSALT19_S8_001"),c("KOC12_S3_001","KOC15_S9_001","KOC19_S6_001"))<br>comparisons = c("KOC vs KOSALT")<br><br><span class="r_comment">##---- sets the working directory. If you move the SL2 folder please update this path ----####</span><br><br>setwd("/home/john/Downloads/mahesh_run2/2x75/SL2/all_genes/de_workflows/KOC_vs_KOSALT")<br><br><span class="r_comment">##---- de input files ----####</span><br><br>de_annotated = read.table(file="data/de_annotated.csv", header=TRUE,row.names = 2, sep='\t', quote='',check.names = TRUE)<br>de_annotated_sig = read.table(file="data/de_annotated_significant.csv", header=TRUE,row.names = 2, sep='\t', quote='',check.names = TRUE)<br><br><span class="r_comment">##---- SL2 Theme ----####</span><br><br>theme_SL2 <- function () { <br> &emsp; &emsp;theme_bw() %+replace% <br> &emsp; &emsp; &emsp; &emsp;theme(<br>	 &emsp; &emsp;panel.grid = element_blank(),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;panel.background = element_blank(),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;panel.border = element_rect(colour = "black", fill=NA, linewidth=1),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.background = element_blank(), <br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.background = element_rect(fill="transparent", colour=NA),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.key = element_rect(fill="transparent", colour=NA),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.title = element_text(size=12, margin = margin(b = 5),hjust=0,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;title = element_text(size = 12, margin = margin(b = 5),hjust=0,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.text.y = element_text(size = 10, margin = margin(r = 5),hjust=1,vjust=0.5, face="bold",colour="black"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.text.x = element_text(size = 10, margin = margin(t = 5),hjust=0.5,vjust=1, face="bold",colour="black"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.title.y = element_text(size = 11, margin = margin(r = 10),angle = 90,hjust=0.5,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.title.x = element_text(size = 11, margin = margin(t = 10),hjust=0.5,vjust=1, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.text=element_text(size=11, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.title=element_blank(), <br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.key.size=unit(2.5,"line"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.margin=unit(c(0.4,0.4,0.4,0.4), "cm")<br> &emsp; &emsp; &emsp; &emsp;)<br>}<br><br><span class="r_comment">##---- Default Three Way Significance Colours &emsp;----####</span><br><br>default_3way_sig_one_colour = "coral3"<br>default_3way_sig_two_colour = "cornflowerblue"<br>default_3way_non_sig_colour = "grey75"<br><br><span class="r_comment">##---- Save Plot Function -----####</span><br><br>save_plot <- function(ggp,plot_height,plot_width,plot_path)<br>{<br> &emsp;png(plot_path, height=plot_height, width=plot_width, pointsize=5)<br> &emsp;print(ggp)<br> &emsp;dev.off()<br><br> &emsp;svg(gsub(".png", ".svg", plot_path), height=plot_height/90, width=plot_width/90)<br> &emsp;print(ggp)<br> &emsp;dev.off()<br><br> &emsp;# clears all devices - as a safety measure<br> &emsp;while (dev.cur()>1) dev.off()<br>}<br><br><span class="r_comment">##---- MA Plot Function ----####</span><br><br>make_MA_plot <- function(de_annotated, sig_up_colour,sig_down_colour, non_sig_colour, dot_size, dot_transparency, sig_up_name, sig_down_name, non_sig_name, x_axis_label, y_axis_label,legend_position, topn, label_size)<br>{<br><br> &emsp;# create a direction column<br> &emsp;de_ma = de_annotated<br> &emsp;de_ma$direction = "c"<br><br> &emsp;de_sig_up = subset(de_ma, log2fold > 0 & sig == "True")<br> &emsp;de_sig_down = subset(de_ma, log2fold < 0 & sig == "True")<br> &emsp;de_non_sig = subset(de_ma, sig == "False")<br><br> &emsp;if (nrow(de_sig_up) > 0)<br> &emsp;{<br> &emsp; &emsp;de_sig_up$direction = "a"<br> &emsp;}<br> &emsp;if (nrow(de_sig_down) > 0)<br> &emsp;{<br> &emsp; &emsp;de_sig_down$direction = "b"<br> &emsp;}<br><br> &emsp;de_ma = rbind(de_non_sig, de_sig_down, de_sig_up)<br><br> &emsp;# create row means column<br> &emsp;de_ma$mean = rowMeans(de_ma[,samples])<br><br><br> &emsp;# get genes to label<br> &emsp;if (topn == "ALL_SIG")<br> &emsp;{<br> &emsp; &emsp;de_topn = subset(de_ma, sig = "True")<br> &emsp;}<br> &emsp;else if (topn == 0)<br> &emsp;{<br> &emsp; &emsp;de_topn = de_ma[0,]<br> &emsp;}<br> &emsp;else<br> &emsp;{<br> &emsp; &emsp;de_topn = de_ma[order(de_ma$p),]<br> &emsp; &emsp;de_topn = de_topn[1:topn,]<br> &emsp;}<br><br> &emsp;# get the limits<br> &emsp;limits = c(-max(abs(de_ma$log2fold)), max(abs(de_ma$log2fold)))<br><br> &emsp;# make the plot<br> &emsp;ggp = ggplot(data=de_ma, aes(x=log10(mean+0.01), y=log2fold, colour=direction)) +<br> &emsp; &emsp;geom_point(size=dot_size,alpha=dot_transparency) +<br> &emsp; &emsp;geom_label_repel(data=de_topn, aes(label=rownames(de_topn)), size=label_size, force = 1,box.padding = 1, show.legend = FALSE, colour = "black") +<br> &emsp; &emsp;scale_color_manual(breaks=c("a","b","c"), values=c(sig_up_colour ,sig_down_colour, non_sig_colour), labels=c(sig_up_name, sig_down_name, non_sig_name)) +<br> &emsp; &emsp;labs(x=x_axis_label, y=y_axis_label) +<br> &emsp; &emsp;ylim(limits) +<br> &emsp; &emsp;theme_SL2() +<br> &emsp; &emsp;theme(legend.position=legend_position, legend.title = element_blank())<br><br> &emsp;return(ggp)<br>}<br><br><span class="r_comment">##---- MA Plot ----####</span><br><br>plot_height = 500<br>plot_width = 600<br>sig_up_colour = default_3way_sig_one_colour<br>sig_down_colour = default_3way_sig_two_colour<br>non_sig_colour = default_3way_non_sig_colour<br>dot_size = 1.5<br>dot_transparency = 1<br>sig_up_name = "upregulated"<br>sig_down_name = "downregulated"<br>non_sig_name = "non-significant"<br>topn = 10 # option is an integer or "ALL_SIG"<br>label_size = 3.5<br>x_axis_label = expression("mean expression (log"[10]* ")")<br>y_axis_label = expression("log"[2]* " fold change")<br>legend_position = "bottom"<br><br>ggp = make_MA_plot(de_annotated, sig_up_colour,sig_down_colour, non_sig_colour, dot_size, dot_transparency, sig_up_name, sig_down_name, non_sig_name, x_axis_label, y_axis_label,legend_position, topn, label_size)<br>save_plot(ggp,plot_height,plot_width,"plots/MA_plot/MA_plot_labelled.png")<br><br>topn = 0<br>ggp = make_MA_plot(de_annotated, sig_up_colour,sig_down_colour, non_sig_colour, dot_size, dot_transparency, sig_up_name, sig_down_name, non_sig_name, x_axis_label, y_axis_label,legend_position, topn, label_size)<br>save_plot(ggp,plot_height,plot_width,"plots/MA_plot/MA_plot_unlabelled.png")<br><br></div>
    </div>
</div>

<div id="Number_of_Significant_Genes" class="plot_sections">
    <h2 class="plot_title">Number of Significant Genes
</h2>
    <div class="plot_wrapper" style="display: block;">
        <div class="plots"><img style="width:33.33333333333333%;" src="plots/number_of_significant_genes/number_of_significant_genes_lollipop.png
"/></div>

        <!-- plot_description is replaced by bin/plot_description/... -->
        <button class="collapsible">plot description</button>
        <div class="plot_description">Here we provide a barchart of the number of significantly up and downregulated genes. The number and direction of the differences between the two sample groups can give insight into the strength and directionality of the difference. If there are many thousands of significantly different genes it can often be useful to increase the stringency of significance, in order to highlight the most important genes.
</div>

        <!-- plot_legend is replaced by bin/figure_legends/... -->
        <button class="collapsible">figure legend</button>
        <div class="plot_legend"><p>Bar chart showing the number of significantly differentially expressed genes (p.adj < 0.05, absolute log2 fold > 1.0), between koc and kosalt. Upregulated genes are higher in koc than in kosalt.
</p></div>

        <!-- r_code is replaced by -->
        <button class="collapsible" id="r_code">r code</button>
        <div class="r_code"><span class="r_comment">##---- general libraries ----####</span><br><br>library(ggplot2)<br>library(ggrepel)<br>library(reshape)<br>library(ggridges)<br><br><span class="r_comment">##---- DE sample and group names ----####</span><br><br>samples = c("KOSALT12_S4_001","KOSALT15_S10_001","KOSALT19_S8_001","KOC12_S3_001","KOC15_S9_001","KOC19_S6_001")<br>sample_groups = c("KOSALT","KOC")<br>sample_groups = factor(sample_groups, levels = sample_groups)<br>sample_groupings = c("KOSALT","KOSALT","KOSALT","KOC","KOC","KOC")<br>sample_groupings = factor(sample_groupings, levels = sample_groups)<br>samples_by_sample_group = list(c("KOSALT12_S4_001","KOSALT15_S10_001","KOSALT19_S8_001"),c("KOC12_S3_001","KOC15_S9_001","KOC19_S6_001"))<br>comparisons = c("KOC vs KOSALT")<br><br><span class="r_comment">##---- sets the working directory. If you move the SL2 folder please update this path ----####</span><br><br>setwd("/home/john/Downloads/mahesh_run2/2x75/SL2/all_genes/de_workflows/KOC_vs_KOSALT")<br><br><span class="r_comment">##---- de input files ----####</span><br><br>de_annotated = read.table(file="data/de_annotated.csv", header=TRUE,row.names = 2, sep='\t', quote='',check.names = TRUE)<br>de_annotated_sig = read.table(file="data/de_annotated_significant.csv", header=TRUE,row.names = 2, sep='\t', quote='',check.names = TRUE)<br><br><span class="r_comment">##---- SL2 Theme ----####</span><br><br>theme_SL2 <- function () { <br> &emsp; &emsp;theme_bw() %+replace% <br> &emsp; &emsp; &emsp; &emsp;theme(<br>	 &emsp; &emsp;panel.grid = element_blank(),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;panel.background = element_blank(),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;panel.border = element_rect(colour = "black", fill=NA, linewidth=1),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.background = element_blank(), <br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.background = element_rect(fill="transparent", colour=NA),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.key = element_rect(fill="transparent", colour=NA),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.title = element_text(size=12, margin = margin(b = 5),hjust=0,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;title = element_text(size = 12, margin = margin(b = 5),hjust=0,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.text.y = element_text(size = 10, margin = margin(r = 5),hjust=1,vjust=0.5, face="bold",colour="black"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.text.x = element_text(size = 10, margin = margin(t = 5),hjust=0.5,vjust=1, face="bold",colour="black"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.title.y = element_text(size = 11, margin = margin(r = 10),angle = 90,hjust=0.5,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.title.x = element_text(size = 11, margin = margin(t = 10),hjust=0.5,vjust=1, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.text=element_text(size=11, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.title=element_blank(), <br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.key.size=unit(2.5,"line"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.margin=unit(c(0.4,0.4,0.4,0.4), "cm")<br> &emsp; &emsp; &emsp; &emsp;)<br>}<br><br><span class="r_comment">##---- Default Three Way Significance Colours &emsp;----####</span><br><br>default_3way_sig_one_colour = "coral3"<br>default_3way_sig_two_colour = "cornflowerblue"<br>default_3way_non_sig_colour = "grey75"<br><br><span class="r_comment">##---- Default Comaparison Labels &emsp;----####</span><br><br>default_comparison_labels = c("KOC vs KOSALT") # note: changing this won't change the order the comparisons appear on the plots. Merely what they are labelled as.<br><br><span class="r_comment">##---- Save Plot Function -----####</span><br><br>save_plot <- function(ggp,plot_height,plot_width,plot_path)<br>{<br> &emsp;png(plot_path, height=plot_height, width=plot_width, pointsize=5)<br> &emsp;print(ggp)<br> &emsp;dev.off()<br><br> &emsp;svg(gsub(".png", ".svg", plot_path), height=plot_height/90, width=plot_width/90)<br> &emsp;print(ggp)<br> &emsp;dev.off()<br><br> &emsp;# clears all devices - as a safety measure<br> &emsp;while (dev.cur()>1) dev.off()<br>}<br><br><span class="r_comment">##---- Number of Significant Genes Lollipop Plot Function ----####</span><br><br>make_number_of_significant_genes_lollipop <- function(x_axis_label, y_axis_label, comparison_labels, direction_labels, up_lollipop_colour, down_lollipop_colour, stick_colour, stick_size, lollipop_size, legend_position, data_label_size)<br>{<br> &emsp;# Parse the plot data<br> &emsp;upregulated = nrow(subset(de_annotated_sig,log2fold > 0))<br> &emsp;downregulated = -nrow(subset(de_annotated_sig,log2fold < 0))<br> &emsp;table_data = data.frame(comparison = factor(comparisons, levels=comparisons), up = upregulated, down = downregulated)<br><br> &emsp;table_data = melt(table_data, id.vars = "comparison")<br> &emsp;table_data_up = subset(table_data, variable == "up")<br> &emsp;table_data_down = subset(table_data, variable == "down")<br><br> &emsp;# limits<br> &emsp;max = max(abs(table_data$value)) *1.5<br> &emsp;limits = c(-max,max)<br><br> &emsp;# make the plot<br> &emsp;ggp = ggplot(table_data, aes(x=value, y=comparison, colour = variable)) +<br> &emsp; &emsp;geom_segment(aes(xend = 0, yend = comparison), size = stick_size, show.legend = FALSE, colour=stick_colour) +<br> &emsp; &emsp;geom_point(size = lollipop_size) +<br> &emsp; &emsp;geom_label(data=table_data_up, aes(label=value),label.padding = unit(0.75, "lines"), show.legend = FALSE, size=data_label_size, nudge_x = max*0.2, vjust = 0.5, colour = "black") +<br> &emsp; &emsp;geom_label(data=table_data_down, aes(label=value),label.padding = unit(0.75, "lines"), show.legend = FALSE, size=data_label_size, nudge_x = -max*0.2, vjust = 0.5, colour = "black") +<br> &emsp; &emsp;geom_vline(xintercept = 0, colour = "black", size = 1) +<br> &emsp; &emsp;scale_color_manual(values=c(up_lollipop_colour,down_lollipop_colour),labels=direction_labels) +<br> &emsp; &emsp;xlim(limits) +<br> &emsp; &emsp;labs(y=y_axis_label, x=x_axis_label) +<br> &emsp; &emsp;theme_SL2() +<br> &emsp; &emsp;theme(legend.position=legend_position, legend.spacing.x = unit(0.25, 'cm'))<br><br> &emsp;return(ggp)<br>}<br><br><span class="r_comment">##---- Number of Significant Genes Lollipop Plot ----####</span><br><br>plot_height = 150<br>plot_width = 500<br>x_axis_label = "number of significant genes (pos = up, neg = down)"<br>y_axis_label = ""<br>comparison_labels = default_comparison_labels # note: changing this won't change the order the comparisons appear in the x axis. Merely what they are labelled as.<br>direction_labels = c("up","down") # note: changing this won't change the order the bars appear in the x axis. Merely what they are labelled as.<br>up_lollipop_colour = default_3way_sig_one_colour<br>down_lollipop_colour = default_3way_sig_two_colour<br>stick_colour = default_3way_non_sig_colour<br>stick_size = 1<br>lollipop_size = 5<br>legend_position = "none"<br>data_label_size = 5<br><br>ggp = make_number_of_significant_genes_lollipop(x_axis_label, y_axis_label, comparison_labels, direction_labels, up_lollipop_colour, down_lollipop_colour, stick_colour, stick_size, lollipop_size, legend_position, data_label_size)<br>save_plot(ggp,plot_height,plot_width,"plots/number_of_significant_genes/number_of_significant_genes_lollipop.png")<br><br></div>
    </div>
</div>

<div id="Significant_Genes_Heatmap" class="plot_sections">
    <h2 class="plot_title">Significant Genes Heatmap
</h2>
    <div class="plot_wrapper" style="display: block;">
        <div class="plots"><img style="width:33.33333333333333%;" src="plots/significant_genes_heatmap/significant_genes_heatmap.png
"/><img style="width:33.33333333333333%;" src="plots/significant_genes_heatmap/significant_genes_heatmap_column_clustered.png
"/></div>

        <!-- plot_description is replaced by bin/plot_description/... -->
        <button class="collapsible">plot description</button>
        <div class="plot_description">It is usually useful to inspect the behaviour of each sample and sample group at each of the differentially expressed genes, using a heatmap. This allows visual validation of the consistency of the differentially expressed genes within and between replicates. In addition the genes have been clustered. In simple terms, grouped by the similarity of their expression profiles. This often highlights any subâˆ’patterns within the data. Finally, in order to focus on the pattern of expression, rather than which gene is generically more highly expressed than any other gene, the data has been row standardised with a zâˆ’score.
</div>

        <!-- plot_legend is replaced by bin/figure_legends/... -->
        <button class="collapsible">figure legend</button>
        <div class="plot_legend"><p>Hierarchically clustered heatmap of the significantly differentially expressed genes (p.adj < 0.05, absolute log2 fold > 1.0) between koc and kosalt, for all sample in the sample groups: kosalt and koc. Samples are on the x axis and genes on the y axis. Colour intensity represents expression level, with blue representing low expression, and red representing high expression. Expression levels have been row scaled into zâˆ’scores. The y-axis (both plots) and x-axis (right plot) have been hierarchically clustered using, Spearman distances, with UPMGA agglomeration and mean reordering.
</p></div>

        <!-- r_code is replaced by -->
        <button class="collapsible" id="r_code">r code</button>
        <div class="r_code"><span class="r_comment">##---- general libraries ----####</span><br><br>library(ggplot2)<br>library(ggrepel)<br>library(reshape)<br>library(ggridges)<br><br><span class="r_comment">##---- heatmap libraries ----####</span><br><br>library(amap)<br><br><span class="r_comment">##---- table libraries ----####</span><br><br>library(grid)<br>library(gridExtra)<br>library(gtable)<br>library(ggplotify)<br><br><span class="r_comment">##---- DE sample and group names ----####</span><br><br>samples = c("KOSALT12_S4_001","KOSALT15_S10_001","KOSALT19_S8_001","KOC12_S3_001","KOC15_S9_001","KOC19_S6_001")<br>sample_groups = c("KOSALT","KOC")<br>sample_groups = factor(sample_groups, levels = sample_groups)<br>sample_groupings = c("KOSALT","KOSALT","KOSALT","KOC","KOC","KOC")<br>sample_groupings = factor(sample_groupings, levels = sample_groups)<br>samples_by_sample_group = list(c("KOSALT12_S4_001","KOSALT15_S10_001","KOSALT19_S8_001"),c("KOC12_S3_001","KOC15_S9_001","KOC19_S6_001"))<br>comparisons = c("KOC vs KOSALT")<br><br><span class="r_comment">##---- sets the working directory. If you move the SL2 folder please update this path ----####</span><br><br>setwd("/home/john/Downloads/mahesh_run2/2x75/SL2/all_genes/de_workflows/KOC_vs_KOSALT")<br><br><span class="r_comment">##---- de input files ----####</span><br><br>de_annotated = read.table(file="data/de_annotated.csv", header=TRUE,row.names = 2, sep='\t', quote='',check.names = TRUE)<br>de_annotated_sig = read.table(file="data/de_annotated_significant.csv", header=TRUE,row.names = 2, sep='\t', quote='',check.names = TRUE)<br><br><span class="r_comment">##---- de parse ne matrix ----####</span><br><br>ne_matrix = de_annotated[,samples]<br>ne_matrix_sig = de_annotated_sig[,samples]<br><br><span class="r_comment">##---- transpose and scale significant genes ne matrix ----####</span><br><br>ne_matrix_sig_scaled = data.frame(t(scale(t(ne_matrix_sig))))<br>rownames(ne_matrix_sig_scaled) = rownames(ne_matrix_sig)<br>ne_matrix_sig_scaled[do.call(cbind, lapply(ne_matrix_sig_scaled, is.nan))] <- 0<br>ne_matrix_sig_scaled = ne_matrix_sig_scaled[is.finite(rowSums(ne_matrix_sig_scaled)), ]<br><br><span class="r_comment">##---- SL2 Theme ----####</span><br><br>theme_SL2 <- function () { <br> &emsp; &emsp;theme_bw() %+replace% <br> &emsp; &emsp; &emsp; &emsp;theme(<br>	 &emsp; &emsp;panel.grid = element_blank(),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;panel.background = element_blank(),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;panel.border = element_rect(colour = "black", fill=NA, linewidth=1),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.background = element_blank(), <br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.background = element_rect(fill="transparent", colour=NA),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.key = element_rect(fill="transparent", colour=NA),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.title = element_text(size=12, margin = margin(b = 5),hjust=0,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;title = element_text(size = 12, margin = margin(b = 5),hjust=0,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.text.y = element_text(size = 10, margin = margin(r = 5),hjust=1,vjust=0.5, face="bold",colour="black"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.text.x = element_text(size = 10, margin = margin(t = 5),hjust=0.5,vjust=1, face="bold",colour="black"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.title.y = element_text(size = 11, margin = margin(r = 10),angle = 90,hjust=0.5,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.title.x = element_text(size = 11, margin = margin(t = 10),hjust=0.5,vjust=1, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.text=element_text(size=11, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.title=element_blank(), <br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.key.size=unit(2.5,"line"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.margin=unit(c(0.4,0.4,0.4,0.4), "cm")<br> &emsp; &emsp; &emsp; &emsp;)<br>}<br><br><span class="r_comment">##---- Default Three Tone Heatmap Colours &emsp;----####</span><br><br>default_three_tone_heatmap_colours = c("purple","black","yellow")<br><br><span class="r_comment">##---- Default Sample Labels &emsp;----####</span><br><br>default_sample_labels = gsub("_"," ",c("KOSALT12_S4_001","KOSALT15_S10_001","KOSALT19_S8_001","KOC12_S3_001","KOC15_S9_001","KOC19_S6_001")) # note: changing this won't change the order the samples appear on the plots. Merely what they are labelled as.<br><br><span class="r_comment">##----- Default GGplot Colours Function -----####</span><br><br>gg_color_hue <- function(n)<br>{<br> &emsp;# default colour blind palettes<br> &emsp;cblind_palette = c("#88CCEE", "#CC6677", "#DDCC77", "#117733", "#332288", "#AA4499", "#44AA99", "#999933", "#882255", "#661100", "#6699CC", "#888888")<br><br> &emsp;# append ggplot colours if groups goes above colour blind palette length<br> &emsp;if (n <= length(cblind_palette))<br> &emsp;{<br> &emsp; &emsp;return(cblind_palette[1:n])<br> &emsp;}<br> &emsp;else<br> &emsp;{<br> &emsp; &emsp;nn = n - length(cblind_palette)<br> &emsp; &emsp;hues = seq(15, 375, length = nn + 1)<br> &emsp; &emsp;return(append(cblind_palette,hcl(h = hues, l = 65, c = 100)[1:nn]))<br> &emsp;}<br>}<br><br><span class="r_comment">##---- Default Sample and Sample Group Colours &emsp;----####</span><br><br>number_of_sample_groups = length(sample_groups)<br>default_sample_group_colours = gg_color_hue(number_of_sample_groups)<br>default_samples_colours = c(default_sample_group_colours[1],default_sample_group_colours[1],default_sample_group_colours[1],default_sample_group_colours[2],default_sample_group_colours[2],default_sample_group_colours[2])<br><br><span class="r_comment">##---- Save Plot Function -----####</span><br><br>save_plot <- function(ggp,plot_height,plot_width,plot_path)<br>{<br> &emsp;png(plot_path, height=plot_height, width=plot_width, pointsize=5)<br> &emsp;print(ggp)<br> &emsp;dev.off()<br><br> &emsp;svg(gsub(".png", ".svg", plot_path), height=plot_height/90, width=plot_width/90)<br> &emsp;print(ggp)<br> &emsp;dev.off()<br><br> &emsp;# clears all devices - as a safety measure<br> &emsp;while (dev.cur()>1) dev.off()<br>}<br><br><span class="r_comment">##----- Clustering Method -----####</span><br><br>cluster_matrix <- function(mx,cluster_x,cluster_y,distance_method,clustering_method,reorder_function)<br>{<br> &emsp;x.cluster = hclust(Dist(t(mx), method=distance_method), method=clustering_method)<br> &emsp;y.cluster = hclust(Dist(mx, method=distance_method), method=clustering_method)<br> &emsp;<br> &emsp;x.dd = as.dendrogram(x.cluster)<br> &emsp;y.dd = as.dendrogram(y.cluster)<br> &emsp;x.dd.reorder = reorder(x.dd,0,FUN=reorder_function)<br> &emsp;y.dd.reorder = reorder(y.dd,0,FUN=reorder_function)<br> &emsp;x.order = order.dendrogram(x.dd.reorder)<br> &emsp;y.order = order.dendrogram(y.dd.reorder)<br><br> &emsp;if(cluster_x == TRUE && cluster_y == TRUE) <br> &emsp;{<br> &emsp; &emsp; &emsp;mx_clustered = mx[y.order,x.order]<br> &emsp;}<br> &emsp;<br> &emsp;if(cluster_x == TRUE && cluster_y == FALSE)<br> &emsp;{<br> &emsp; &emsp; &emsp;mx_clustered = mx[,x.order]<br> &emsp;}<br> &emsp;<br> &emsp;if(cluster_x == FALSE && cluster_y == TRUE)<br> &emsp;{<br> &emsp; &emsp; &emsp;mx_clustered = mx[y.order,]<br> &emsp;}<br> &emsp;if(cluster_x == FALSE && cluster_y == FALSE) <br> &emsp;{<br> &emsp; &emsp; &emsp;mx_clustered = mx<br> &emsp;}<br> &emsp;return(mx_clustered)<br>}<br><br><span class="r_comment">##---- Significant Genes Heatmap Function &emsp;----####</span><br><br>make_significant_genes_heatmap <- function(matrix_scaled,colours,sample_names,cluster_x,cluster_y,distance_method,clustering_method,reorder_function, sample_groupings, default_samples_colours)<br>{<br><br> &emsp;# tests for enough genes to make a heatmap<br> &emsp;if(nrow(matrix_scaled) >= 2) <br> &emsp;{<br><br> &emsp; &emsp;# parse the table<br> &emsp; &emsp;matrix_scaled_renamed = as.matrix(matrix_scaled)<br> &emsp; &emsp;colnames(matrix_scaled_renamed) = sample_names<br><br> &emsp; &emsp;# cluster the table<br> &emsp; &emsp;matrix_scaled_renamed_clustered = cluster_matrix(matrix_scaled_renamed,cluster_x,cluster_y,distance_method,clustering_method,reorder_function)<br> &emsp; &emsp;matrix_scaled_renamed_clustered_melted = melt(matrix_scaled_renamed_clustered)<br> &emsp; &emsp;matrix_scaled_renamed_clustered_melted$X1 = factor(matrix_scaled_renamed_clustered_melted$X1, levels=row.names(matrix_scaled_renamed_clustered))<br> &emsp; &emsp;matrix_scaled_renamed_clustered_melted$X2 = factor(matrix_scaled_renamed_clustered_melted$X2, levels=colnames(matrix_scaled_renamed_clustered))<br><br> &emsp; &emsp;# colours<br> &emsp; &emsp;hm.palette = colorRampPalette(colours)<br><br> &emsp; &emsp;# make the heatmap<br> &emsp; &emsp;ggp.hm = ggplot(matrix_scaled_renamed_clustered_melted, aes(x = X2, y = X1, fill = value)) +<br> &emsp; &emsp; &emsp;geom_tile() +<br> &emsp; &emsp; &emsp;scale_fill_gradientn(colours = hm.palette(100)) +<br> &emsp; &emsp; &emsp;ylab('') +<br> &emsp; &emsp; &emsp;xlab('') +<br> &emsp; &emsp; &emsp;theme_SL2() +<br> &emsp; &emsp; &emsp;theme(legend.position="right", legend.title = element_blank(), legend.spacing.x = unit(0.25, 'cm'),axis.text.x = element_text(angle = 90, hjust = 1), axis.ticks=element_blank()) +<br> &emsp; &emsp; &emsp;scale_x_discrete(expand=c(0,0)) +<br> &emsp; &emsp; &emsp;scale_y_discrete(expand=c(0,0))<br><br><br> &emsp; &emsp;## create the rug data<br> &emsp; &emsp;rug_data = data.frame(default_samples_colours)<br> &emsp; &emsp;names(rug_data) = "colours"<br> &emsp; &emsp;row.names(rug_data) = samples<br> &emsp; &emsp;rug_data$Y = "1"<br> &emsp; &emsp;rug_data = rug_data[ gsub(" ","_",colnames(matrix_scaled_renamed_clustered)),]<br> &emsp; &emsp;rug_data$X = row.names(rug_data)<br> &emsp; &emsp;rug_data$X = factor(row.names(rug_data), row.names(rug_data))<br><br><br> &emsp; &emsp;# create rug<br> &emsp; &emsp;ggp.rug = ggplot(rug_data, aes(x = X, y = Y)) +<br> &emsp; &emsp; &emsp;geom_tile(fill = rug_data$colours) +<br> &emsp; &emsp; &emsp;theme_void() +<br> &emsp; &emsp; &emsp;theme(legend.position = "none") +<br> &emsp; &emsp; &emsp;ylab('') +<br> &emsp; &emsp; &emsp;xlab('') +<br> &emsp; &emsp; &emsp;scale_x_discrete(expand=c(0,0)) +<br> &emsp; &emsp; &emsp;scale_y_discrete(expand=c(0,0))<br><br> &emsp; &emsp;# sets the widths<br> &emsp; &emsp;max_widths = unit.pmax(ggplotGrob(ggp.hm)$widths)<br> &emsp; &emsp;ggp.rug = ggplotGrob(ggp.rug)<br> &emsp; &emsp;ggp.rug$widths = max_widths<br><br> &emsp; &emsp;# combine<br> &emsp; &emsp;ggp = as.ggplot(grid.arrange(ggp.rug, ggp.hm, &emsp;ncol = 1, heights = c(1, 15)))<br><br> &emsp; &emsp;# clears all devices - as a safety measure<br> &emsp; &emsp;pdf(NULL)<br> &emsp; &emsp;while (dev.cur()>1) dev.off()<br><br> &emsp; &emsp;return(ggp)<br> &emsp;}<br> &emsp;else<br> &emsp;{<br> &emsp; &emsp;return(ggplot(data.frame()) + theme_SL2() + geom_blank() + ggtitle("There were too few significant genes to plot this."))<br> &emsp;}<br>}<br><br><span class="r_comment">##---- Significant Genes Heatmap &emsp;----####</span><br><br>plot_height = 750<br>plot_width = 500<br>colours = default_three_tone_heatmap_colours<br>sample_names = default_sample_labels<br>distance_method = "spearman"<br>clustering_method = "average"<br>reorder_function = "average"<br><br>cluster_x = FALSE<br>cluster_y = TRUE<br><br>ggp = make_significant_genes_heatmap(ne_matrix_sig_scaled,colours,sample_names,cluster_x,cluster_y,distance_method,clustering_method,reorder_function, sample_groupings, default_samples_colours)<br>save_plot(ggp,plot_height,plot_width,"plots/significant_genes_heatmap/significant_genes_heatmap.png")<br><br>cluster_x = TRUE<br>cluster_y = TRUE<br><br>ggp = make_significant_genes_heatmap(ne_matrix_sig_scaled,colours,sample_names,cluster_x,cluster_y,distance_method,clustering_method,reorder_function, sample_groupings, default_samples_colours)<br>save_plot(ggp,plot_height,plot_width,"plots/significant_genes_heatmap/significant_genes_heatmap_column_clustered.png")<br><br></div>
    </div>
</div>

<div id="Most_Differential_Genes_Tables" class="plot_sections">
    <h2 class="plot_title">Most Differential Genes Tables
</h2>
    <div class="plot_wrapper" style="display: block;">
        <div class="plots"><h4>Upregulated By P-value
</h4><img style="width:33.33333333333333%;" src="plots/most_differential_genes/most_upregulated_by_p_value/most_upregulated_genes_table.png
"/><h4>Upregulated By Log2 Fold
</h4><img style="width:33.33333333333333%;" src="plots/most_differential_genes/most_upregulated_by_log2fold/most_upregulated_genes_table.png
"/><h4>Downregulated By P-value
</h4><img style="width:33.33333333333333%;" src="plots/most_differential_genes/most_downregulated_by_p_value/most_downregulated_genes_table.png
"/><h4>Downregulated By Log2 Fold
</h4><img style="width:33.33333333333333%;" src="plots/most_differential_genes/most_downregulated_by_log2fold/most_downregulated_genes_table.png
"/></div>

        <!-- plot_description is replaced by bin/plot_description/... -->
        <button class="collapsible">plot description</button>
        <div class="plot_description">Here we provide a list of the ten most highly upregulated and most highly downregulated genes by p-value and log2 fold change. This can be useful in some cases for identifying candidate genes. As a rule of thumb, more highly or downregulated upregulated genes tend to be more reliable, and less peripheral to the biological difference between two sample groups. Thus the ten most highly upregulated or downregulated genes can also give a useful snapshot as to the types of genes that differ between the two sample groups.
</div>

        <!-- plot_legend is replaced by bin/figure_legends/... -->
        <button class="collapsible">figure legend</button>
        <div class="plot_legend"><p>Table showing the 10 most upregulated and downregulated genes by adjusted p-value and log2fold change. Upregulated genes are higher and downregulated genes lower in koc than in kosalt.
</p></div>

        <!-- r_code is replaced by -->
        <button class="collapsible" id="r_code">r code</button>
        <div class="r_code"><span class="r_comment">##---- general libraries ----####</span><br><br>library(ggplot2)<br>library(ggrepel)<br>library(reshape)<br>library(ggridges)<br><br><span class="r_comment">##---- table libraries ----####</span><br><br>library(grid)<br>library(gridExtra)<br>library(gtable)<br>library(ggplotify)<br><br><span class="r_comment">##---- DE sample and group names ----####</span><br><br>samples = c("KOSALT12_S4_001","KOSALT15_S10_001","KOSALT19_S8_001","KOC12_S3_001","KOC15_S9_001","KOC19_S6_001")<br>sample_groups = c("KOSALT","KOC")<br>sample_groups = factor(sample_groups, levels = sample_groups)<br>sample_groupings = c("KOSALT","KOSALT","KOSALT","KOC","KOC","KOC")<br>sample_groupings = factor(sample_groupings, levels = sample_groups)<br>samples_by_sample_group = list(c("KOSALT12_S4_001","KOSALT15_S10_001","KOSALT19_S8_001"),c("KOC12_S3_001","KOC15_S9_001","KOC19_S6_001"))<br>comparisons = c("KOC vs KOSALT")<br><br><span class="r_comment">##---- sets the working directory. If you move the SL2 folder please update this path ----####</span><br><br>setwd("/home/john/Downloads/mahesh_run2/2x75/SL2/all_genes/de_workflows/KOC_vs_KOSALT")<br><br><span class="r_comment">##---- de input files ----####</span><br><br>de_annotated = read.table(file="data/de_annotated.csv", header=TRUE,row.names = 2, sep='\t', quote='',check.names = TRUE)<br>de_annotated_sig = read.table(file="data/de_annotated_significant.csv", header=TRUE,row.names = 2, sep='\t', quote='',check.names = TRUE)<br><br><span class="r_comment">##---- de parse ne matrix ----####</span><br><br>ne_matrix = de_annotated[,samples]<br>ne_matrix_sig = de_annotated_sig[,samples]<br><br><span class="r_comment">##---- transpose and scale ne matrix ----####</span><br><br>ne_matrix_transposed = data.frame(t(ne_matrix))<br>colnames(ne_matrix_transposed) = rownames(ne_matrix)<br><br>ne_matrix_scaled = data.frame(t(scale(t(ne_matrix))))<br>rownames(ne_matrix_scaled) = rownames(ne_matrix)<br>ne_matrix_scaled[do.call(cbind, lapply(ne_matrix_scaled, is.nan))] <- 0<br>ne_matrix_scaled = ne_matrix_scaled[is.finite(rowSums(ne_matrix_scaled)), ]<br><br>ne_matrix_scaled_transposed = data.frame(t(ne_matrix_scaled))<br>colnames(ne_matrix_scaled_transposed) = rownames(ne_matrix_scaled)<br><br><span class="r_comment">##---- SL2 Theme ----####</span><br><br>theme_SL2 <- function () { <br> &emsp; &emsp;theme_bw() %+replace% <br> &emsp; &emsp; &emsp; &emsp;theme(<br>	 &emsp; &emsp;panel.grid = element_blank(),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;panel.background = element_blank(),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;panel.border = element_rect(colour = "black", fill=NA, linewidth=1),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.background = element_blank(), <br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.background = element_rect(fill="transparent", colour=NA),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.key = element_rect(fill="transparent", colour=NA),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.title = element_text(size=12, margin = margin(b = 5),hjust=0,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;title = element_text(size = 12, margin = margin(b = 5),hjust=0,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.text.y = element_text(size = 10, margin = margin(r = 5),hjust=1,vjust=0.5, face="bold",colour="black"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.text.x = element_text(size = 10, margin = margin(t = 5),hjust=0.5,vjust=1, face="bold",colour="black"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.title.y = element_text(size = 11, margin = margin(r = 10),angle = 90,hjust=0.5,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.title.x = element_text(size = 11, margin = margin(t = 10),hjust=0.5,vjust=1, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.text=element_text(size=11, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.title=element_blank(), <br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.key.size=unit(2.5,"line"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.margin=unit(c(0.4,0.4,0.4,0.4), "cm")<br> &emsp; &emsp; &emsp; &emsp;)<br>}<br><br><span class="r_comment">##---- Save Table Function -----####</span><br><br>save_table <- function(gt,plot_height,plot_width,plot_path)<br>{<br> &emsp;png(plot_path, height=plot_height, width=plot_width, pointsize=5)<br> &emsp;grid.arrange(gt)<br> &emsp;dev.off()<br><br> &emsp;while (dev.cur()>1) dev.off()<br>}<br><br><span class="r_comment">##---- Top 10 Differential Genes Function ----####</span><br><br>get_top_10_differential_genes <- function(direction,type) {<br><br> &emsp;if(type=="log2fold") <br> &emsp;{<br> &emsp; &emsp;top_10_genes = row.names(head(de_annotated[order(de_annotated$log2fold,decreasing=direction),],10))<br> &emsp;}<br> &emsp;if(type=="p")<br> &emsp;{<br> &emsp; &emsp;if (direction == TRUE) {de_annotated_direction = subset(de_annotated,log2fold > 0)}<br> &emsp; &emsp;else {de_annotated_direction = subset(de_annotated,log2fold < 0)}<br> &emsp; &emsp;top_10_genes = row.names(head(de_annotated_direction[order(de_annotated_direction$p,decreasing=FALSE),],10))<br> &emsp;}<br> &emsp;return(top_10_genes)<br>}<br><br><span class="r_comment">##---- Most Differential Genes Table Function ----####</span><br><br>make_most_differential_genes_table <- function(top_10_genes,header_size,text_size,border_thickness)<br>{<br> &emsp;de_annotated_top_10 = de_annotated[top_10_genes,]<br> &emsp;de_annotated_top_10$Gene = row.names(de_annotated_top_10)<br> &emsp;de_annotated_top_10 = de_annotated_top_10[,c("Gene","log2fold","p","p.adj")]<br><br> &emsp;table_theme <- gridExtra::ttheme_minimal(core = list(fg_params=list(cex = text_size)),colhead = list(fg_params=list(cex = header_size)))<br><br> &emsp;gt = tableGrob(de_annotated_top_10, theme=table_theme, rows=NULL)<br> &emsp;gt = gtable_add_grob(gt,grobs = rectGrob(gp = gpar(fill = NA, lwd = border_thickness)),t = 2, b = nrow(gt), l = 1, r = ncol(gt))<br> &emsp;gt = gtable_add_grob(gt,grobs = rectGrob(gp = gpar(fill = NA, lwd = border_thickness)),t = 1, l = 1, r = ncol(gt))<br> &emsp;gt = gtable_add_grob(gt,grobs = rectGrob(gp = gpar(fill = NA, lwd = border_thickness)),t = 1, b = nrow(gt), l = 1, r = 1)<br> &emsp;<br> &emsp;return(gt)<br>}<br><br><span class="r_comment">##---- Most Differential Genes (Tables) ----####</span><br><br>plot_height = 300<br>plot_width = 400<br>header_size = 1.25<br>text_size = 1.25<br>border_thickness = 2<br><br>top_10_genes = get_top_10_differential_genes(TRUE,"p") # upregulated, by p<br>gt = make_most_differential_genes_table(top_10_genes,header_size,text_size,border_thickness)<br>save_table(gt,plot_height,plot_width,"plots/most_differential_genes/most_upregulated_by_p_value/most_upregulated_genes_table.png")<br><br>top_10_genes = get_top_10_differential_genes(FALSE,"p") # downregulated, by p<br>gt = make_most_differential_genes_table(top_10_genes,header_size,text_size,border_thickness)<br>save_table(gt,plot_height,plot_width,"plots/most_differential_genes/most_downregulated_by_p_value/most_downregulated_genes_table.png")<br><br>top_10_genes = get_top_10_differential_genes(TRUE,"log2fold") # upregulated, by log2fold<br>gt = make_most_differential_genes_table(top_10_genes,header_size,text_size,border_thickness)<br>save_table(gt,plot_height,plot_width,"plots/most_differential_genes/most_upregulated_by_log2fold/most_upregulated_genes_table.png")<br><br>top_10_genes = get_top_10_differential_genes(FALSE,"log2fold") # downregulated, by log2fold<br>gt = make_most_differential_genes_table(top_10_genes,header_size,text_size,border_thickness)<br>save_table(gt,plot_height,plot_width,"plots/most_differential_genes/most_downregulated_by_log2fold/most_downregulated_genes_table.png")<br><br></div>
    </div>
</div>

<div id="Most_Differential_Genes_Boxplots" class="plot_sections">
    <h2 class="plot_title">Most Differential Genes Boxplots
</h2>
    <div class="plot_wrapper" style="display: block;">
        <div class="plots"><h4>Upregulated By P-value
</h4><img style="width:20.0%;" src="plots/most_differential_genes/most_upregulated_by_p_value/no.1_most_upregulated_gene.png
"/><img style="width:20.0%;" src="plots/most_differential_genes/most_upregulated_by_p_value/no.2_most_upregulated_gene.png
"/><img style="width:20.0%;" src="plots/most_differential_genes/most_upregulated_by_p_value/no.3_most_upregulated_gene.png
"/><img style="width:20.0%;" src="plots/most_differential_genes/most_upregulated_by_p_value/no.4_most_upregulated_gene.png
"/><img style="width:20.0%;" src="plots/most_differential_genes/most_upregulated_by_p_value/no.5_most_upregulated_gene.png
"/><img style="width:20.0%;" src="plots/most_differential_genes/most_upregulated_by_p_value/no.6_most_upregulated_gene.png
"/><img style="width:20.0%;" src="plots/most_differential_genes/most_upregulated_by_p_value/no.7_most_upregulated_gene.png
"/><img style="width:20.0%;" src="plots/most_differential_genes/most_upregulated_by_p_value/no.8_most_upregulated_gene.png
"/><img style="width:20.0%;" src="plots/most_differential_genes/most_upregulated_by_p_value/no.9_most_upregulated_gene.png
"/><img style="width:20.0%;" src="plots/most_differential_genes/most_upregulated_by_p_value/no.10_most_upregulated_gene.png
"/><h4>Upregulated By Log2 Fold
</h4><img style="width:20.0%;" src="plots/most_differential_genes/most_upregulated_by_log2fold/no.1_most_upregulated_gene.png
"/><img style="width:20.0%;" src="plots/most_differential_genes/most_upregulated_by_log2fold/no.2_most_upregulated_gene.png
"/><img style="width:20.0%;" src="plots/most_differential_genes/most_upregulated_by_log2fold/no.3_most_upregulated_gene.png
"/><img style="width:20.0%;" src="plots/most_differential_genes/most_upregulated_by_log2fold/no.4_most_upregulated_gene.png
"/><img style="width:20.0%;" src="plots/most_differential_genes/most_upregulated_by_log2fold/no.5_most_upregulated_gene.png
"/><img style="width:20.0%;" src="plots/most_differential_genes/most_upregulated_by_log2fold/no.6_most_upregulated_gene.png
"/><img style="width:20.0%;" src="plots/most_differential_genes/most_upregulated_by_log2fold/no.7_most_upregulated_gene.png
"/><img style="width:20.0%;" src="plots/most_differential_genes/most_upregulated_by_log2fold/no.8_most_upregulated_gene.png
"/><img style="width:20.0%;" src="plots/most_differential_genes/most_upregulated_by_log2fold/no.9_most_upregulated_gene.png
"/><img style="width:20.0%;" src="plots/most_differential_genes/most_upregulated_by_log2fold/no.10_most_upregulated_gene.png
"/><h4>Downregulated By P-value
</h4><img style="width:20.0%;" src="plots/most_differential_genes/most_downregulated_by_p_value/no.1_most_downregulated_gene.png
"/><img style="width:20.0%;" src="plots/most_differential_genes/most_downregulated_by_p_value/no.2_most_downregulated_gene.png
"/><img style="width:20.0%;" src="plots/most_differential_genes/most_downregulated_by_p_value/no.3_most_downregulated_gene.png
"/><img style="width:20.0%;" src="plots/most_differential_genes/most_downregulated_by_p_value/no.4_most_downregulated_gene.png
"/><img style="width:20.0%;" src="plots/most_differential_genes/most_downregulated_by_p_value/no.5_most_downregulated_gene.png
"/><img style="width:20.0%;" src="plots/most_differential_genes/most_downregulated_by_p_value/no.6_most_downregulated_gene.png
"/><img style="width:20.0%;" src="plots/most_differential_genes/most_downregulated_by_p_value/no.7_most_downregulated_gene.png
"/><img style="width:20.0%;" src="plots/most_differential_genes/most_downregulated_by_p_value/no.8_most_downregulated_gene.png
"/><img style="width:20.0%;" src="plots/most_differential_genes/most_downregulated_by_p_value/no.9_most_downregulated_gene.png
"/><img style="width:20.0%;" src="plots/most_differential_genes/most_downregulated_by_p_value/no.10_most_downregulated_gene.png
"/><h4>Downregulated By Log2 Fold
</h4><img style="width:20.0%;" src="plots/most_differential_genes/most_downregulated_by_log2fold/no.1_most_downregulated_gene.png
"/><img style="width:20.0%;" src="plots/most_differential_genes/most_downregulated_by_log2fold/no.2_most_downregulated_gene.png
"/><img style="width:20.0%;" src="plots/most_differential_genes/most_downregulated_by_log2fold/no.3_most_downregulated_gene.png
"/><img style="width:20.0%;" src="plots/most_differential_genes/most_downregulated_by_log2fold/no.4_most_downregulated_gene.png
"/><img style="width:20.0%;" src="plots/most_differential_genes/most_downregulated_by_log2fold/no.5_most_downregulated_gene.png
"/><img style="width:20.0%;" src="plots/most_differential_genes/most_downregulated_by_log2fold/no.6_most_downregulated_gene.png
"/><img style="width:20.0%;" src="plots/most_differential_genes/most_downregulated_by_log2fold/no.7_most_downregulated_gene.png
"/><img style="width:20.0%;" src="plots/most_differential_genes/most_downregulated_by_log2fold/no.8_most_downregulated_gene.png
"/><img style="width:20.0%;" src="plots/most_differential_genes/most_downregulated_by_log2fold/no.9_most_downregulated_gene.png
"/><img style="width:20.0%;" src="plots/most_differential_genes/most_downregulated_by_log2fold/no.10_most_downregulated_gene.png
"/></div>

        <!-- plot_description is replaced by bin/plot_description/... -->
        <button class="collapsible">plot description</button>
        <div class="plot_description">Here we provide violin plots for the ten most highly upregulated and most highly downregulated genes by p-value and log2 fold change. It is often useful to visually inspect the behaviour of the samples and the general expression levels at these genes. Generally we expect significant genes to have obvious visual differences between the sample groups, and not seeing this can indicate issues with the method for calling differential expression or a too relaxed p or fold cut-off. The spread of the samples within a sample group gives an idea of sample heterogeneity at a given gene. The levels of expression at a given gene can be useful for choosing candidate genes for validation (more highly expressed genes are often easier to validate). 
</div>

        <!-- plot_legend is replaced by bin/figure_legends/... -->
        <button class="collapsible">figure legend</button>
        <div class="plot_legend"><p>Violin plots showing for each of the 10 most upregulated and downregulated genes by adjusted p-value and log2fold change. Showing expression values for all samples in the sample groups: kosalt and koc. Each dot is one sample, with sample groups given on the x-axis and gene expression on the y-axis. The mean and standard error for each sample group is given as a red dot and line.
</p></div>

        <!-- r_code is replaced by -->
        <button class="collapsible" id="r_code">r code</button>
        <div class="r_code"><span class="r_comment">##---- general libraries ----####</span><br><br>library(ggplot2)<br>library(ggrepel)<br>library(reshape)<br>library(ggridges)<br><br><span class="r_comment">##---- table libraries ----####</span><br><br>library(grid)<br>library(gridExtra)<br>library(gtable)<br>library(ggplotify)<br><br><span class="r_comment">##---- DE sample and group names ----####</span><br><br>samples = c("KOSALT12_S4_001","KOSALT15_S10_001","KOSALT19_S8_001","KOC12_S3_001","KOC15_S9_001","KOC19_S6_001")<br>sample_groups = c("KOSALT","KOC")<br>sample_groups = factor(sample_groups, levels = sample_groups)<br>sample_groupings = c("KOSALT","KOSALT","KOSALT","KOC","KOC","KOC")<br>sample_groupings = factor(sample_groupings, levels = sample_groups)<br>samples_by_sample_group = list(c("KOSALT12_S4_001","KOSALT15_S10_001","KOSALT19_S8_001"),c("KOC12_S3_001","KOC15_S9_001","KOC19_S6_001"))<br>comparisons = c("KOC vs KOSALT")<br><br><span class="r_comment">##---- sets the working directory. If you move the SL2 folder please update this path ----####</span><br><br>setwd("/home/john/Downloads/mahesh_run2/2x75/SL2/all_genes/de_workflows/KOC_vs_KOSALT")<br><br><span class="r_comment">##---- de input files ----####</span><br><br>de_annotated = read.table(file="data/de_annotated.csv", header=TRUE,row.names = 2, sep='\t', quote='',check.names = TRUE)<br>de_annotated_sig = read.table(file="data/de_annotated_significant.csv", header=TRUE,row.names = 2, sep='\t', quote='',check.names = TRUE)<br><br><span class="r_comment">##---- de parse ne matrix ----####</span><br><br>ne_matrix = de_annotated[,samples]<br>ne_matrix_sig = de_annotated_sig[,samples]<br><br><span class="r_comment">##---- transpose and scale ne matrix ----####</span><br><br>ne_matrix_transposed = data.frame(t(ne_matrix))<br>colnames(ne_matrix_transposed) = rownames(ne_matrix)<br><br>ne_matrix_scaled = data.frame(t(scale(t(ne_matrix))))<br>rownames(ne_matrix_scaled) = rownames(ne_matrix)<br>ne_matrix_scaled[do.call(cbind, lapply(ne_matrix_scaled, is.nan))] <- 0<br>ne_matrix_scaled = ne_matrix_scaled[is.finite(rowSums(ne_matrix_scaled)), ]<br><br>ne_matrix_scaled_transposed = data.frame(t(ne_matrix_scaled))<br>colnames(ne_matrix_scaled_transposed) = rownames(ne_matrix_scaled)<br><br><span class="r_comment">##---- SL2 Theme ----####</span><br><br>theme_SL2 <- function () { <br> &emsp; &emsp;theme_bw() %+replace% <br> &emsp; &emsp; &emsp; &emsp;theme(<br>	 &emsp; &emsp;panel.grid = element_blank(),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;panel.background = element_blank(),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;panel.border = element_rect(colour = "black", fill=NA, linewidth=1),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.background = element_blank(), <br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.background = element_rect(fill="transparent", colour=NA),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.key = element_rect(fill="transparent", colour=NA),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.title = element_text(size=12, margin = margin(b = 5),hjust=0,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;title = element_text(size = 12, margin = margin(b = 5),hjust=0,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.text.y = element_text(size = 10, margin = margin(r = 5),hjust=1,vjust=0.5, face="bold",colour="black"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.text.x = element_text(size = 10, margin = margin(t = 5),hjust=0.5,vjust=1, face="bold",colour="black"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.title.y = element_text(size = 11, margin = margin(r = 10),angle = 90,hjust=0.5,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.title.x = element_text(size = 11, margin = margin(t = 10),hjust=0.5,vjust=1, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.text=element_text(size=11, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.title=element_blank(), <br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.key.size=unit(2.5,"line"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.margin=unit(c(0.4,0.4,0.4,0.4), "cm")<br> &emsp; &emsp; &emsp; &emsp;)<br>}<br><br><span class="r_comment">##----- Default GGplot Colours Function -----####</span><br><br>gg_color_hue <- function(n)<br>{<br> &emsp;# default colour blind palettes<br> &emsp;cblind_palette = c("#88CCEE", "#CC6677", "#DDCC77", "#117733", "#332288", "#AA4499", "#44AA99", "#999933", "#882255", "#661100", "#6699CC", "#888888")<br><br> &emsp;# append ggplot colours if groups goes above colour blind palette length<br> &emsp;if (n <= length(cblind_palette))<br> &emsp;{<br> &emsp; &emsp;return(cblind_palette[1:n])<br> &emsp;}<br> &emsp;else<br> &emsp;{<br> &emsp; &emsp;nn = n - length(cblind_palette)<br> &emsp; &emsp;hues = seq(15, 375, length = nn + 1)<br> &emsp; &emsp;return(append(cblind_palette,hcl(h = hues, l = 65, c = 100)[1:nn]))<br> &emsp;}<br>}<br><br><span class="r_comment">##---- Default Sample and Sample Group Colours &emsp;----####</span><br><br>number_of_sample_groups = length(sample_groups)<br>default_sample_group_colours = gg_color_hue(number_of_sample_groups)<br>default_samples_colours = c(default_sample_group_colours[1],default_sample_group_colours[1],default_sample_group_colours[1],default_sample_group_colours[2],default_sample_group_colours[2],default_sample_group_colours[2])<br><br><span class="r_comment">##---- Default Sample Group Labels &emsp;----####</span><br><br>default_sample_group_labels = gsub("_"," ",c("KOSALT","KOC")) # note: changing this won't change the order the groups appear on the plots. Merely what they are labelled as.<br><br><span class="r_comment">##---- Save Plot Function -----####</span><br><br>save_plot <- function(ggp,plot_height,plot_width,plot_path)<br>{<br> &emsp;png(plot_path, height=plot_height, width=plot_width, pointsize=5)<br> &emsp;print(ggp)<br> &emsp;dev.off()<br><br> &emsp;svg(gsub(".png", ".svg", plot_path), height=plot_height/90, width=plot_width/90)<br> &emsp;print(ggp)<br> &emsp;dev.off()<br><br> &emsp;# clears all devices - as a safety measure<br> &emsp;while (dev.cur()>1) dev.off()<br>}<br><br><span class="r_comment">##---- Top 10 Differential Genes Function ----####</span><br><br>get_top_10_differential_genes <- function(direction,type) {<br><br> &emsp;if(type=="log2fold") <br> &emsp;{<br> &emsp; &emsp;top_10_genes = row.names(head(de_annotated[order(de_annotated$log2fold,decreasing=direction),],10))<br> &emsp;}<br> &emsp;if(type=="p")<br> &emsp;{<br> &emsp; &emsp;if (direction == TRUE) {de_annotated_direction = subset(de_annotated,log2fold > 0)}<br> &emsp; &emsp;else {de_annotated_direction = subset(de_annotated,log2fold < 0)}<br> &emsp; &emsp;top_10_genes = row.names(head(de_annotated_direction[order(de_annotated_direction$p,decreasing=FALSE),],10))<br> &emsp;}<br> &emsp;return(top_10_genes)<br>}<br><br><span class="r_comment">##---- Gene Expression Boxplot Function &emsp;----####</span><br><br>make_gene_expression_boxplot <- function(e_matrix, gene, box_transparency, box_width, box_line_thickness, box_colours, sample_labels, jitter_dot_size, jitter_dot_colour, jitter_dot_width,x_axis_label, y_axis_label, legend_position)<br>{<br> &emsp;# parse data<br> &emsp;gene_data = data.frame(t(e_matrix[gene,]))<br> &emsp;names(gene_data) = "expression"<br><br> &emsp;# check for negative values<br> &emsp;if (min(gene_data$expression) < 0)<br> &emsp;{<br> &emsp; &emsp;ymin = min(gene_data$expression) * 1.25<br> &emsp;}<br> &emsp;else<br> &emsp;{<br> &emsp; &emsp;ymin = 0<br> &emsp;}<br><br> &emsp;# make the plot<br> &emsp;ggp = ggplot(gene_data, aes(x=sample_groupings, y=expression, fill=sample_groupings)) +<br> &emsp; &emsp;geom_boxplot(width = box_width, alpha = box_transparency, size = box_line_thickness) +<br> &emsp; &emsp;geom_jitter(size=jitter_dot_size, colour=jitter_dot_colour, width=jitter_dot_width, height=0, show.legend = FALSE) +<br> &emsp; &emsp;scale_fill_manual(values=box_colours,labels=sample_labels) +<br> &emsp; &emsp;scale_x_discrete(labels=sample_labels) +<br> &emsp; &emsp;ylim(ymin, max(gene_data) * 1.25) +<br> &emsp; &emsp;labs(x=x_axis_label, y=y_axis_label, title= gene) +<br> &emsp; &emsp;theme_SL2() +<br> &emsp; &emsp;theme(legend.position=legend_position, axis.text.x = element_text(angle = 45, hjust = 1))<br><br> &emsp;return(ggp)<br>}<br><br><span class="r_comment">##---- Most Differential Genes (Boxplots) ----####</span><br><br>plot_height = 350<br>plot_width = 300<br>box_transparency = 0.75<br>box_width = 0.75<br>box_line_thickness = 0.5<br>box_colours = default_sample_group_colours 	# note: changing this won't change the order the groups appear in the x axis. Merely what they are coloured as.<br>sample_labels = default_sample_group_labels	# note: changing this won't change the order the groups appear in the x axis. Merely what they are named as.<br>jitter_dot_size = 2<br>jitter_dot_colour = "black"<br>jitter_dot_width = 0.2<br>x_axis_label = ""<br>y_axis_label = "expression"<br>legend_position = "none"<br><br>top_10_genes = get_top_10_differential_genes(TRUE,"p") # upregulated, by p<br>for (gene_index in 1:10)<br>{<br> &emsp;gene = top_10_genes[gene_index]<br> &emsp;ggp = make_gene_expression_boxplot(ne_matrix, gene, box_transparency, box_width, box_line_thickness, box_colours, sample_labels, jitter_dot_size, jitter_dot_colour, jitter_dot_width,x_axis_label, y_axis_label, legend_position)<br> &emsp;save_plot(ggp,plot_height,plot_width,paste("plots/most_differential_genes/most_upregulated_by_p_value/no.",gene_index,"_most_upregulated_gene.png",sep=""))<br>}<br><br>top_10_genes = get_top_10_differential_genes(FALSE,"p") # downregulated, by p<br>for (gene_index in 1:10)<br>{<br> &emsp;gene = top_10_genes[gene_index]<br> &emsp;ggp = make_gene_expression_boxplot(ne_matrix, gene, box_transparency, box_width, box_line_thickness, box_colours, sample_labels, jitter_dot_size, jitter_dot_colour, jitter_dot_width,x_axis_label, y_axis_label, legend_position)<br> &emsp;save_plot(ggp,plot_height,plot_width,paste("plots/most_differential_genes/most_downregulated_by_p_value/no.",gene_index,"_most_downregulated_gene.png",sep=""))<br>}<br><br>top_10_genes = get_top_10_differential_genes(TRUE,"log2fold") # upregulated, by log2fold<br>for (gene_index in 1:10)<br>{<br> &emsp;gene = top_10_genes[gene_index]<br> &emsp;ggp = make_gene_expression_boxplot(ne_matrix, gene, box_transparency, box_width, box_line_thickness, box_colours, sample_labels, jitter_dot_size, jitter_dot_colour, jitter_dot_width,x_axis_label, y_axis_label, legend_position)<br> &emsp;save_plot(ggp,plot_height,plot_width,paste("plots/most_differential_genes/most_upregulated_by_log2fold/no.",gene_index,"_most_upregulated_gene.png",sep=""))<br>}<br><br>top_10_genes = get_top_10_differential_genes(FALSE,"log2fold") # downregulated, by log2fold<br>for (gene_index in 1:10)<br>{<br> &emsp;gene = top_10_genes[gene_index]<br> &emsp;ggp = make_gene_expression_boxplot(ne_matrix, gene, box_transparency, box_width, box_line_thickness, box_colours, sample_labels, jitter_dot_size, jitter_dot_colour, jitter_dot_width,x_axis_label, y_axis_label, legend_position)<br> &emsp; &emsp; &emsp; &emsp; save_plot(ggp,plot_height,plot_width,paste("plots/most_differential_genes/most_downregulated_by_log2fold/no.",gene_index,"_most_downregulated_gene.png",sep=""))<br>}<br><br></div>
    </div>
</div>

<div id="Enrichment_Summary" class="plot_sections">
    <h2 class="plot_title">Enrichment Summary
</h2>
    <div class="plot_wrapper" style="display: block;">
        <div class="plots"><h4>Most Enriched Amongst All Significant Genes (STRING_11.5)
</h4><img style="width:50.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_all_significant_genes/enriched_gene_sets_barchart.png
"/><img style="width:50.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_all_significant_genes/enriched_gene_sets_ridgeplot_sig.png
"/><h4>Most Enriched Amongst Upregulated Genes (STRING_11.5)
</h4><img style="width:50.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_upregulated_genes/enriched_gene_sets_barchart.png
"/><img style="width:50.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_upregulated_genes/enriched_gene_sets_ridgeplot_sig.png
"/><h4>Most Enriched Amongst Downregulated Genes (STRING_11.5)
</h4><img style="width:50.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_downregulated_genes/enriched_gene_sets_barchart.png
"/><img style="width:50.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_downregulated_genes/enriched_gene_sets_ridgeplot_sig.png
"/></div>

        <!-- plot_description is replaced by bin/plot_description/... -->
        <button class="collapsible">plot description</button>
        <div class="plot_description">One of the core aims of differential expression analysis is to understand which pathways and areas of biology any differentially expressed genes represent. One method to determine this is Over Representation Analysis (ORA). In ORA the significantly different genes are compared to a database (e.g. GO terms, KEGG) of pre-defined lists of genes (gene-sets) - each of which represents a specific pathway or area of biology (e.g. Cell Cycle, TNF signalling). A hyper-geometric test is then used to determine whether each gene-set is enriched or not for the significantly differential genes. The most enriched gene-sets have the lowest p-values.
<br />
<br />
Here we provide bar charts showing ten most enriched gene-sets when using: all significantly differential genes, significantly upregulated genes, and significantly downregulated genes. The ten-most enriched gene-sets and the strength of enrichment gives a useful snapshot as to the pathways or general areas of biology that are most affected by the differential expression, and whether they are being up or downregulated. It can be interesting to note whether different gene-sets are being up or downregulated.
</div>

        <!-- plot_legend is replaced by bin/figure_legends/... -->
        <button class="collapsible">figure legend</button>
        <div class="plot_legend"><p>Summary bar chart of the ten most enriched genes sets for: all significantly differentially expressed genes (p.adj < 0.05, absolute log2 fold > 1.0) between koc and kosalt, significantly upregulated genes and significantly downregulated genes. Upregulated genes are higher expression in koc than in kosalt. For each plot the -log10 enrichment p-value is given on the x-axis and the gene set name on the y-axis. Data labels represent the number of significantly differential expressed genes within each gene set. Enrichment analysis was performed using Hypergeometric Gene Set Enrichment on the gene set databases STRING 11.5. Significantly enriched gene sets (p.adj < 0.05, absolute log2 fold > 0.0) are coloured red and non-significant black.
</p></div>

        <!-- r_code is replaced by -->
        <button class="collapsible" id="r_code">r code</button>
        <div class="r_code"><span class="r_comment">##---- general libraries ----####</span><br><br>library(ggplot2)<br>library(ggrepel)<br>library(reshape)<br>library(ggridges)<br><br><span class="r_comment">##---- DE sample and group names ----####</span><br><br>samples = c("KOSALT12_S4_001","KOSALT15_S10_001","KOSALT19_S8_001","KOC12_S3_001","KOC15_S9_001","KOC19_S6_001")<br>sample_groups = c("KOSALT","KOC")<br>sample_groups = factor(sample_groups, levels = sample_groups)<br>sample_groupings = c("KOSALT","KOSALT","KOSALT","KOC","KOC","KOC")<br>sample_groupings = factor(sample_groupings, levels = sample_groups)<br>samples_by_sample_group = list(c("KOSALT12_S4_001","KOSALT15_S10_001","KOSALT19_S8_001"),c("KOC12_S3_001","KOC15_S9_001","KOC19_S6_001"))<br>comparisons = c("KOC vs KOSALT")<br><br><span class="r_comment">##---- sets the working directory. If you move the SL2 folder please update this path ----####</span><br><br>setwd("/home/john/Downloads/mahesh_run2/2x75/SL2/all_genes/de_workflows/KOC_vs_KOSALT")<br><br><span class="r_comment">##---- de hypergeometric gene set enrichment files ----####</span><br><br>STRING_11.5_all_gene_sets = read.table(file="data/statistical_analysis/over_representation_analysis/STRING_11.5/all_significant_genes/all_gene_sets_results.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br>STRING_11.5_all_signficant_genes_enriched_gene_sets = read.table(file="data/statistical_analysis/over_representation_analysis/STRING_11.5/all_significant_genes/enriched_gene_sets_results.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br>STRING_11.5_all_signficant_genes_underenriched_gene_sets = read.table(file="data/statistical_analysis/over_representation_analysis/STRING_11.5/all_significant_genes/underenriched_gene_sets_results.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br>STRING_11.5_upregulated_gene_sets = read.table(file="data/statistical_analysis/over_representation_analysis/STRING_11.5/upregulated_genes/all_gene_sets_results.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br>STRING_11.5_upregulated_enriched_gene_sets = read.table(file="data/statistical_analysis/over_representation_analysis/STRING_11.5/upregulated_genes/enriched_gene_sets_results.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br>STRING_11.5_upregulated_underenriched_gene_sets = read.table(file="data/statistical_analysis/over_representation_analysis/STRING_11.5/upregulated_genes/underenriched_gene_sets_results.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br>STRING_11.5_downregulated_gene_sets = read.table(file="data/statistical_analysis/over_representation_analysis/STRING_11.5/downregulated_genes/all_gene_sets_results.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br>STRING_11.5_downregulated_enriched_gene_sets = read.table(file="data/statistical_analysis/over_representation_analysis/STRING_11.5/downregulated_genes/enriched_gene_sets_results.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br>STRING_11.5_downregulated_underenriched_gene_sets = read.table(file="data/statistical_analysis/over_representation_analysis/STRING_11.5/downregulated_genes/underenriched_gene_sets_results.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br><br><span class="r_comment">##---- SL2 Theme ----####</span><br><br>theme_SL2 <- function () { <br> &emsp; &emsp;theme_bw() %+replace% <br> &emsp; &emsp; &emsp; &emsp;theme(<br>	 &emsp; &emsp;panel.grid = element_blank(),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;panel.background = element_blank(),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;panel.border = element_rect(colour = "black", fill=NA, linewidth=1),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.background = element_blank(), <br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.background = element_rect(fill="transparent", colour=NA),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.key = element_rect(fill="transparent", colour=NA),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.title = element_text(size=12, margin = margin(b = 5),hjust=0,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;title = element_text(size = 12, margin = margin(b = 5),hjust=0,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.text.y = element_text(size = 10, margin = margin(r = 5),hjust=1,vjust=0.5, face="bold",colour="black"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.text.x = element_text(size = 10, margin = margin(t = 5),hjust=0.5,vjust=1, face="bold",colour="black"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.title.y = element_text(size = 11, margin = margin(r = 10),angle = 90,hjust=0.5,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.title.x = element_text(size = 11, margin = margin(t = 10),hjust=0.5,vjust=1, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.text=element_text(size=11, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.title=element_blank(), <br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.key.size=unit(2.5,"line"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.margin=unit(c(0.4,0.4,0.4,0.4), "cm")<br> &emsp; &emsp; &emsp; &emsp;)<br>}<br><br><span class="r_comment">##---- Default Two Way Significance Colours ----####</span><br><br>default_2way_sig_colour = "coral3"<br>default_2way_non_sig_colour = "grey50"<br><br><span class="r_comment">##---- Save Plot Function -----####</span><br><br>save_plot <- function(ggp,plot_height,plot_width,plot_path)<br>{<br> &emsp;png(plot_path, height=plot_height, width=plot_width, pointsize=5)<br> &emsp;print(ggp)<br> &emsp;dev.off()<br><br> &emsp;svg(gsub(".png", ".svg", plot_path), height=plot_height/90, width=plot_width/90)<br> &emsp;print(ggp)<br> &emsp;dev.off()<br><br> &emsp;# clears all devices - as a safety measure<br> &emsp;while (dev.cur()>1) dev.off()<br>}<br><br><span class="r_comment">##---- Top 10 Gene Sets by ORA P-value Function ----####</span><br><br>get_top10_oras_by_p_value <- function(gene_sets,enriched_gene_sets,type) {<br><br> &emsp;if(type=="enrichment")<br> &emsp;{<br> &emsp; &emsp;top_10_gene_sets = head(gene_sets[order(gene_sets$enrichment_p_value,decreasing=FALSE),],10)<br> &emsp; &emsp;top_10_gene_sets = top_10_gene_sets[,c("gene_set","enrichment_p_value","overlapping_genes","log2_fold_enrichment","overlapping_gene_names")]<br> &emsp;}<br> &emsp;if(type=="underenrichment")<br> &emsp;{<br> &emsp; &emsp;top_10_gene_sets = head(gene_sets[order(gene_sets$underenrichment_p_value,decreasing=FALSE),],10)<br> &emsp; &emsp;top_10_gene_sets = top_10_gene_sets[,c("gene_set","underenrichment_p_value","overlapping_genes","log2_fold_enrichment","overlapping_gene_names")]<br> &emsp;}<br> &emsp;top_10_gene_sets$significant = top_10_gene_sets$gene_set %in% enriched_gene_sets$gene_set<br> &emsp;top_10_gene_sets$significant = as.character(top_10_gene_sets$significant)<br> &emsp;colnames(top_10_gene_sets) = c("gene_set","p","overlapping_genes","log2_fold_enrichment","overlapping_gene_names","significant")<br> &emsp;return(top_10_gene_sets)<br>}<br><br><span class="r_comment">##---- ORA Gene Sets Bar Chart Function ----####</span><br><br>make_oras_bar_chart <- function(top_10_gene_sets, x_axis_label, y_axis_label, non_significant_colour, significant_colour, bar_outline_size, bar_transparency, legend_position, significant_name, non_significant_name, data_label_size)<br>{ <br> &emsp;top_10_gene_sets = top_10_gene_sets[seq(dim(top_10_gene_sets)[1],1),]<br> &emsp;ggp = ggplot(data=top_10_gene_sets , aes(x=gene_set, y=-log10(p), fill=significant,group=significant)) +<br> &emsp; &emsp; &emsp; &emsp;geom_bar(colour="black",stat="identity", position = "dodge", alpha = bar_transparency) + <br> &emsp; &emsp; &emsp; &emsp;coord_flip() + <br> &emsp; &emsp; &emsp; &emsp;scale_x_discrete(limits = top_10_gene_sets$gene_set) + <br> &emsp; &emsp; &emsp; &emsp;ylab(x_axis_label) &emsp;+ <br> &emsp; &emsp; &emsp; &emsp;xlab(y_axis_label) + <br> &emsp; &emsp; &emsp; &emsp;scale_fill_manual(values=c(non_significant_colour,significant_colour),breaks=c("TRUE","FALSE"),labels=c(significant_name, non_significant_name)) + <br> &emsp; &emsp; &emsp; &emsp;theme_SL2() + <br> &emsp; &emsp; &emsp; &emsp;theme(axis.text.y = element_text(hjust=1),legend.position=legend_position, legend.spacing.x = unit(0.15, 'cm')) + <br> &emsp; &emsp; &emsp; &emsp;scale_y_continuous(expand=c(0,0), limits=c(0,max(-log10(top_10_gene_sets$p)*1.25))) + <br> &emsp; &emsp; &emsp; &emsp;geom_text(aes(label=overlapping_genes), position=position_dodge(width=0.9), size=data_label_size, show.legend = FALSE, hjust=-0.5)<br><br> &emsp;return(ggp)<br>}<br><br><span class="r_comment">##----ORA Enriched Gene Sets (Bar Charts) ----####</span><br><br>plot_height = 350<br>plot_width = 1000<br>x_axis_label = "-log10 p-value"<br>y_axis_label = ""<br>non_significant_colour = default_2way_sig_colour<br>significant_colour = default_2way_non_sig_colour<br>bar_outline_size = 1<br>bar_transparency = 0.75<br>legend_position = "bottom"<br>significant_name = "significant"<br>non_significant_name = "non-significant"<br>data_label_size = 5<br><br>top_10_gene_sets = get_top10_oras_by_p_value(STRING_11.5_all_gene_sets,STRING_11.5_all_signficant_genes_enriched_gene_sets,"enrichment")<br>ggp = make_oras_bar_chart(top_10_gene_sets,x_axis_label,y_axis_label,non_significant_colour,significant_colour,bar_outline_size,bar_transparency,legend_position,significant_name,non_significant_name,data_label_size)<br>save_plot(ggp,plot_height,plot_width,"plots/over_representation_analysis/STRING_11.5/enriched_amongst_all_significant_genes/enriched_gene_sets_barchart.png")<br><br>top_10_gene_sets = get_top10_oras_by_p_value(STRING_11.5_downregulated_gene_sets,STRING_11.5_downregulated_enriched_gene_sets,"enrichment")<br>ggp = make_oras_bar_chart(top_10_gene_sets,x_axis_label,y_axis_label,non_significant_colour,significant_colour,bar_outline_size,bar_transparency,legend_position,significant_name,non_significant_name,data_label_size)<br>save_plot(ggp,plot_height,plot_width,"plots/over_representation_analysis/STRING_11.5/enriched_amongst_downregulated_genes/enriched_gene_sets_barchart.png")<br><br>top_10_gene_sets = get_top10_oras_by_p_value(STRING_11.5_upregulated_gene_sets,STRING_11.5_upregulated_enriched_gene_sets,"enrichment")<br>ggp = make_oras_bar_chart(top_10_gene_sets,x_axis_label,y_axis_label,non_significant_colour,significant_colour,bar_outline_size,bar_transparency,legend_position,significant_name,non_significant_name,data_label_size)<br>save_plot(ggp,plot_height,plot_width,"plots/over_representation_analysis/STRING_11.5/enriched_amongst_upregulated_genes/enriched_gene_sets_barchart.png")<br><br></div>
    </div>
</div>

<div id="Enrichment_Boxplots" class="plot_sections">
    <h2 class="plot_title">Enrichment Boxplots
</h2>
    <div class="plot_wrapper" style="display: block;">
        <div class="plots"><h4>Most Enriched Amongst All Significant Genes (STRING_11.5)
</h4><img style="width:50.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_all_significant_genes/no.1_most_enriched_gene_set.png
"/><img style="width:50.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_all_significant_genes/no.2_most_enriched_gene_set.png
"/><img style="width:50.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_all_significant_genes/no.3_most_enriched_gene_set.png
"/><img style="width:50.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_all_significant_genes/no.4_most_enriched_gene_set.png
"/><img style="width:50.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_all_significant_genes/no.5_most_enriched_gene_set.png
"/><img style="width:50.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_all_significant_genes/no.6_most_enriched_gene_set.png
"/><img style="width:50.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_all_significant_genes/no.7_most_enriched_gene_set.png
"/><img style="width:50.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_all_significant_genes/no.8_most_enriched_gene_set.png
"/><img style="width:50.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_all_significant_genes/no.9_most_enriched_gene_set.png
"/><img style="width:50.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_all_significant_genes/no.10_most_enriched_gene_set.png
"/><h4>Most Enriched Amongst Upregulated  Genes (STRING_11.5)
</h4><img style="width:50.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_upregulated_genes/no.1_most_enriched_gene_set.png
"/><img style="width:50.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_upregulated_genes/no.2_most_enriched_gene_set.png
"/><img style="width:50.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_upregulated_genes/no.3_most_enriched_gene_set.png
"/><img style="width:50.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_upregulated_genes/no.4_most_enriched_gene_set.png
"/><img style="width:50.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_upregulated_genes/no.5_most_enriched_gene_set.png
"/><img style="width:50.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_upregulated_genes/no.6_most_enriched_gene_set.png
"/><img style="width:50.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_upregulated_genes/no.7_most_enriched_gene_set.png
"/><img style="width:50.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_upregulated_genes/no.8_most_enriched_gene_set.png
"/><img style="width:50.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_upregulated_genes/no.9_most_enriched_gene_set.png
"/><img style="width:50.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_upregulated_genes/no.10_most_enriched_gene_set.png
"/><h4>Most Enriched Amongst Downregulated Genes (STRING_11.5)
</h4><img style="width:50.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_downregulated_genes/no.1_most_enriched_gene_set.png
"/><img style="width:50.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_downregulated_genes/no.2_most_enriched_gene_set.png
"/><img style="width:50.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_downregulated_genes/no.3_most_enriched_gene_set.png
"/><img style="width:50.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_downregulated_genes/no.4_most_enriched_gene_set.png
"/><img style="width:50.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_downregulated_genes/no.5_most_enriched_gene_set.png
"/><img style="width:50.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_downregulated_genes/no.6_most_enriched_gene_set.png
"/><img style="width:50.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_downregulated_genes/no.7_most_enriched_gene_set.png
"/><img style="width:50.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_downregulated_genes/no.8_most_enriched_gene_set.png
"/><img style="width:50.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_downregulated_genes/no.9_most_enriched_gene_set.png
"/><img style="width:50.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_downregulated_genes/no.10_most_enriched_gene_set.png
"/></div>

        <!-- plot_description is replaced by bin/plot_description/... -->
        <button class="collapsible">plot description</button>
        <div class="plot_description">One of the core aims of differential expression analysis is to understand which pathways and areas of biology any differentially expresseOver Representation Analysis (ORA). In ORA the significantly different genes are compared to a database (e.g. GO terms, KEGG) of pre-defined lists of genes (gene-sets) - each of which represents a specific pathway or area of biology (e.g. Cell Cycle, TNF signalling). A hyper-geometric test is then used to determine whether each gene-set is enriched or not for the significantly differential genes. The most enriched gene-sets have the lowest p-values.
<br />
<br />
Here we provide boxplots showing the ten most highly enriched gene-sets, and the expression values for the significantly differential genes within each gene-set. The expression values are given as z-scores. Boxplots of the top ten enriched gene-sets are given for: all significantly differential genes, significantly upregulated genes and significantly downregulated genes. As well providing a useful snapshot of the pathways or general areas of biology that are most affected by the differential expression this also shows how convincing the differences are, and which individual genes within the gene-set are most affected.
</div>

        <!-- plot_legend is replaced by bin/figure_legends/... -->
        <button class="collapsible">figure legend</button>
        <div class="plot_legend"><p>Summary boxplots of the ten most enriched genes sets for: all significantly differentially expressed genes (p.adj < 0.05, absolute log2 fold > 1.0) between koc and kosalt, significantly upregulated genes and significantly downregulated genes. Upregulated genes are higher expression in koc than in kosalt. For each plot the significantly differential genes within the gene set are given on the x-axis and the expression values on the y-axis. Boxes are shown for the following sample groups: kosalt and koc. Expression levels have been scaled into per gene zâˆ’scores. Enrichment analysis was performed using Hypergeometric Gene Set Enrichment on the gene set databases STRING 11.5.
</p></div>

        <!-- r_code is replaced by -->
        <button class="collapsible" id="r_code">r code</button>
        <div class="r_code"><span class="r_comment">##---- general libraries ----####</span><br><br>library(ggplot2)<br>library(ggrepel)<br>library(reshape)<br>library(ggridges)<br><br><span class="r_comment">##---- DE sample and group names ----####</span><br><br>samples = c("KOSALT12_S4_001","KOSALT15_S10_001","KOSALT19_S8_001","KOC12_S3_001","KOC15_S9_001","KOC19_S6_001")<br>sample_groups = c("KOSALT","KOC")<br>sample_groups = factor(sample_groups, levels = sample_groups)<br>sample_groupings = c("KOSALT","KOSALT","KOSALT","KOC","KOC","KOC")<br>sample_groupings = factor(sample_groupings, levels = sample_groups)<br>samples_by_sample_group = list(c("KOSALT12_S4_001","KOSALT15_S10_001","KOSALT19_S8_001"),c("KOC12_S3_001","KOC15_S9_001","KOC19_S6_001"))<br>comparisons = c("KOC vs KOSALT")<br><br><span class="r_comment">##---- sets the working directory. If you move the SL2 folder please update this path ----####</span><br><br>setwd("/home/john/Downloads/mahesh_run2/2x75/SL2/all_genes/de_workflows/KOC_vs_KOSALT")<br><br><span class="r_comment">##---- de input files ----####</span><br><br>de_annotated = read.table(file="data/de_annotated.csv", header=TRUE,row.names = 2, sep='\t', quote='',check.names = TRUE)<br>de_annotated_sig = read.table(file="data/de_annotated_significant.csv", header=TRUE,row.names = 2, sep='\t', quote='',check.names = TRUE)<br><br><span class="r_comment">##---- de hypergeometric gene set enrichment files ----####</span><br><br>STRING_11.5_all_gene_sets = read.table(file="data/statistical_analysis/over_representation_analysis/STRING_11.5/all_significant_genes/all_gene_sets_results.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br>STRING_11.5_all_signficant_genes_enriched_gene_sets = read.table(file="data/statistical_analysis/over_representation_analysis/STRING_11.5/all_significant_genes/enriched_gene_sets_results.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br>STRING_11.5_all_signficant_genes_underenriched_gene_sets = read.table(file="data/statistical_analysis/over_representation_analysis/STRING_11.5/all_significant_genes/underenriched_gene_sets_results.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br>STRING_11.5_upregulated_gene_sets = read.table(file="data/statistical_analysis/over_representation_analysis/STRING_11.5/upregulated_genes/all_gene_sets_results.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br>STRING_11.5_upregulated_enriched_gene_sets = read.table(file="data/statistical_analysis/over_representation_analysis/STRING_11.5/upregulated_genes/enriched_gene_sets_results.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br>STRING_11.5_upregulated_underenriched_gene_sets = read.table(file="data/statistical_analysis/over_representation_analysis/STRING_11.5/upregulated_genes/underenriched_gene_sets_results.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br>STRING_11.5_downregulated_gene_sets = read.table(file="data/statistical_analysis/over_representation_analysis/STRING_11.5/downregulated_genes/all_gene_sets_results.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br>STRING_11.5_downregulated_enriched_gene_sets = read.table(file="data/statistical_analysis/over_representation_analysis/STRING_11.5/downregulated_genes/enriched_gene_sets_results.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br>STRING_11.5_downregulated_underenriched_gene_sets = read.table(file="data/statistical_analysis/over_representation_analysis/STRING_11.5/downregulated_genes/underenriched_gene_sets_results.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br><br><span class="r_comment">##---- de parse ne matrix ----####</span><br><br>ne_matrix = de_annotated[,samples]<br>ne_matrix_sig = de_annotated_sig[,samples]<br><br><span class="r_comment">##---- transpose and scale ne matrix ----####</span><br><br>ne_matrix_transposed = data.frame(t(ne_matrix))<br>colnames(ne_matrix_transposed) = rownames(ne_matrix)<br><br>ne_matrix_scaled = data.frame(t(scale(t(ne_matrix))))<br>rownames(ne_matrix_scaled) = rownames(ne_matrix)<br>ne_matrix_scaled[do.call(cbind, lapply(ne_matrix_scaled, is.nan))] <- 0<br>ne_matrix_scaled = ne_matrix_scaled[is.finite(rowSums(ne_matrix_scaled)), ]<br><br>ne_matrix_scaled_transposed = data.frame(t(ne_matrix_scaled))<br>colnames(ne_matrix_scaled_transposed) = rownames(ne_matrix_scaled)<br><br><span class="r_comment">##---- SL2 Theme ----####</span><br><br>theme_SL2 <- function () { <br> &emsp; &emsp;theme_bw() %+replace% <br> &emsp; &emsp; &emsp; &emsp;theme(<br>	 &emsp; &emsp;panel.grid = element_blank(),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;panel.background = element_blank(),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;panel.border = element_rect(colour = "black", fill=NA, linewidth=1),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.background = element_blank(), <br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.background = element_rect(fill="transparent", colour=NA),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.key = element_rect(fill="transparent", colour=NA),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.title = element_text(size=12, margin = margin(b = 5),hjust=0,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;title = element_text(size = 12, margin = margin(b = 5),hjust=0,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.text.y = element_text(size = 10, margin = margin(r = 5),hjust=1,vjust=0.5, face="bold",colour="black"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.text.x = element_text(size = 10, margin = margin(t = 5),hjust=0.5,vjust=1, face="bold",colour="black"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.title.y = element_text(size = 11, margin = margin(r = 10),angle = 90,hjust=0.5,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.title.x = element_text(size = 11, margin = margin(t = 10),hjust=0.5,vjust=1, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.text=element_text(size=11, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.title=element_blank(), <br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.key.size=unit(2.5,"line"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.margin=unit(c(0.4,0.4,0.4,0.4), "cm")<br> &emsp; &emsp; &emsp; &emsp;)<br>}<br><br><span class="r_comment">##----- Default GGplot Colours Function -----####</span><br><br>gg_color_hue <- function(n)<br>{<br> &emsp;# default colour blind palettes<br> &emsp;cblind_palette = c("#88CCEE", "#CC6677", "#DDCC77", "#117733", "#332288", "#AA4499", "#44AA99", "#999933", "#882255", "#661100", "#6699CC", "#888888")<br><br> &emsp;# append ggplot colours if groups goes above colour blind palette length<br> &emsp;if (n <= length(cblind_palette))<br> &emsp;{<br> &emsp; &emsp;return(cblind_palette[1:n])<br> &emsp;}<br> &emsp;else<br> &emsp;{<br> &emsp; &emsp;nn = n - length(cblind_palette)<br> &emsp; &emsp;hues = seq(15, 375, length = nn + 1)<br> &emsp; &emsp;return(append(cblind_palette,hcl(h = hues, l = 65, c = 100)[1:nn]))<br> &emsp;}<br>}<br><br><span class="r_comment">##---- Default Sample and Sample Group Colours &emsp;----####</span><br><br>number_of_sample_groups = length(sample_groups)<br>default_sample_group_colours = gg_color_hue(number_of_sample_groups)<br>default_samples_colours = c(default_sample_group_colours[1],default_sample_group_colours[1],default_sample_group_colours[1],default_sample_group_colours[2],default_sample_group_colours[2],default_sample_group_colours[2])<br><br><span class="r_comment">##---- Default Sample Group Labels &emsp;----####</span><br><br>default_sample_group_labels = gsub("_"," ",c("KOSALT","KOC")) # note: changing this won't change the order the groups appear on the plots. Merely what they are labelled as.<br><br><span class="r_comment">##---- Save Plot Function -----####</span><br><br>save_plot <- function(ggp,plot_height,plot_width,plot_path)<br>{<br> &emsp;png(plot_path, height=plot_height, width=plot_width, pointsize=5)<br> &emsp;print(ggp)<br> &emsp;dev.off()<br><br> &emsp;svg(gsub(".png", ".svg", plot_path), height=plot_height/90, width=plot_width/90)<br> &emsp;print(ggp)<br> &emsp;dev.off()<br><br> &emsp;# clears all devices - as a safety measure<br> &emsp;while (dev.cur()>1) dev.off()<br>}<br><br><span class="r_comment">##---- Top 10 Gene Sets by ORA P-value Function ----####</span><br><br>get_top10_oras_by_p_value <- function(gene_sets,enriched_gene_sets,type) {<br><br> &emsp;if(type=="enrichment")<br> &emsp;{<br> &emsp; &emsp;top_10_gene_sets = head(gene_sets[order(gene_sets$enrichment_p_value,decreasing=FALSE),],10)<br> &emsp; &emsp;top_10_gene_sets = top_10_gene_sets[,c("gene_set","enrichment_p_value","overlapping_genes","log2_fold_enrichment","overlapping_gene_names")]<br> &emsp;}<br> &emsp;if(type=="underenrichment")<br> &emsp;{<br> &emsp; &emsp;top_10_gene_sets = head(gene_sets[order(gene_sets$underenrichment_p_value,decreasing=FALSE),],10)<br> &emsp; &emsp;top_10_gene_sets = top_10_gene_sets[,c("gene_set","underenrichment_p_value","overlapping_genes","log2_fold_enrichment","overlapping_gene_names")]<br> &emsp;}<br> &emsp;top_10_gene_sets$significant = top_10_gene_sets$gene_set %in% enriched_gene_sets$gene_set<br> &emsp;top_10_gene_sets$significant = as.character(top_10_gene_sets$significant)<br> &emsp;colnames(top_10_gene_sets) = c("gene_set","p","overlapping_genes","log2_fold_enrichment","overlapping_gene_names","significant")<br> &emsp;return(top_10_gene_sets)<br>}<br><br><span class="r_comment">##---- ORA Gene Set Boxplot Function ----####</span><br><br>make_ora_boxplot <- function(gene_set,plot_title,box_transparency,box_line_thickness,box_colours,box_labels,x_axis_label,y_axis_label,legend_position)<br>{<br> &emsp;genes_in_gene_set = unlist(lapply(gene_set$overlapping_gene_names, function(x) unlist(strsplit(as.character(x),","))))<br> &emsp;if(length(genes_in_gene_set) > 1) <br> &emsp;{<br> &emsp; &emsp;ne_matrix_scaled_transposed_genes_in_gene_set = ne_matrix_scaled_transposed[,genes_in_gene_set]<br> &emsp; &emsp;ne_matrix_scaled_transposed_genes_in_gene_set$sample_groupings = sample_groupings<br> &emsp; &emsp;ne_matrix_scaled_transposed_genes_in_gene_set_melted = melt(ne_matrix_scaled_transposed_genes_in_gene_set,id.vars = "sample_groupings")<br> &emsp; &emsp;<br> &emsp; &emsp;ggp = ggplot(ne_matrix_scaled_transposed_genes_in_gene_set_melted, aes(x=variable, y=value, fill=sample_groupings)) + <br> &emsp; &emsp; &emsp;geom_boxplot(position=position_dodge(0.75),outlier.shape = NA, alpha=box_transparency,size=box_line_thickness) + <br> &emsp; &emsp; &emsp;theme_SL2() + <br> &emsp; &emsp; &emsp;theme(axis.text.x = element_text(angle = 90, hjust=1, vjust=0.5),legend.position=legend_position) + <br> &emsp; &emsp; &emsp;ggtitle(plot_title) + <br> &emsp; &emsp; &emsp;scale_color_manual(values=box_colours,breaks=sample_groups,labels=box_labels) + <br> &emsp; &emsp; &emsp;scale_fill_manual(values=box_colours,breaks=sample_groups,labels=box_labels) + <br> &emsp; &emsp; &emsp;xlab(x_axis_label) + <br> &emsp; &emsp; &emsp;ylab(y_axis_label)<br> &emsp; &emsp;<br> &emsp; &emsp;return(ggp)<br> &emsp;}<br> &emsp;return(ggplot(data.frame()) + theme_SL2() + geom_blank() + ggtitle("There were too few genes to sensibly plot this."))<br>}<br><br><span class="r_comment">##---- ORA Enriched Gene Sets (Boxplot) ----####</span><br><br>plot_height = 250<br>plot_width = 1000<br>box_transparency = 0.75<br>box_line_thickness = 0.75<br>box_colours = default_sample_group_colours 	# note: changing this won't change the order the groups appear in the x axis. Merely what they are coloured as.<br>box_labels = default_sample_group_labels	# note: changing this won't change the order the groups appear in the x axis. Merely what they are named as.<br>x_axis_label = ""<br>y_axis_label = "expression (z-score)"<br>legend_position = "right"<br><br>top_10_gene_sets = get_top10_oras_by_p_value(STRING_11.5_all_gene_sets,STRING_11.5_all_signficant_genes_enriched_gene_sets,"enrichment")<br>for(gene_set_index in 1:10)<br>{<br> &emsp;gene_set = top_10_gene_sets[gene_set_index,]<br> &emsp;plot_title = gene_set[["gene_set"]]<br> &emsp;ggp = make_ora_boxplot(gene_set,plot_title,box_transparency,box_line_thickness,box_colours,box_labels,x_axis_label,y_axis_label,legend_position)<br> &emsp;save_plot(ggp,plot_height,plot_width,paste("plots/over_representation_analysis/STRING_11.5/enriched_amongst_all_significant_genes/no.",gene_set_index,"_most_enriched_gene_set.png",sep=""))<br>}<br><br>top_10_gene_sets = get_top10_oras_by_p_value(STRING_11.5_downregulated_gene_sets,STRING_11.5_downregulated_enriched_gene_sets,"enrichment")<br>for(gene_set_index in 1:10)<br>{<br> &emsp;gene_set = top_10_gene_sets[gene_set_index,]<br> &emsp;plot_title = gene_set[["gene_set"]]<br> &emsp;ggp = make_ora_boxplot(gene_set,plot_title,box_transparency,box_line_thickness,box_colours,box_labels,x_axis_label,y_axis_label,legend_position)<br> &emsp;save_plot(ggp,plot_height,plot_width,paste("plots/over_representation_analysis/STRING_11.5/enriched_amongst_downregulated_genes/no.",gene_set_index,"_most_enriched_gene_set.png",sep=""))<br>}<br><br>top_10_gene_sets = get_top10_oras_by_p_value(STRING_11.5_upregulated_gene_sets,STRING_11.5_upregulated_enriched_gene_sets,"enrichment")<br>for(gene_set_index in 1:10)<br>{<br> &emsp;gene_set = top_10_gene_sets[gene_set_index,]<br> &emsp;plot_title = gene_set[["gene_set"]]<br> &emsp;ggp = make_ora_boxplot(gene_set,plot_title,box_transparency,box_line_thickness,box_colours,box_labels,x_axis_label,y_axis_label,legend_position)<br> &emsp;save_plot(ggp,plot_height,plot_width,paste("plots/over_representation_analysis/STRING_11.5/enriched_amongst_upregulated_genes/no.",gene_set_index,"_most_enriched_gene_set.png",sep=""))<br>}<br><br><br></div>
    </div>
</div>

<div id="Enrichment_Network_Plots" class="plot_sections">
    <h2 class="plot_title">Enrichment Network Plots
</h2>
    <div class="plot_wrapper" style="display: block;">
        <div class="plots"><h4>Most Enriched Amongst All Significant Genes (STRING_11.5)
</h4><img style="width:25.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_all_significant_genes/enriched_gene_sets_network_plot.png
"/><img style="width:25.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_all_significant_genes/enriched_gene_sets_network_plot_unlabelled.png
"/><img style="width:25.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_all_significant_genes/enriched_gene_sets_network_cluster_wordcloud.png
"/><img style="width:25.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_all_significant_genes/enriched_gene_sets_network_cluster_genes.png
"/><h4>Most Enriched Amongst Upregulated Genes (STRING_11.5)
</h4><img style="width:25.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_upregulated_genes/enriched_gene_sets_network_plot.png
"/><img style="width:25.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_upregulated_genes/enriched_gene_sets_network_plot_unlabelled.png
"/><img style="width:25.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_upregulated_genes/enriched_gene_sets_network_cluster_wordcloud.png
"/><img style="width:25.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_upregulated_genes/enriched_gene_sets_network_cluster_genes.png
"/><h4>Most Enriched Amongst Downregulated Genes (STRING_11.5)
</h4><img style="width:25.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_downregulated_genes/enriched_gene_sets_network_plot.png
"/><img style="width:25.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_downregulated_genes/enriched_gene_sets_network_plot_unlabelled.png
"/><img style="width:25.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_downregulated_genes/enriched_gene_sets_network_cluster_wordcloud.png
"/><img style="width:25.0%;" src="plots/over_representation_analysis/STRING_11.5/enriched_amongst_downregulated_genes/enriched_gene_sets_network_cluster_genes.png
"/></div>

        <!-- plot_description is replaced by bin/plot_description/... -->
        <button class="collapsible">plot description</button>
        <div class="plot_description">One of the core aims of differential expression analysis is to understand which pathways and areas of biology any differentially expressed genes represent. One method to determine this is Over Representation Analysis (ORA). In ORA the significantly different genes are compared to a database (e.g. GO terms, KEGG) of pre-defined lists of genes (gene-sets) - each of which represents a specific pathway or area of biology (e.g. Cell Cycle, TNF signalling). A hyper-geometric test is then used to determine whether each gene-set is enriched or not for the significantly differential genes. The most enriched gene-sets have the lowest p-values.
<br />
<br />
As individual genes can have several different functions, and several different gene-sets in a database might describe a similar biology (e.g. cell-cycle, G1 phase, replication) there is often considerable redundancy in gene-set databases and thus enrichment results. I.e. often there are many enriched gene-sets that contain very similar genes, and so are themselves very similar. This can often mean that the ten most enriched gene-sets are highly similar, and only represent a portion of the enriched biology. It is therefore highly useful to view all gene-set enrichment results as a network, where each node is a gene-set and each edge links gene-sets with highly similar genes. This often gives rise to clusters of gene-sets with similar function. When viewing these plots it is more important to consider the general biological theme of each cluster rather than the exact details of the gene-sets or genes within them. Networks are given for the enriched gene-sets for: all significantly differential genes, significantly upregulated genes and significantly downregulated genes.
</div>

        <!-- plot_legend is replaced by bin/figure_legends/... -->
        <button class="collapsible">figure legend</button>
        <div class="plot_legend"><p>Network of all significantly enriched genes sets (p.adj < 0.05, absolute log2 fold > 0.0), highlighting significance and gene set interconnectivity. Networks are given for enrichment amongst: all significantly differentially expressed genes (p.adj < 0.05, absolute log2 fold > 1.0) between koc and kosalt, significantly upregulated genes and significantly downregulated genes. Enrichment analysis was performed using Hypergeometric Gene Set Enrichment on the gene set databases STRING 11.5. For each network nodes represent gene sets and edges represent two gene sets with a Szymkiewicz-Simpson coefficient of at least 0.5. Node colour represents -log10 enrichment p-value, with red as higher significance and white as lower signficance. Node size represents the number of significantly differential genes within the node.
</p></div>

        <!-- r_code is replaced by -->
        <button class="collapsible" id="r_code">r code</button>
        <div class="r_code"><span class="r_comment">##---- network libraries ----####</span><br><br>library(igraph)<br>library(network)<br>library(GGally)<br>library(ggwordcloud)<br>library(ggfittext)<br><br><span class="r_comment">##---- sets the working directory. If you move the SL2 folder please update this path ----####</span><br><br>setwd("/home/john/Downloads/mahesh_run2/2x75/SL2/all_genes/de_workflows/KOC_vs_KOSALT")<br><br><span class="r_comment">##---- de hypergeometric gene set enrichment files ----####</span><br><br>STRING_11.5_all_gene_sets = read.table(file="data/statistical_analysis/over_representation_analysis/STRING_11.5/all_significant_genes/all_gene_sets_results.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br>STRING_11.5_all_signficant_genes_enriched_gene_sets = read.table(file="data/statistical_analysis/over_representation_analysis/STRING_11.5/all_significant_genes/enriched_gene_sets_results.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br>STRING_11.5_all_signficant_genes_underenriched_gene_sets = read.table(file="data/statistical_analysis/over_representation_analysis/STRING_11.5/all_significant_genes/underenriched_gene_sets_results.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br>STRING_11.5_upregulated_gene_sets = read.table(file="data/statistical_analysis/over_representation_analysis/STRING_11.5/upregulated_genes/all_gene_sets_results.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br>STRING_11.5_upregulated_enriched_gene_sets = read.table(file="data/statistical_analysis/over_representation_analysis/STRING_11.5/upregulated_genes/enriched_gene_sets_results.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br>STRING_11.5_upregulated_underenriched_gene_sets = read.table(file="data/statistical_analysis/over_representation_analysis/STRING_11.5/upregulated_genes/underenriched_gene_sets_results.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br>STRING_11.5_downregulated_gene_sets = read.table(file="data/statistical_analysis/over_representation_analysis/STRING_11.5/downregulated_genes/all_gene_sets_results.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br>STRING_11.5_downregulated_enriched_gene_sets = read.table(file="data/statistical_analysis/over_representation_analysis/STRING_11.5/downregulated_genes/enriched_gene_sets_results.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br>STRING_11.5_downregulated_underenriched_gene_sets = read.table(file="data/statistical_analysis/over_representation_analysis/STRING_11.5/downregulated_genes/underenriched_gene_sets_results.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br><br><span class="r_comment">##---- de hypergeometric gene set network files ----####</span><br><br>STRING_11.5_all_significant_gene_sets_nodes = read.table(file="data/statistical_analysis/over_representation_analysis/STRING_11.5/all_significant_genes/network_data/enriched_gene_sets_nodes.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br>STRING_11.5_all_significant_gene_sets_edges = read.table(file="data/statistical_analysis/over_representation_analysis/STRING_11.5/all_significant_genes/network_data/enriched_gene_sets_edges.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br>STRING_11.5_upregulated_gene_sets_nodes = read.table(file="data/statistical_analysis/over_representation_analysis/STRING_11.5/upregulated_genes/network_data/enriched_gene_sets_nodes.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br>STRING_11.5_upregulated_gene_sets_edges = read.table(file="data/statistical_analysis/over_representation_analysis/STRING_11.5/upregulated_genes/network_data/enriched_gene_sets_edges.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br>STRING_11.5_downregulated_gene_sets_nodes = read.table(file="data/statistical_analysis/over_representation_analysis/STRING_11.5/downregulated_genes/network_data/enriched_gene_sets_nodes.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br>STRING_11.5_downregulated_gene_sets_edges = read.table(file="data/statistical_analysis/over_representation_analysis/STRING_11.5/downregulated_genes/network_data/enriched_gene_sets_edges.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br><br><span class="r_comment">##---- SL2 Theme ----####</span><br><br>theme_SL2 <- function () { <br> &emsp; &emsp;theme_bw() %+replace% <br> &emsp; &emsp; &emsp; &emsp;theme(<br>	 &emsp; &emsp;panel.grid = element_blank(),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;panel.background = element_blank(),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;panel.border = element_rect(colour = "black", fill=NA, linewidth=1),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.background = element_blank(), <br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.background = element_rect(fill="transparent", colour=NA),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.key = element_rect(fill="transparent", colour=NA),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.title = element_text(size=12, margin = margin(b = 5),hjust=0,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;title = element_text(size = 12, margin = margin(b = 5),hjust=0,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.text.y = element_text(size = 10, margin = margin(r = 5),hjust=1,vjust=0.5, face="bold",colour="black"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.text.x = element_text(size = 10, margin = margin(t = 5),hjust=0.5,vjust=1, face="bold",colour="black"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.title.y = element_text(size = 11, margin = margin(r = 10),angle = 90,hjust=0.5,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.title.x = element_text(size = 11, margin = margin(t = 10),hjust=0.5,vjust=1, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.text=element_text(size=11, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.title=element_blank(), <br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.key.size=unit(2.5,"line"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.margin=unit(c(0.4,0.4,0.4,0.4), "cm")<br> &emsp; &emsp; &emsp; &emsp;)<br>}<br><br><span class="r_comment">##----- Default GGplot Colours Function -----####</span><br><br>gg_color_hue <- function(n)<br>{<br> &emsp;# default colour blind palettes<br> &emsp;cblind_palette = c("#88CCEE", "#CC6677", "#DDCC77", "#117733", "#332288", "#AA4499", "#44AA99", "#999933", "#882255", "#661100", "#6699CC", "#888888")<br><br> &emsp;# append ggplot colours if groups goes above colour blind palette length<br> &emsp;if (n <= length(cblind_palette))<br> &emsp;{<br> &emsp; &emsp;return(cblind_palette[1:n])<br> &emsp;}<br> &emsp;else<br> &emsp;{<br> &emsp; &emsp;nn = n - length(cblind_palette)<br> &emsp; &emsp;hues = seq(15, 375, length = nn + 1)<br> &emsp; &emsp;return(append(cblind_palette,hcl(h = hues, l = 65, c = 100)[1:nn]))<br> &emsp;}<br>}<br><br><span class="r_comment">##---- Default Two Tone Network Node Colours &emsp;----####</span><br><br>default_two_tone_network_node_colours = c("grey90","red")<br><br><span class="r_comment">##---- Save Plot Function -----####</span><br><br>save_plot <- function(ggp,plot_height,plot_width,plot_path)<br>{<br> &emsp;png(plot_path, height=plot_height, width=plot_width, pointsize=5)<br> &emsp;print(ggp)<br> &emsp;dev.off()<br><br> &emsp;svg(gsub(".png", ".svg", plot_path), height=plot_height/90, width=plot_width/90)<br> &emsp;print(ggp)<br> &emsp;dev.off()<br><br> &emsp;# clears all devices - as a safety measure<br> &emsp;while (dev.cur()>1) dev.off()<br>}<br><br><span class="r_comment">##---- Cluster Network Function ----####</span><br><br>cluster_network <- function(edges, nodes)<br>{<br> &emsp;# sets the row names for nodes<br> &emsp;row.names(nodes) = nodes$node<br><br> &emsp;# gets the network clusters<br> &emsp;g = graph_from_data_frame(edges, directed=TRUE, vertices=nodes)<br> &emsp;fc = fastgreedy.community(as.undirected(g))<br> &emsp;clusters = data.frame(cbind(fc$names, fc$membership))<br> &emsp;names(clusters) = c("name", "cluster")<br> &emsp;clusters$cluster = as.numeric(clusters$cluster)<br> &emsp;nodes = merge(nodes, clusters, by = 1)<br><br> &emsp;return(nodes)<br>}<br><br><span class="r_comment">##---- ORA Gene Set Network Plot Function ----####</span><br><br>make_ora_network_plot <- function(edges, nodes, node_colours, node_outer_max_size, node_inner_max_size, node_label_size, label_nodes)<br>{<br> &emsp;set.seed(1)<br><br> &emsp;if (nrow(edges) > 0) {<br><br> &emsp; &emsp;# cluster network<br> &emsp; &emsp;nodes = cluster_network(edges, nodes)<br><br> &emsp; &emsp;# build network<br> &emsp; &emsp;net = 0<br> &emsp; &emsp;net = network(edges[,c(1,2)], directed = FALSE)<br><br><br> &emsp; &emsp;# gets the colours<br> &emsp; &emsp;colours = gg_color_hue(max(as.numeric(unique(nodes$cluster))))<br><br> &emsp; &emsp;# adds colours to nodes<br> &emsp; &emsp;nodes$node_colour = "white"<br> &emsp; &emsp; &emsp;for (row_index in 1:nrow(nodes))<br> &emsp; &emsp; &emsp;{<br> &emsp; &emsp; &emsp; &emsp;nodes[row_index,"node_colour"] = colours[nodes[row_index,"cluster"]]<br> &emsp; &emsp; &emsp;}<br><br> &emsp; &emsp;x = data.frame(node = network.vertex.names(net))<br> &emsp; &emsp;x = merge(x, nodes, by = "node", sort = FALSE)$node_colour<br> &emsp; &emsp;net %v% "node_colour" = as.character(x)<br><br><br> &emsp; &emsp;# make the plot<br> &emsp; &emsp;if (label_nodes)<br> &emsp; &emsp;{<br> &emsp; &emsp; &emsp;ggp = ggnet2(net, color = 'node_colour', size = 0) +<br> &emsp; &emsp; &emsp; &emsp;xlab("") +<br> &emsp; &emsp; &emsp; &emsp;ylab("") +<br> &emsp; &emsp; &emsp; &emsp;theme_SL2() +<br> &emsp; &emsp; &emsp; &emsp;theme(legend.position="none") +<br> &emsp; &emsp; &emsp; &emsp;geom_point(aes(color = "black"), size=node_outer_max_size, alpha = 1) +<br> &emsp; &emsp; &emsp; &emsp;geom_point(aes(color = color), size=node_inner_max_size, alpha = 1) +<br> &emsp; &emsp; &emsp; &emsp;geom_label(aes(label = network.vertex.names(net), color = color) , size=node_label_size)<br> &emsp; &emsp;}<br> &emsp; &emsp;else<br> &emsp; &emsp;{<br> &emsp; &emsp; &emsp;ggp = ggnet2(net, color = 'node_colour', size = 0) +<br> &emsp; &emsp; &emsp; &emsp;xlab("") +<br> &emsp; &emsp; &emsp; &emsp;ylab("") +<br> &emsp; &emsp; &emsp; &emsp;theme_SL2() +<br> &emsp; &emsp; &emsp; &emsp;theme(legend.position="none") +<br> &emsp; &emsp; &emsp; &emsp;geom_point(aes(color = "black"), size=node_outer_max_size, alpha = 1) +<br> &emsp; &emsp; &emsp; &emsp;geom_point(aes(color = color), size=node_inner_max_size, alpha = 1)<br> &emsp; &emsp;}<br> &emsp; &emsp;return(ggp)<br> &emsp;}<br><br> &emsp;return(ggplot(data.frame()) + theme_SL2() + geom_blank() + ggtitle("There were too few significant nodes to plot this."))<br>}<br><br><span class="r_comment">##---- ORA Gene Set Network Word Cloud Function ----####</span><br><br>make_ora_network_word_cloud <- function(edges, nodes, sig_gene_sets, topn, table_path, plot_path, plot_height, plot_width)<br>{<br><br> &emsp;set.seed(1)<br><br> &emsp;if (nrow(edges) > 0) {<br><br> &emsp; &emsp;# cluster network<br> &emsp; &emsp;nodes = cluster_network(edges, nodes)<br><br> &emsp; &emsp;# sets the gene-sets<br> &emsp; &emsp;row.names(sig_gene_sets) = sig_gene_sets$gene_set<br><br> &emsp; &emsp;# table to store lists of genes and gene-sets per cluster<br> &emsp; &emsp;cluster_summary = data.frame(matrix("",nrow=max(nodes$cluster), ncol=3))<br> &emsp; &emsp;names(cluster_summary) = c("cluster", "names", "genes")<br><br> &emsp; &emsp;# creates combined lists of genes and gene-sets per cluster<br> &emsp; &emsp;for (row_index in 1:nrow(nodes))<br> &emsp; &emsp;{<br> &emsp; &emsp; node_data = &emsp;nodes[row_index,]<br> &emsp; &emsp; gene_set_data = sig_gene_sets[node_data$node,]<br> &emsp; &emsp; gene_set_name = gene_set_data$gene_set<br> &emsp; &emsp; gene_set_genes = gene_set_data$overlapping_gene_names<br><br> &emsp; &emsp; cluster_number = node_data$cluster<br> &emsp; &emsp; cluster_summary[cluster_number,"cluster"] = cluster_number<br> &emsp; &emsp; cluster_summary[cluster_number,"names"] = paste(cluster_summary[cluster_number,"names"], gene_set_name, sep=",")<br> &emsp; &emsp; cluster_summary[cluster_number,"genes"] = paste(cluster_summary[cluster_number,"genes"], gene_set_genes, sep=",")<br> &emsp; &emsp;}<br><br> &emsp; &emsp;# trim the unnecessary first comma<br> &emsp; &emsp;for (row_index in 1:nrow(cluster_summary))<br> &emsp; &emsp;{<br> &emsp; &emsp; &emsp;cluster_summary[row_index,"names"] = sub(',', '', cluster_summary[row_index,"names"])<br> &emsp; &emsp; &emsp;cluster_summary[row_index,"genes"] = sub(',', '', cluster_summary[row_index,"genes"])<br> &emsp; &emsp;}<br><br> &emsp; &emsp;# Collapse the duplicates in the gene lists<br> &emsp; &emsp;for (row_index in 1:nrow(cluster_summary))<br> &emsp; &emsp;{<br> &emsp; &emsp; &emsp;genes = unique(unlist(lapply(cluster_summary[row_index,"genes"], function(x) unlist(strsplit(as.character(x),",")))))<br> &emsp; &emsp; &emsp;genes = sort(genes[2:length(genes)])<br> &emsp; &emsp; &emsp;cluster_summary[row_index,"genes"] = paste(genes,collapse=",")<br> &emsp; &emsp;}<br><br> &emsp; &emsp;# saves the table<br> &emsp; &emsp;write.table(cluster_summary, file=paste0(table_path,"enriched_gene_sets_network_clusters.csv") ,sep="\t", quote = FALSE, row.names = FALSE)<br><br> &emsp; &emsp;# filters to include only the top n clusters<br> &emsp; &emsp;cluster_summary_subset = subset(cluster_summary, as.numeric(cluster_summary$cluster) <= topn)<br><br><br> &emsp; &emsp;# removes commas from gene set names<br> &emsp; &emsp;cluster_summary_subset$names = gsub(",", "_",cluster_summary_subset$names)<br><br> &emsp; &emsp;# collapses the gene-set names, counting the times each word is used.<br> &emsp; &emsp;gene_sets_wc = data.frame()<br> &emsp; &emsp;for (row_index in 1:nrow(cluster_summary_subset))<br> &emsp; &emsp;{<br> &emsp; &emsp; &emsp;gene_sets = unlist(lapply(cluster_summary_subset[row_index,"names"], function(x) unlist(strsplit(as.character(x), "[, _]+"))))<br> &emsp; &emsp; &emsp;gene_sets = sort(gene_sets)<br> &emsp; &emsp; &emsp;gene_sets = table(gene_sets)<br> &emsp; &emsp; &emsp;gene_sets = data.frame(gene_sets)<br> &emsp; &emsp; &emsp;gene_sets = gene_sets[order(gene_sets$Freq, decreasing = TRUE),]<br> &emsp; &emsp; &emsp;gene_sets$cluster = row_index<br> &emsp; &emsp; &emsp;gene_sets_wc = rbind(gene_sets_wc, gene_sets)<br> &emsp; &emsp;}<br><br> &emsp; &emsp;# creates the gene-set name plots<br> &emsp; &emsp;ggp_wc = ggplot(gene_sets_wc, aes(label = gene_sets, size = Freq, colour=factor(gene_sets_wc$cluster))) +<br> &emsp; &emsp; &emsp;geom_text_wordcloud() +<br> &emsp; &emsp; &emsp;scale_size_area(max_size = 24)+<br> &emsp; &emsp; &emsp;facet_wrap(~cluster) +<br> &emsp; &emsp; &emsp;scale_colour_manual(values=gg_color_hue(max(as.numeric(nodes$cluster)))) +<br> &emsp; &emsp; &emsp;theme_minimal()<br> &emsp; &emsp;save_plot(ggp_wc,plot_height,plot_width,paste0(plot_path,"_network_cluster_wordcloud.png"))<br><br><br> &emsp; &emsp;# removes commas from gene names<br> &emsp; &emsp;cluster_summary_subset$genes = gsub(",", ", ",cluster_summary_subset$genes)<br><br> &emsp; &emsp;# plot the genes<br> &emsp; &emsp;cluster_summary_subset$cluster = factor(cluster_summary_subset$cluster, levels=rev(cluster_summary_subset$cluster))<br> &emsp; &emsp;ggp_genes = ggplot(cluster_summary_subset, aes(x=1, y=cluster, label=genes)) +<br> &emsp; &emsp; &emsp;geom_tile(fill = "white", colour = "black") +<br> &emsp; &emsp; &emsp;geom_fit_text(reflow = TRUE) + #,grow=TRUE<br> &emsp; &emsp; &emsp;theme_minimal() +<br> &emsp; &emsp; &emsp;theme(axis.title.x = element_blank(),axis.text.x = element_blank()) +<br> &emsp; &emsp; &emsp;labs(x="")<br> &emsp; &emsp;save_plot(ggp_genes,plot_height,plot_width,paste0(plot_path,"_network_cluster_genes.png"))<br><br> &emsp;}<br> &emsp;else<br> &emsp;{<br> &emsp; &emsp;ggp = ggplot(data.frame()) + theme_SL2() + geom_blank() + ggtitle("There were too few significant nodes to plot this.")<br> &emsp; &emsp;save_plot(ggp,plot_height,plot_width,paste0(plot_path,"_network_cluster_wordcloud.png"))<br> &emsp; &emsp;save_plot(ggp,plot_height,plot_width,paste0(plot_path,"_network_cluster_genes.png"))<br> &emsp;}<br>}<br><br><span class="r_comment">##---- ORA Enriched Gene Sets (Network Plots) ----####</span><br><br>node_colours = default_two_tone_network_node_colours<br>plot_height = 2000<br>plot_width = 2000<br>node_outer_max_size = 10<br>node_inner_max_size = 8<br>node_label_size = 3<br><br>ggp = make_ora_network_plot(STRING_11.5_all_significant_gene_sets_edges,STRING_11.5_all_significant_gene_sets_nodes,node_colours,node_outer_max_size,node_inner_max_size,node_label_size,TRUE)<br>save_plot(ggp,plot_height,plot_width,"plots/over_representation_analysis/STRING_11.5/enriched_amongst_all_significant_genes/enriched_gene_sets_network_plot.png")<br><br>ggp = make_ora_network_plot(STRING_11.5_upregulated_gene_sets_edges,STRING_11.5_upregulated_gene_sets_nodes,node_colours,node_outer_max_size,node_inner_max_size,node_label_size, TRUE)<br>save_plot(ggp,plot_height,plot_width,"plots/over_representation_analysis/STRING_11.5/enriched_amongst_upregulated_genes/enriched_gene_sets_network_plot.png")<br><br>ggp = make_ora_network_plot(STRING_11.5_downregulated_gene_sets_edges,STRING_11.5_downregulated_gene_sets_nodes,node_colours,node_outer_max_size,node_inner_max_size,node_label_size, TRUE)<br>save_plot(ggp,plot_height,plot_width,"plots/over_representation_analysis/STRING_11.5/enriched_amongst_downregulated_genes/enriched_gene_sets_network_plot.png")<br><br>ggp = make_ora_network_plot(STRING_11.5_all_significant_gene_sets_edges,STRING_11.5_all_significant_gene_sets_nodes,node_colours,node_outer_max_size,node_inner_max_size,node_label_size, FALSE)<br>save_plot(ggp,plot_height,plot_width,"plots/over_representation_analysis/STRING_11.5/enriched_amongst_all_significant_genes/enriched_gene_sets_network_plot_unlabelled.png")<br><br>ggp = make_ora_network_plot(STRING_11.5_upregulated_gene_sets_edges,STRING_11.5_upregulated_gene_sets_nodes,node_colours,node_outer_max_size,node_inner_max_size,node_label_size, FALSE)<br>save_plot(ggp,plot_height,plot_width,"plots/over_representation_analysis/STRING_11.5/enriched_amongst_upregulated_genes/enriched_gene_sets_network_plot_unlabelled.png")<br><br>ggp = make_ora_network_plot(STRING_11.5_downregulated_gene_sets_edges,STRING_11.5_downregulated_gene_sets_nodes,node_colours,node_outer_max_size,node_inner_max_size,node_label_size, FALSE)<br>save_plot(ggp,plot_height,plot_width,"plots/over_representation_analysis/STRING_11.5/enriched_amongst_downregulated_genes/enriched_gene_sets_network_plot_unlabelled.png")<br><br><br><span class="r_comment">##---- ORA Enriched Gene Sets (Network Word Clouds) ----####</span><br><br>plot_height = 2000<br>plot_width = 2000<br>topn = 16<br><br>table_path = "data/statistical_analysis/over_representation_analysis/STRING_11.5/all_significant_genes/network_data/"<br>plot_path = "plots/over_representation_analysis/STRING_11.5/enriched_amongst_all_significant_genes/enriched_gene_sets"<br>make_ora_network_word_cloud(STRING_11.5_all_significant_gene_sets_edges, STRING_11.5_all_significant_gene_sets_nodes, STRING_11.5_all_signficant_genes_enriched_gene_sets, topn, table_path, plot_path, plot_height, plot_width)<br><br>table_path = "data/statistical_analysis/over_representation_analysis/STRING_11.5/upregulated_genes/network_data/"<br>plot_path = "plots/over_representation_analysis/STRING_11.5/enriched_amongst_upregulated_genes/enriched_gene_sets"<br>make_ora_network_word_cloud(STRING_11.5_upregulated_gene_sets_edges, STRING_11.5_upregulated_gene_sets_nodes, STRING_11.5_upregulated_enriched_gene_sets, topn, table_path, plot_path, plot_height, plot_width)<br><br>table_path = "data/statistical_analysis/over_representation_analysis/STRING_11.5/downregulated_genes/network_data/"<br>plot_path = "plots/over_representation_analysis/STRING_11.5/enriched_amongst_downregulated_genes/enriched_gene_sets"<br>make_ora_network_word_cloud(STRING_11.5_downregulated_gene_sets_edges, STRING_11.5_downregulated_gene_sets_nodes, STRING_11.5_downregulated_enriched_gene_sets, topn, table_path, plot_path, plot_height, plot_width)<br><br><br></div>
    </div>
</div>

<div id="Underenrichment_Barcharts" class="plot_sections">
    <h2 class="plot_title">Underenrichment Barcharts
</h2>
    <div class="plot_wrapper" style="display: block;">
        <div class="plots"><h4>Most Underenriched Amongst All Significant Genes (STRING_11.5)
</h4><img style="width:50.0%;" src="plots/over_representation_analysis/STRING_11.5/underenriched_amongst_all_significant_genes/underenriched_gene_sets_barchart.png
"/><h4>Most Underenriched Amongst Upregulated Genes (STRING_11.5)
</h4><img style="width:50.0%;" src="plots/over_representation_analysis/STRING_11.5/underenriched_amongst_upregulated_genes/underenriched_gene_sets_barchart.png
"/><h4>Most Underenriched Amongst Downregulated Genes (STRING_11.5)
</h4><img style="width:50.0%;" src="plots/over_representation_analysis/STRING_11.5/underenriched_amongst_downregulated_genes/underenriched_gene_sets_barchart.png
"/></div>

        <!-- plot_description is replaced by bin/plot_description/... -->
        <button class="collapsible">plot description</button>
        <div class="plot_description">One of the core aims of differential expression analysis is to understand which pathways and areas of biology any differentially expressed genes represent. One method to determine this is Over Representation Analysis (ORA). In ORA the significantly different genes are compared to a database (e.g. GO terms, KEGG) of pre-defined lists of genes (gene-sets) - each of which represents a specific pathway or area of biology (e.g. Cell Cycle, TNF signalling). A hyper-geometric test is then used to determine whether each gene-set is enriched or not for the significantly differential genes. The most enriched gene-sets have the lowest p-values.
<br />
<br />
As well as understanding what biology is different between to sample groups, it can sometimes be useful to understand what biology is not different. More specifically using HGSEA it is possible to identify gene sets that are underenriched. I.e. gene-sets that are less differential than they should be given the number of significantly differential genes. This can give an indication as to the types of biology that the cell / tissue is actively maintaining. Here we provide bar charts showing the most underenriched gene-sets for: all significantly differential genes, significantly upregulated genes and significantly downregulated genes.
</div>

        <!-- plot_legend is replaced by bin/figure_legends/... -->
        <button class="collapsible">figure legend</button>
        <div class="plot_legend"><p>Summary bar chart of the ten most underenriched genes sets for: all significantly differentially expressed genes (p.adj < 0.05, absolute log2 fold > 1.0) between koc and kosalt, significantly upregulated genes and significantly downregulated genes. Upregulated genes are higher expression in koc than in kosalt. For each plot the -log10 under enrichment p-value is given on the x-axis and the gene set name on the y-axis. Data labels represent the number of significantly differential expressed genes within each gene set. Enrichment analysis was performed using Hypergeometric Gene Set Enrichment on the gene set databases STRING 11.5. Significantly underenriched gene sets (p.adj < 0.05, absolute log2 fold > 0.0) are coloured red and non-significant black.
</p></div>

        <!-- r_code is replaced by -->
        <button class="collapsible" id="r_code">r code</button>
        <div class="r_code"><span class="r_comment">##---- general libraries ----####</span><br><br>library(ggplot2)<br>library(ggrepel)<br>library(reshape)<br>library(ggridges)<br><br><span class="r_comment">##---- DE sample and group names ----####</span><br><br>samples = c("KOSALT12_S4_001","KOSALT15_S10_001","KOSALT19_S8_001","KOC12_S3_001","KOC15_S9_001","KOC19_S6_001")<br>sample_groups = c("KOSALT","KOC")<br>sample_groups = factor(sample_groups, levels = sample_groups)<br>sample_groupings = c("KOSALT","KOSALT","KOSALT","KOC","KOC","KOC")<br>sample_groupings = factor(sample_groupings, levels = sample_groups)<br>samples_by_sample_group = list(c("KOSALT12_S4_001","KOSALT15_S10_001","KOSALT19_S8_001"),c("KOC12_S3_001","KOC15_S9_001","KOC19_S6_001"))<br>comparisons = c("KOC vs KOSALT")<br><br><span class="r_comment">##---- sets the working directory. If you move the SL2 folder please update this path ----####</span><br><br>setwd("/home/john/Downloads/mahesh_run2/2x75/SL2/all_genes/de_workflows/KOC_vs_KOSALT")<br><br><span class="r_comment">##---- de hypergeometric gene set enrichment files ----####</span><br><br>STRING_11.5_all_gene_sets = read.table(file="data/statistical_analysis/over_representation_analysis/STRING_11.5/all_significant_genes/all_gene_sets_results.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br>STRING_11.5_all_signficant_genes_enriched_gene_sets = read.table(file="data/statistical_analysis/over_representation_analysis/STRING_11.5/all_significant_genes/enriched_gene_sets_results.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br>STRING_11.5_all_signficant_genes_underenriched_gene_sets = read.table(file="data/statistical_analysis/over_representation_analysis/STRING_11.5/all_significant_genes/underenriched_gene_sets_results.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br>STRING_11.5_upregulated_gene_sets = read.table(file="data/statistical_analysis/over_representation_analysis/STRING_11.5/upregulated_genes/all_gene_sets_results.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br>STRING_11.5_upregulated_enriched_gene_sets = read.table(file="data/statistical_analysis/over_representation_analysis/STRING_11.5/upregulated_genes/enriched_gene_sets_results.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br>STRING_11.5_upregulated_underenriched_gene_sets = read.table(file="data/statistical_analysis/over_representation_analysis/STRING_11.5/upregulated_genes/underenriched_gene_sets_results.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br>STRING_11.5_downregulated_gene_sets = read.table(file="data/statistical_analysis/over_representation_analysis/STRING_11.5/downregulated_genes/all_gene_sets_results.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br>STRING_11.5_downregulated_enriched_gene_sets = read.table(file="data/statistical_analysis/over_representation_analysis/STRING_11.5/downregulated_genes/enriched_gene_sets_results.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br>STRING_11.5_downregulated_underenriched_gene_sets = read.table(file="data/statistical_analysis/over_representation_analysis/STRING_11.5/downregulated_genes/underenriched_gene_sets_results.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br><br><span class="r_comment">##---- SL2 Theme ----####</span><br><br>theme_SL2 <- function () { <br> &emsp; &emsp;theme_bw() %+replace% <br> &emsp; &emsp; &emsp; &emsp;theme(<br>	 &emsp; &emsp;panel.grid = element_blank(),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;panel.background = element_blank(),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;panel.border = element_rect(colour = "black", fill=NA, linewidth=1),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.background = element_blank(), <br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.background = element_rect(fill="transparent", colour=NA),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.key = element_rect(fill="transparent", colour=NA),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.title = element_text(size=12, margin = margin(b = 5),hjust=0,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;title = element_text(size = 12, margin = margin(b = 5),hjust=0,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.text.y = element_text(size = 10, margin = margin(r = 5),hjust=1,vjust=0.5, face="bold",colour="black"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.text.x = element_text(size = 10, margin = margin(t = 5),hjust=0.5,vjust=1, face="bold",colour="black"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.title.y = element_text(size = 11, margin = margin(r = 10),angle = 90,hjust=0.5,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.title.x = element_text(size = 11, margin = margin(t = 10),hjust=0.5,vjust=1, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.text=element_text(size=11, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.title=element_blank(), <br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.key.size=unit(2.5,"line"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.margin=unit(c(0.4,0.4,0.4,0.4), "cm")<br> &emsp; &emsp; &emsp; &emsp;)<br>}<br><br><span class="r_comment">##---- Default Two Way Significance Colours ----####</span><br><br>default_2way_sig_colour = "coral3"<br>default_2way_non_sig_colour = "grey50"<br><br><span class="r_comment">##---- Save Plot Function -----####</span><br><br>save_plot <- function(ggp,plot_height,plot_width,plot_path)<br>{<br> &emsp;png(plot_path, height=plot_height, width=plot_width, pointsize=5)<br> &emsp;print(ggp)<br> &emsp;dev.off()<br><br> &emsp;svg(gsub(".png", ".svg", plot_path), height=plot_height/90, width=plot_width/90)<br> &emsp;print(ggp)<br> &emsp;dev.off()<br><br> &emsp;# clears all devices - as a safety measure<br> &emsp;while (dev.cur()>1) dev.off()<br>}<br><br><span class="r_comment">##---- Top 10 Gene Sets by ORA P-value Function ----####</span><br><br>get_top10_oras_by_p_value <- function(gene_sets,enriched_gene_sets,type) {<br><br> &emsp;if(type=="enrichment")<br> &emsp;{<br> &emsp; &emsp;top_10_gene_sets = head(gene_sets[order(gene_sets$enrichment_p_value,decreasing=FALSE),],10)<br> &emsp; &emsp;top_10_gene_sets = top_10_gene_sets[,c("gene_set","enrichment_p_value","overlapping_genes","log2_fold_enrichment","overlapping_gene_names")]<br> &emsp;}<br> &emsp;if(type=="underenrichment")<br> &emsp;{<br> &emsp; &emsp;top_10_gene_sets = head(gene_sets[order(gene_sets$underenrichment_p_value,decreasing=FALSE),],10)<br> &emsp; &emsp;top_10_gene_sets = top_10_gene_sets[,c("gene_set","underenrichment_p_value","overlapping_genes","log2_fold_enrichment","overlapping_gene_names")]<br> &emsp;}<br> &emsp;top_10_gene_sets$significant = top_10_gene_sets$gene_set %in% enriched_gene_sets$gene_set<br> &emsp;top_10_gene_sets$significant = as.character(top_10_gene_sets$significant)<br> &emsp;colnames(top_10_gene_sets) = c("gene_set","p","overlapping_genes","log2_fold_enrichment","overlapping_gene_names","significant")<br> &emsp;return(top_10_gene_sets)<br>}<br><br><span class="r_comment">##---- ORA Gene Sets Bar Chart Function ----####</span><br><br>make_oras_bar_chart <- function(top_10_gene_sets, x_axis_label, y_axis_label, non_significant_colour, significant_colour, bar_outline_size, bar_transparency, legend_position, significant_name, non_significant_name, data_label_size)<br>{ <br> &emsp;top_10_gene_sets = top_10_gene_sets[seq(dim(top_10_gene_sets)[1],1),]<br> &emsp;ggp = ggplot(data=top_10_gene_sets , aes(x=gene_set, y=-log10(p), fill=significant,group=significant)) +<br> &emsp; &emsp; &emsp; &emsp;geom_bar(colour="black",stat="identity", position = "dodge", alpha = bar_transparency) + <br> &emsp; &emsp; &emsp; &emsp;coord_flip() + <br> &emsp; &emsp; &emsp; &emsp;scale_x_discrete(limits = top_10_gene_sets$gene_set) + <br> &emsp; &emsp; &emsp; &emsp;ylab(x_axis_label) &emsp;+ <br> &emsp; &emsp; &emsp; &emsp;xlab(y_axis_label) + <br> &emsp; &emsp; &emsp; &emsp;scale_fill_manual(values=c(non_significant_colour,significant_colour),breaks=c("TRUE","FALSE"),labels=c(significant_name, non_significant_name)) + <br> &emsp; &emsp; &emsp; &emsp;theme_SL2() + <br> &emsp; &emsp; &emsp; &emsp;theme(axis.text.y = element_text(hjust=1),legend.position=legend_position, legend.spacing.x = unit(0.15, 'cm')) + <br> &emsp; &emsp; &emsp; &emsp;scale_y_continuous(expand=c(0,0), limits=c(0,max(-log10(top_10_gene_sets$p)*1.25))) + <br> &emsp; &emsp; &emsp; &emsp;geom_text(aes(label=overlapping_genes), position=position_dodge(width=0.9), size=data_label_size, show.legend = FALSE, hjust=-0.5)<br><br> &emsp;return(ggp)<br>}<br><br><span class="r_comment">##---- ORA Underenriched Gene Sets (Bar Charts) ----####</span><br><br>plot_height = 350<br>plot_width = 1000<br>x_axis_label = "-log10 p-value"<br>y_axis_label = ""<br>non_significant_colour = default_2way_sig_colour<br>significant_colour = default_2way_non_sig_colour<br>bar_outline_size = 1<br>bar_transparency = 0.75<br>legend_position = "bottom"<br>significant_name = "significant"<br>non_significant_name = "non-significant"<br>data_label_size = 5<br><br>top_10_gene_sets = get_top10_oras_by_p_value(STRING_11.5_all_gene_sets,STRING_11.5_all_signficant_genes_underenriched_gene_sets,"underenrichment")<br>ggp = make_oras_bar_chart(top_10_gene_sets,x_axis_label,y_axis_label,non_significant_colour,significant_colour,bar_outline_size,bar_transparency,legend_position,significant_name,non_significant_name,data_label_size)<br>save_plot(ggp,plot_height,plot_width,"plots/over_representation_analysis/STRING_11.5/underenriched_amongst_all_significant_genes/underenriched_gene_sets_barchart.png")<br><br>top_10_gene_sets = get_top10_oras_by_p_value(STRING_11.5_downregulated_gene_sets,STRING_11.5_downregulated_underenriched_gene_sets,"underenrichment")<br>ggp = make_oras_bar_chart(top_10_gene_sets,x_axis_label,y_axis_label,non_significant_colour,significant_colour,bar_outline_size,bar_transparency,legend_position,significant_name,non_significant_name,data_label_size)<br>save_plot(ggp,plot_height,plot_width,"plots/over_representation_analysis/STRING_11.5/underenriched_amongst_downregulated_genes/underenriched_gene_sets_barchart.png")<br><br>top_10_gene_sets = get_top10_oras_by_p_value(STRING_11.5_upregulated_gene_sets,STRING_11.5_upregulated_underenriched_gene_sets,"underenrichment")<br>ggp = make_oras_bar_chart(top_10_gene_sets,x_axis_label,y_axis_label,non_significant_colour,significant_colour,bar_outline_size,bar_transparency,legend_position,significant_name,non_significant_name,data_label_size)<br>save_plot(ggp,plot_height,plot_width,"plots/over_representation_analysis/STRING_11.5/underenriched_amongst_upregulated_genes/underenriched_gene_sets_barchart.png")<br><br><br></div>
    </div>
</div>

<div id="Spatial_Enrichment_Table" class="plot_sections">
    <h2 class="plot_title">Spatial Enrichment Table
</h2>
    <div class="plot_wrapper" style="display: block;">
        <div class="plots"><img style="width:100.0%;" src="plots/spatial_enrichment/spatial_enrichment_table.png
"/></div>

        <!-- plot_description is replaced by bin/plot_description/... -->
        <button class="collapsible">plot description</button>
        <div class="plot_description">The spatial enrichment table gives statistics on the genomic distribution of expressed and differentially expressed genes. It provides three main measures of spatial enrichment: 1) Whether a chromosome has more expressed genes than expected by random chance - taking into account the number of expressed genes globally and the number of genes on the chromosome. 2) Whether a chromosome has more differentially expressed genes than expected by random chance - taking into account the number of differentially expressed genes globally and the number of genes on the chromosome. 3) Whether a chromosome has any significant bias towards up or downregulated genes - taking into account the proportion of upregulated and downregulated genes globally and the number of genes on the chromosome. For more details please see the methods section. Any significant bias may be indicative of spatial or epigenetic regulation at these loci. 
</div>

        <!-- plot_legend is replaced by bin/figure_legends/... -->
        <button class="collapsible">figure legend</button>
        <div class="plot_legend"><p>Table showing spatial enrichment statistics for the differentially expressed genes between koc and kosalt. Three measures of spatial enrichment are provided for each chromosome: enrichment of expressed genes (columns 8-9), enrichment of differentially expressed genes (columns 10-11) and bias towards upregulation or downregulation (columns 12-13). The per chromosome gene count statistics from which all statistical analysis is based are also provided (columns 2-7). Upregulated genes are higher in koc than in kosalt.
</p></div>

        <!-- r_code is replaced by -->
        <button class="collapsible" id="r_code">r code</button>
        <div class="r_code"><span class="r_comment">##---- general libraries ----####</span><br><br>library(ggplot2)<br>library(ggrepel)<br>library(reshape)<br>library(ggridges)<br><br><span class="r_comment">##---- table libraries ----####</span><br><br>library(grid)<br>library(gridExtra)<br>library(gtable)<br>library(ggplotify)<br><br><span class="r_comment">##---- sets the working directory. If you move the SL2 folder please update this path ----####</span><br><br>setwd("/home/john/Downloads/mahesh_run2/2x75/SL2/all_genes/de_workflows/KOC_vs_KOSALT")<br><br><span class="r_comment">##---- spatial analysis input files ----####</span><br><br>spatial_enrichment_summary = read.table(file="data/statistical_analysis/spatial_enrichment/spatial_enrichment_summary.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br>spatial_enrichment_gene_data = read.table(file="data/statistical_analysis/spatial_enrichment/spatial_enrichment_gene_data.csv", header=TRUE, sep='\t', quote='',check.names = TRUE)<br><br><span class="r_comment">##---- SL2 Theme ----####</span><br><br>theme_SL2 <- function () { <br> &emsp; &emsp;theme_bw() %+replace% <br> &emsp; &emsp; &emsp; &emsp;theme(<br>	 &emsp; &emsp;panel.grid = element_blank(),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;panel.background = element_blank(),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;panel.border = element_rect(colour = "black", fill=NA, linewidth=1),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.background = element_blank(), <br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.background = element_rect(fill="transparent", colour=NA),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.key = element_rect(fill="transparent", colour=NA),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.title = element_text(size=12, margin = margin(b = 5),hjust=0,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;title = element_text(size = 12, margin = margin(b = 5),hjust=0,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.text.y = element_text(size = 10, margin = margin(r = 5),hjust=1,vjust=0.5, face="bold",colour="black"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.text.x = element_text(size = 10, margin = margin(t = 5),hjust=0.5,vjust=1, face="bold",colour="black"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.title.y = element_text(size = 11, margin = margin(r = 10),angle = 90,hjust=0.5,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.title.x = element_text(size = 11, margin = margin(t = 10),hjust=0.5,vjust=1, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.text=element_text(size=11, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.title=element_blank(), <br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.key.size=unit(2.5,"line"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.margin=unit(c(0.4,0.4,0.4,0.4), "cm")<br> &emsp; &emsp; &emsp; &emsp;)<br>}<br><br><span class="r_comment">##---- Save Table Function -----####</span><br><br>save_table <- function(gt,plot_height,plot_width,plot_path)<br>{<br> &emsp;png(plot_path, height=plot_height, width=plot_width, pointsize=5)<br> &emsp;grid.arrange(gt)<br> &emsp;dev.off()<br><br> &emsp;while (dev.cur()>1) dev.off()<br>}<br><br><span class="r_comment">##---- Spatial Enrichement Table Function &emsp;----####</span><br><br>make_spatial_enrichment_table <- function(spatial_enrichment_summary,header_size,text_size,border_thickness)<br>{<br> &emsp;spatial_enrichment_summary_table = spatial_enrichment_summary[,c("chromosome","total_genes","expressed_genes","de_valid_genes","significant_genes","upregulated_genes","downregulated_genes","expressed_genes_bias_log2fold","expressed_genes_bias_p","significant_genes_bias_log2fold","significant_genes_bias_p","direction_bias_swing","direction_bias_p")]<br> &emsp;colnames(spatial_enrichment_summary_table) = c("chromosome","total genes","expressed genes","de valid genes","significant genes","upregulated genes","downregulated genes","expressed genes bias (log2fold)","expressed genes bias (p)","significant genes bias (log2fold)","significant genes bias (p)","direction bias (swing)","direction bias (p)")<br> &emsp;spatial_enrichment_summary_table = spatial_enrichment_summary_table[order(as.numeric(as.character(spatial_enrichment_summary_table$chromosome))),]<br> &emsp;table_theme <- gridExtra::ttheme_minimal(core = list(fg_params=list(cex = text_size)),colhead = list(fg_params=list(cex = header_size)))<br><br> &emsp;gt = tableGrob(spatial_enrichment_summary_table, theme=table_theme, rows=NULL)<br> &emsp;gt = gtable_add_grob(gt,grobs = rectGrob(gp = gpar(fill = NA, lwd = border_thickness)),t = 2, b = nrow(gt), l = 1, r = ncol(gt))<br> &emsp;gt = gtable_add_grob(gt,grobs = rectGrob(gp = gpar(fill = NA, lwd = border_thickness)),t = 1, l = 1, r = ncol(gt))<br> &emsp;gt = gtable_add_grob(gt,grobs = rectGrob(gp = gpar(fill = NA, lwd = border_thickness)),t = 1, b = nrow(gt), l = 1, r = 1)<br> &emsp;<br> &emsp;return(gt)<br>}<br><br><span class="r_comment">##---- Spatial Enrichement (Table) ----####</span><br><br>plot_height = 700<br>plot_width = 2100<br>header_size = 1.25<br>text_size = 1.25<br>border_thickness = 2<br><br>gt = make_spatial_enrichment_table(spatial_enrichment_summary,header_size,text_size,border_thickness)<br>save_table(gt,plot_height,plot_width,"plots/spatial_enrichment/spatial_enrichment_table.png")<br><br></div>
    </div>
</div>

<div id="Spatial_Enrichment_Distribution_Plots" class="plot_sections">
    <h2 class="plot_title">Spatial Enrichment Distribution Plots
</h2>
    <div class="plot_wrapper" style="display: block;">
        <div class="plots"><img style="width:50.0%;" src="plots/spatial_enrichment/fold_distribution_chr_1_unlabelled.png
"/><img style="width:50.0%;" src="plots/spatial_enrichment/gene_distribution_chr_1.png
"/><img style="width:50.0%;" src="plots/spatial_enrichment/fold_distribution_chr_2_unlabelled.png
"/><img style="width:50.0%;" src="plots/spatial_enrichment/gene_distribution_chr_2.png
"/><img style="width:50.0%;" src="plots/spatial_enrichment/fold_distribution_chr_3_unlabelled.png
"/><img style="width:50.0%;" src="plots/spatial_enrichment/gene_distribution_chr_3.png
"/><img style="width:50.0%;" src="plots/spatial_enrichment/fold_distribution_chr_4_unlabelled.png
"/><img style="width:50.0%;" src="plots/spatial_enrichment/gene_distribution_chr_4.png
"/><img style="width:50.0%;" src="plots/spatial_enrichment/fold_distribution_chr_5_unlabelled.png
"/><img style="width:50.0%;" src="plots/spatial_enrichment/gene_distribution_chr_5.png
"/><img style="width:50.0%;" src="plots/spatial_enrichment/fold_distribution_chr_Mt_unlabelled.png
"/><img style="width:50.0%;" src="plots/spatial_enrichment/gene_distribution_chr_Mt.png
"/><img style="width:50.0%;" src="plots/spatial_enrichment/fold_distribution_chr_Pt_unlabelled.png
"/><img style="width:50.0%;" src="plots/spatial_enrichment/gene_distribution_chr_Pt.png
"/></div>

        <!-- plot_description is replaced by bin/plot_description/... -->
        <button class="collapsible">plot description</button>
        <div class="plot_description">Here we provide spatial enrichment distribution plots. These show for each chromosome: the location of each gene and its associated log2 fold change and significance level (left plots), and the overall distribution of significantly different compared to non-significantly different genes (right plots). These plots can be useful to visually confirm any observations shown in the spatial enrichment table.
</div>

        <!-- plot_legend is replaced by bin/figure_legends/... -->
        <button class="collapsible">figure legend</button>
        <div class="plot_legend"><p>Per chromosome differentially expressed genes spatial distribution plots. Showing for each chromosome, Left plots: a scatterplot of the genomic locations of each gene (y-axis) and its log2 fold change between between koc and kosalt. Significantly differential genes (p.adj < 0.05, absolute log2 fold > 1.0) are shown in red and non-significant genes in black. Right plots - density plot showing the density of genes by locus across the chromosome. The distribution of significantly differential genes is shown in red and non-significant genes in black.
</p></div>

        <!-- r_code is replaced by -->
        <button class="collapsible" id="r_code">r code</button>
        <div class="r_code"><span class="r_comment">##---- general libraries ----####</span><br><br>library(ggplot2)<br>library(ggrepel)<br>library(reshape)<br>library(ggridges)<br><br><span class="r_comment">##---- sets the working directory. If you move the SL2 folder please update this path ----####</span><br><br>setwd("/home/john/Downloads/mahesh_run2/2x75/SL2/all_genes/de_workflows/KOC_vs_KOSALT")<br><br><span class="r_comment">##---- de input files ----####</span><br><br>de_annotated = read.table(file="data/de_annotated.csv", header=TRUE,row.names = 2, sep='\t', quote='',check.names = TRUE)<br>de_annotated_sig = read.table(file="data/de_annotated_significant.csv", header=TRUE,row.names = 2, sep='\t', quote='',check.names = TRUE)<br><br><span class="r_comment">##---- SL2 Theme ----####</span><br><br>theme_SL2 <- function () { <br> &emsp; &emsp;theme_bw() %+replace% <br> &emsp; &emsp; &emsp; &emsp;theme(<br>	 &emsp; &emsp;panel.grid = element_blank(),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;panel.background = element_blank(),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;panel.border = element_rect(colour = "black", fill=NA, linewidth=1),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.background = element_blank(), <br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.background = element_rect(fill="transparent", colour=NA),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.key = element_rect(fill="transparent", colour=NA),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.title = element_text(size=12, margin = margin(b = 5),hjust=0,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;title = element_text(size = 12, margin = margin(b = 5),hjust=0,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.text.y = element_text(size = 10, margin = margin(r = 5),hjust=1,vjust=0.5, face="bold",colour="black"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.text.x = element_text(size = 10, margin = margin(t = 5),hjust=0.5,vjust=1, face="bold",colour="black"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.title.y = element_text(size = 11, margin = margin(r = 10),angle = 90,hjust=0.5,vjust=0.5, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;axis.title.x = element_text(size = 11, margin = margin(t = 10),hjust=0.5,vjust=1, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.text=element_text(size=11, face="bold"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.title=element_blank(), <br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;legend.key.size=unit(2.5,"line"),<br> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;plot.margin=unit(c(0.4,0.4,0.4,0.4), "cm")<br> &emsp; &emsp; &emsp; &emsp;)<br>}<br><br><span class="r_comment">##---- Default Three Way Significance Colours &emsp;----####</span><br><br>default_3way_sig_one_colour = "coral3"<br>default_3way_sig_two_colour = "cornflowerblue"<br>default_3way_non_sig_colour = "grey75"<br><br><span class="r_comment">##---- Save Plot Function -----####</span><br><br>save_plot <- function(ggp,plot_height,plot_width,plot_path)<br>{<br> &emsp;png(plot_path, height=plot_height, width=plot_width, pointsize=5)<br> &emsp;print(ggp)<br> &emsp;dev.off()<br><br> &emsp;svg(gsub(".png", ".svg", plot_path), height=plot_height/90, width=plot_width/90)<br> &emsp;print(ggp)<br> &emsp;dev.off()<br><br> &emsp;# clears all devices - as a safety measure<br> &emsp;while (dev.cur()>1) dev.off()<br>}<br><br><span class="r_comment">##---- Spatial Analysis Fold Distribution Function ----####</span><br><br>make_spatial_enrichment_fold_distribution_plot <- function(current_chromosome,sig_up_colour,sig_down_colour,non_sig_colour,dot_size,dot_transparency,sig_up_name,sig_down_name,non_sig_name,x_axis_label,y_axis_label,legend_position,data_label_size, show_data_labels)<br>{<br> &emsp;# parse the table<br> &emsp;chr_data = &emsp;subset(de_annotated,chromosome==current_chromosome)<br> &emsp;chr_data$direction = "c"<br> &emsp;chr_data_non_sig = subset(chr_data, de_valid=="True" & sig == "False")<br> &emsp;chr_data_sig_up = subset(chr_data, de_valid=="True" & sig == "True" & log2fold >0)<br> &emsp;chr_data_sig_down = subset(chr_data, de_valid=="True" & sig == "True" & log2fold <0)<br><br> &emsp;if (nrow(chr_data_sig_up) > 0)<br> &emsp;{<br> &emsp; &emsp;chr_data_sig_up$direction = "a"<br> &emsp;}<br> &emsp;if (nrow(chr_data_sig_down) > 0)<br> &emsp;{<br> &emsp; &emsp;chr_data_sig_down$direction = "b"<br> &emsp;}<br><br> &emsp;chr_data = rbind(chr_data_sig_up, chr_data_sig_down, chr_data_non_sig)<br><br> &emsp;# get labels<br> &emsp;if (show_data_labels == TRUE)<br> &emsp;{<br> &emsp; &emsp;chr_data_sig = subset(chr_data, sig == "True")<br> &emsp;}<br> &emsp;else<br> &emsp;{<br> &emsp; &emsp;chr_data_sig = chr_data[0,]<br> &emsp;}<br><br> &emsp;# make the plot<br> &emsp;if (nrow(chr_data) > 1)<br> &emsp;{<br> &emsp; &emsp;ggp = ggplot(data=chr_data, aes(x=start+((stop-start)/2), y=log2fold, colour=direction)) +<br> &emsp; &emsp; &emsp;geom_point(size=dot_size,alpha=dot_transparency) +<br> &emsp; &emsp; &emsp;geom_label(data=chr_data_sig,aes(label=rownames(chr_data_sig)), size=data_label_size, show.legend = FALSE) +<br> &emsp; &emsp; &emsp;scale_color_manual(breaks=c("a","b","c"), values=c(sig_up_colour,sig_down_colour, non_sig_colour),labels=c(sig_up_name,sig_down_name,non_sig_name)) +<br> &emsp; &emsp; &emsp;ylim(c(-max(abs(chr_data$log2fold))*1.1,max(abs(chr_data$log2fold))*1.1)) +<br> &emsp; &emsp; &emsp;labs(x=x_axis_label, y=y_axis_label, title=paste("chromosome ",current_chromosome,sep="")) +<br> &emsp; &emsp; &emsp;theme_SL2() +<br> &emsp; &emsp; &emsp;theme(legend.position=legend_position, legend.title = element_blank())<br><br> &emsp; &emsp;return(ggp)<br> &emsp;}<br> &emsp;else<br> &emsp;{<br> &emsp; &emsp;return(ggplot(data.frame()) + theme_SL2() + geom_blank() + ggtitle("There were too few genes to plot this."))<br> &emsp;}<br>}<br><br><span class="r_comment">##---- Spatial Enrichment Gene Distribution Function ----####</span><br><br>make_spatial_enrichment_gene_distribution_plot <- function(current_chromosome,sig_up_colour,sig_down_colour,non_sig_colour,density_curve_transparency,density_curve_line_thickness,sig_up_name,sig_down_name,non_sig_name,x_axis_label,y_axis_label,legend_position)<br>{<br><br> &emsp;# parse the table<br> &emsp;chr_data = &emsp;subset(de_annotated,chromosome==current_chromosome)<br> &emsp;chr_data$direction = "c"<br> &emsp;chr_data_non_sig = subset(chr_data, de_valid=="True" & sig == "False")<br> &emsp;chr_data_sig_up = subset(chr_data, de_valid=="True" & sig == "True" & log2fold >0)<br> &emsp;chr_data_sig_down = subset(chr_data, de_valid=="True" & sig == "True" & log2fold <0)<br><br> &emsp;if (nrow(chr_data_sig_up) > 0)<br> &emsp;{<br> &emsp; &emsp;chr_data_sig_up$direction = "a"<br> &emsp;}<br> &emsp;if (nrow(chr_data_sig_down) > 0)<br> &emsp;{<br> &emsp; &emsp;chr_data_sig_down$direction = "b"<br> &emsp;}<br><br> &emsp;chr_data = rbind(chr_data_sig_up, chr_data_sig_down, chr_data_non_sig)<br><br> &emsp;# make the plot<br> &emsp;if (nrow(chr_data) > 1)<br> &emsp;{<br> &emsp; &emsp;ggp = ggplot(chr_data, aes(x=start+((stop-start)/2), group=direction, fill=direction, colour = direction)) +<br> &emsp; &emsp; &emsp;geom_density(data = chr_data_non_sig, alpha=density_curve_transparency,size=density_curve_line_thickness, adjust=0.25) +<br> &emsp; &emsp; &emsp;geom_density(data = chr_data_non_sig, aes( y = -(..density..)), alpha=density_curve_transparency,size=density_curve_line_thickness, adjust=0.25) +<br> &emsp; &emsp; &emsp;geom_density(data = chr_data_sig_up, alpha=density_curve_transparency,size=density_curve_line_thickness, adjust=0.25) +<br> &emsp; &emsp; &emsp;geom_density(data = chr_data_sig_down, aes( y = -(..density..)), alpha=density_curve_transparency,size=density_curve_line_thickness, adjust=0.25) +<br> &emsp; &emsp; &emsp;scale_fill_manual(breaks=c("a","b","c"), values=c(sig_up_colour,sig_down_colour, non_sig_colour),labels=c(sig_up_name,sig_down_name,non_sig_name)) +<br> &emsp; &emsp; &emsp;scale_colour_manual(breaks=c("a","b","c"), values=c(sig_up_colour,sig_down_colour, non_sig_colour),labels=c(sig_up_name,sig_down_name,non_sig_name)) +<br> &emsp; &emsp; &emsp;labs(x=x_axis_label, y=y_axis_label, &emsp;title=paste("chromosome ",current_chromosome,sep="")) +<br> &emsp; &emsp; &emsp;theme_SL2() +<br> &emsp; &emsp; &emsp;theme(legend.position=legend_position, legend.title = element_blank())<br><br> &emsp; &emsp;return(ggp)<br> &emsp;}<br> &emsp;else<br> &emsp;{<br> &emsp; &emsp;return(ggplot(data.frame()) + theme_SL2() + geom_blank() + ggtitle("There were too few &emsp;genes to plot this."))<br> &emsp;}<br>}<br><br><span class="r_comment">##---- Spatial Enrichment Distribution Plots ----####</span><br><br>plot_height = 300<br>plot_width = 750<br>sig_up_colour = default_3way_sig_one_colour<br>sig_down_colour = default_3way_sig_two_colour<br>non_sig_colour = default_3way_non_sig_colour<br>dot_size = 1.5<br>dot_transparency = 1<br>data_label_size = 3.5<br>show_data_labels = TRUE<br>sig_up_name = "upregulated"<br>sig_down_name = "downregulated"<br>non_sig_name = "non-significant"<br>x_axis_label = "gene location"<br>y_axis_label = expression("log"[2]* " fold change")<br>legend_position = "bottom"<br>density_curve_transparency = 0.75<br>density_curve_line_thickness = 0<br><br><br>chromosomes = unique(factor(de_annotated$chromosome))<br><br>for (chromosome in chromosomes)<br>{<br>show_data_labels = TRUE<br>ggp = make_spatial_enrichment_fold_distribution_plot(chromosome,sig_up_colour,sig_down_colour,non_sig_colour,dot_size,dot_transparency,sig_up_name,sig_down_name,non_sig_name,x_axis_label,y_axis_label,legend_position,data_label_size, show_data_labels)<br>save_plot(ggp,plot_height,plot_width,paste("plots/spatial_enrichment/fold_distribution_chr_",chromosome,"_labelled.png",sep=""))<br><br>show_data_labels = FALSE<br>ggp = make_spatial_enrichment_fold_distribution_plot(chromosome,sig_up_colour,sig_down_colour,non_sig_colour,dot_size,dot_transparency,sig_up_name,sig_down_name,non_sig_name,x_axis_label,y_axis_label,legend_position,data_label_size, show_data_labels)<br>save_plot(ggp,plot_height,plot_width,paste("plots/spatial_enrichment/fold_distribution_chr_",chromosome,"_unlabelled.png",sep=""))<br>}<br><br>x_axis_label = "gene location"<br>y_axis_label = "gene density"<br><br>for (chromosome in chromosomes)<br>{<br>ggp = make_spatial_enrichment_gene_distribution_plot(chromosome,sig_up_colour,sig_down_colour,non_sig_colour,density_curve_transparency,density_curve_line_thickness,sig_up_name,sig_down_name,non_sig_name,x_axis_label,y_axis_label,legend_position)<br>save_plot(ggp,plot_height,plot_width,paste("plots/spatial_enrichment/gene_distribution_chr_",chromosome,".png",sep=""))<br>}<br><br></div>
    </div>
</div>

<div class=text id=Core Workflow><h2>Core Workflow</h2><div>This pairwise differential expression (de) workfow report was generated using Searchlight 2 (v2.0.3). The normalised expression matrix at /home/john/Downloads/mahesh_run2/2x75/EM_2x75.txt was combined with the background file at /home/john/Downloads/mahesh_run2/BACKGROUND.txt, retaining only the 32833 genes that were present in both, to create the "expression set". Per sample group mean expression values and standard devitations per gene were calulated using numpy.

<br><br>

Next the differential expression table at /home/john/Downloads/mahesh_run2/2x75/DM_KOCvKOSalt_2x75.txt which compares the two sample groups koc and kosalt, was combined with the expression set, retaining only the 22597 genes that were present in both, to create the "differential expression set". Next a list of significantly differentially expressed genes was generated from the differential expression set using the adjusted p threshold of 0.05 and the absolute log2fold theshold of > 1.0. Finally, all plots were generated as described in the figure legends of this report, using the R script at /de/KOC vs KOSALT/plots/workflow.r, and as described in the figure legends of this report.

</div></div><div class=text id=Spatial Analysis><h2>Spatial Analysis</h2><div>The spatial enrichment analysis provides three statistical measures for spatial bias. 1) Bias towards gene expression at a given chromosome. The p value is calulated using a fishers exact test as: number of expression set genes in chromosome, number of expressed genes in chromosome vs number of  expression set genes in genome ,number of expressed genes in geneome. 2) Bias towards differential expression at a given chromosome. The p value is calulated using a fishers exact test as: number of differential expression set genes in chromosome, number of differentially expressed genes in chromsome, vs number of differential expression set genes in genome, number of differentially expressed genes in genome. 3) Bias towards either up or downregulation at a given chromosome. The p value is calulated using a fishers exact test as: expected number of upregulated genes at a given chromosome, the number of upregulated genes at a given chromosome, expected number of downregulated genes at a given chromosome, the number of downregulated genes at a given chromosome.
</div></div><div class=text id=Over Representation Analysis><h2>Over Representation Analysis</h2><div>To perform Over Representation Analysis (ORA), firstly the STRING 11.5 database was filtered to include only genes that were also in both the normalised expression file and the background file (the expression set). Next the database was filtered to include only gene sets with between 5.0 and 250.0 valid genes. For each set of candidate genes (i.e. all significant, all upregulated and all downregulated genes) the number of overlapping genes with each gene set in the database was determined. Next a hypergeometric test was performed for each candidate set and gene set combination using the number of genes in the expression set as the background size. To correct for multisampling a Benjamini-Hochberg correction was applied. Gene sets with an adjusted p-value of 0.05 and an absolute log2fold enrichment above 0.0 were considered significant.

</div></div><div class=text id=Citation><h2>Citation</h2><div>Cole, J.J., Faydaci, B.A., McGuinness, D. et al. Searchlight: automated bulk RNA-seq exploration and visualisation using dynamically generated R scripts. BMC Bioinformatics 22, 411 (2021). https://doi.org/10.1186/s12859-021-04321-2
https://github.com/Searchlight2/Searchlight2/
</div></div></div>
    </div>
    <!-- end of report -->

    <script src="https://code.jquery.com/jquery-1.10.2.js"></script>
<script>
$(".collapsible").on("click", function () {
    var content = this.nextElementSibling;
    this.classList.toggle("active");
    if (content.style.display == "block") {
        content.style.display = "none";
    } else {
        content.style.display = "block";
    }
    });

$(".plot_title").on("click", function () {
    var content = this.nextElementSibling;
    this.classList.toggle("active");
    if (content.style.display == "block") {
        content.style.display = "none";
    } else {
        content.style.display = "block";
    }
    });


</script>
</body>
</html>
